diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/Makefile	2017-05-01 22:29:00.150393195 +0200
@@ -134,7 +134,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/Makefile	2017-05-01 22:29:26.857682453 +0200
@@ -135,7 +135,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/msvc-generate/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/msvc-generate/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/msvc-generate/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/build/msvc/msvc-generate/Makefile	2017-05-01 22:31:42.860016638 +0200
@@ -98,7 +98,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/Makefile	2017-05-01 22:31:42.864017012 +0200
@@ -135,7 +135,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/rpm/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/rpm/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/rpm/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/distro/rpm/Makefile	2017-05-01 22:31:42.860016638 +0200
@@ -98,7 +98,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/doc/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/doc/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/doc/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/doc/Makefile	2017-05-01 22:31:42.864017012 +0200
@@ -132,7 +132,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/include/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/include/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/include/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/include/Makefile	2017-05-01 22:31:42.868017386 +0200
@@ -128,7 +128,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/Makefile	2016-10-11 17:53:09.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/Makefile	2017-05-01 22:28:31.718673776 +0200
@@ -211,7 +211,7 @@
 AWK = gawk
 CC = $(CROSS)gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = $(CROSS)gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
@@ -834,7 +834,7 @@
 	done
 install: $(BUILT_SOURCES)
 	#$(MAKE) $(AM_MAKEFLAGS) install-recursive
-	$(STRIP) src/openvpn/openvpn
+	$(SSTRIP) src/openvpn/openvpn
 	cp src/openvpn/openvpn ../../target/sbin
 	chmod +x easy-rsa/EasyRSA-2.2.2/*
 	cp -af easy-rsa/EasyRSA-2.2.2/keys/*  ../../target/usr/openvpn/.
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/sample/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/sample/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/sample/Makefile	2016-10-11 17:53:07.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/sample/Makefile	2017-05-01 22:31:42.868017386 +0200
@@ -125,7 +125,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/compat/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/compat/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/compat/Makefile	2016-10-11 17:53:07.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/compat/Makefile	2017-05-01 22:31:42.868017386 +0200
@@ -117,7 +117,7 @@
 AWK = gawk
 CC = $(CROSS)gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = $(CROSS)gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/Makefile	2017-05-01 22:31:42.880018508 +0200
@@ -135,7 +135,7 @@
 AWK = gawk
 CC = $(CROSS)gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = $(CROSS)gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpn/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpn/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpn/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpn/Makefile	2017-05-01 22:31:42.872017760 +0200
@@ -188,7 +188,7 @@
 AWK = gawk
 CC = $(CROSS)gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = $(CROSS)gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpnserv/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpnserv/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpnserv/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/openvpnserv/Makefile	2017-05-01 22:31:42.872017760 +0200
@@ -130,7 +130,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/auth-pam/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/auth-pam/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/auth-pam/Makefile	2016-10-11 17:53:07.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/auth-pam/Makefile	2017-05-01 22:48:20.137209724 +0200
@@ -150,7 +150,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS = -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/down-root/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/down-root/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/down-root/Makefile	2016-10-11 17:53:07.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/down-root/Makefile	2017-05-01 22:48:21.425282804 +0200
@@ -149,7 +149,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS = -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/Makefile	2016-10-11 17:53:07.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/src/plugins/Makefile	2017-05-01 22:31:42.880018508 +0200
@@ -135,7 +135,7 @@
 AWK = gawk
 CC = $(CROSS)gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = $(CROSS)gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
diff -urN D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/tests/Makefile D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/tests/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/tests/Makefile	2016-10-11 17:53:08.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/openvpn-2.3.2/tests/Makefile	2017-05-01 22:31:42.884018882 +0200
@@ -100,7 +100,7 @@
 AWK = gawk
 CC = gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2
+CFLAGS += -Os
 CPP = gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
