--- work/bridge-utils-1.7.1/bridge-utils-1.7.1/configure.ac
+++ work/bridge-utils-1.7.1/bridge-utils-1.7.1/configure.ac
@@ -1,5 +1,5 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT([bridge-utils],[1.7])
+AC_INIT([bridge-utils],[1.7.1])
 AC_CONFIG_HEADERS(libbridge/config.h)
 
 AC_ARG_WITH([linux-headers],
@@ -28,5 +28,5 @@ AC_CHECK_FUNCS(strlcpy strlcat)
 
 AC_SUBST(KERNEL_HEADERS)
 
-AC_CONFIG_FILES([doc/Makefile libbridge/Makefile brctl/Makefile Makefile bridge-utils.spec])
+AC_CONFIG_FILES([libbridge/Makefile brctl/Makefile Makefile])
 AC_OUTPUT
