diff -urN D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/libevent/Makefile D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/libevent/Makefile
--- D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/libevent/Makefile	2016-10-11 17:57:14.000000000 +0200
+++ D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/libevent/Makefile	2017-05-01 21:51:00.491113358 +0200
@@ -13,7 +13,7 @@
 OBJS=event.o event_tcp.o hash.o event_sig.o
 SRCS=$(OBJS:.o=.c)
 HDRS=event.h event_tcp.h eventpriv.h hash.h
-CFLAGS=-g -O2 -Wall -Wstrict-prototypes -ansi -pedantic -I..
+CFLAGS +=  -Os -Wall -Wstrict-prototypes -ansi -pedantic -I..
 
 all: libevent.a
 
diff -urN D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/Makefile D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/Makefile
--- D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/Makefile	2016-10-11 17:57:14.000000000 +0200
+++ D7000_V1.0.1.50_WW_src/Source/apps/rp-pppoe/Makefile	2017-05-01 22:08:23.438772953 +0200
@@ -60,7 +60,7 @@
 	'-DPLUGIN_PATH="$(PLUGIN_PATH)"' \
 	'-DPPPOE_SERVER_OPTIONS="$(PPPOESERVER_PPPD_OPTIONS)"'
 
-CFLAGS= -s -Os -Wall -Werror -Wstrict-prototypes -Wno-unused-but-set-variable -ansi  -D_GNU_SOURCE $(LIC_INCDIR) $(DEFINES) $(LIC_DEFINE) $(PATHS) -Ilibevent
+CFLAGS += -s -Os -Wall -Werror -Wstrict-prototypes -Wno-unused-but-set-variable -ansi  -D_GNU_SOURCE $(LIC_INCDIR) $(DEFINES) $(LIC_DEFINE) $(PATHS) -Ilibevent
 TARGETS=pppoe #pppoe-server pppoe-sniff pppoe-relay
 PPPOE_SERVER_LIBS=$(LIC_LIBDIR) $(LIC_LIB)
 
@@ -79,11 +79,11 @@
 pppoe: pppoe.o common.o ppp.o discovery.o if.o relay.o common.o#debug.o 
 	$(CROSS)gcc -s -o pppoe $^ 
 	ln -sf pppoe pppoe-relay
-	$(CROSS)strip pppoe
+	$(SSTRIP) pppoe
 
 pppoe-relay: relay.o if.o common.o #debug.o
 	$(CROSS)gcc -o pppoe-relay relay.o if.o common.o #debug.o common.o
-	$(CROSS)strip pppoe-relay
+	$(SSTRIP) pppoe-relay
 
 pppoe.o: pppoe.c pppoe.h
 	$(CROSS)gcc $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o pppoe.o pppoe.c
