diff -urN work/libarchive-3.3.1/Makefile work/libarchive-3.3.1/Makefile
--- work/libarchive-3.3.1/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ work/libarchive-3.3.1/Makefile	2017-03-23 20:35:44.381701657 +0100
@@ -0,0 +1,33 @@
+include ../Rules.mak
+
+all: config
+	$(MAKE) -C $(LIBARCHIVE)
+	$(STRIP) $(LIBARCHIVE)/.libs/*.so.*
+	$(STRIP) $(STRIPFLAGS) $(LIBARCHIVE)/.libs/*.a
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(LIBARCHIVE) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) --disable-bsdtar --disable-bsdcat --disable-bsdcpio CFLAGS="$(CFLAGS) -Os -s"
+
+install:
+	@echo "Nothing to install..."
+#	$(SSTRIP) $(LIBARCHIVE)/bsdtar
+#	$(SSTRIP) $(LIBARCHIVE)/bsdcat
+#	$(SSTRIP) $(LIBARCHIVE)/bsdcpio
+#	install -m 0755 -d $(LIBDIR)
+#	install -m 0755 -d $(BINDIR)
+#	cp -rf $(LIBARCHIVE)/.libs/*.so.* $(LIBDIR)
+#	cp -rf $(LIBARCHIVE)/bsdtar $(BINDIR)/tar
+
+clean:
+	$(MAKE) -C $(LIBARCHIVE) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(LIBARCHIVE) distclean
+
