diff -urN D7000_V1.0.1.44_WW_src/Source/apps/smtpc/Makefile D7000_V1.0.1.44_WW_src/Source/apps/smtpc/Makefile
--- D7000_V1.0.1.44_WW_src/Source/apps/smtpc/Makefile	2016-10-11 17:56:25.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/apps/smtpc/Makefile	2017-04-29 23:47:54.774184504 +0200
@@ -1,9 +1,8 @@
 include ../../Rules.mak
 
 CC=$(CROSS)gcc
-#CC=gcc
 BIN=smtpc
-CFLAGS = -Wall -Werror -Os -s
+CFLAGS += -Wall -Werror -Os -s
 CFLAGS += -I$(SHARED_DIR)
 CFLAGS += -DBOARD_ID=\"$(PROJECT)\"
 
@@ -21,4 +20,5 @@
 	rm -f $(BIN)
 	rm -f ../../target/usr/sbin/smtpc 
 install:
+	$(SSTRIP) $(BIN)
 	cp -a smtpc ../../target/usr/sbin
