diff -urN work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/Makefile.in work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/Makefile.in
--- work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/Makefile.in	2015-07-20 19:30:24.000000000 +0200
+++ work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/Makefile.in	2017-03-23 21:20:45.828649220 +0100
@@ -209,7 +209,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = dist-bzip2 dist-zip
-SUBDIRS = src doc
+SUBDIRS = src #doc
 EXTRA_DIST = README.* Release_Notes inet_ntop.c autogen.sh m4
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
diff -urN work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/src/Makefile.in work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/src/Makefile.in
--- work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/src/Makefile.in	2015-07-20 19:30:24.000000000 +0200
+++ work/netperf-netperf-2.7.0/netperf-netperf-2.7.0/src/Makefile.in	2017-03-23 21:20:45.828649220 +0100
@@ -35,7 +35,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-bin_PROGRAMS = netperf$(EXEEXT) netserver$(EXEEXT)
+bin_PROGRAMS = netperf$(EXEEXT) #netserver$(EXEEXT)
 subdir = src
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in \
 	$(srcdir)/netperf_version.h.in
@@ -72,7 +72,7 @@
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
-SOURCES = $(netperf_SOURCES) $(netserver_SOURCES)
+SOURCES = $(netperf_SOURCES) #$(netserver_SOURCES)
 DIST_SOURCES = $(netperf_SOURCES) $(netserver_SOURCES)
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
 	html-recursive info-recursive install-data-recursive \
@@ -215,8 +215,8 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = missing
-CLEANFILES = netperf_version.h
+SUBDIRS = #missing
+CLEANFILES = #netperf_version.h
 AM_CFLAGS = $(NETPERF_CFLAGS)
 USE_CPU_SOURCE = netcpu_@NETCPU_SOURCE@.c
 EXTRA_DIST = netcpu_none.c netcpu_looper.c netcpu_pstat.c netcpu_pstatnew.c netcpu_perfstat.c netcpu_procstat.c netcpu_kstat.c netcpu_kstat10.c netcpu_sysctl.c netcpu_ntperf.c netcpu_osx.c dirs NetPerfDir/* NetServerDir/* netperf_version.h.in Makefile.uw
