diff -urN D7000_V1.0.1.50_WW_src/Source/apps/udhcp-0.9.7/Makefile D7000_V1.0.1.50_WW_src/Source/apps/udhcp-0.9.7/Makefile
--- D7000_V1.0.1.50_WW_src/Source/apps/udhcp-0.9.7/Makefile	2016-10-11 17:57:17.000000000 +0200
+++ D7000_V1.0.1.50_WW_src/Source/apps/udhcp-0.9.7/Makefile	2017-05-01 21:49:32.197798478 +0200
@@ -14,11 +14,10 @@
 #DEBUG=1
 
 # Uncomment this to output messages to syslog, otherwise, messages go to stdout
-CFLAGS += -Os -s -Wall
+CFLAGS +=  -Os -s -Wall
 CFLAGS += -Werror -Wno-strict-aliasing -Wno-unused-but-set-parameter -Wno-unused-but-set-variable
 
 include ../../Rules.mak
-#CROSS_COMPILE=mips-uclibc-
 CROSS_COMPILE=$(CROSS)
 CC = $(CROSS_COMPILE)gcc
 LD = $(CROSS_COMPILE)gcc
@@ -108,6 +107,7 @@
 	$(LD) $(STRIP_FLAG) $(OBJS3) -o $(EXEC3)
 
 install: 
+	$(SSTRIP) udhcpd
 	rm -f udhcpc
 	ln -s udhcpd udhcpc
 	cp -a udhcpd ../../target/usr/sbin
