diff -urN work/openvpn-2.4.8/openvpn-2.4.8/src/openvpn/openssl_compat.h work/openvpn-2.4.8/openvpn-2.4.8/src/openvpn/openssl_compat.h
--- work/openvpn-2.4.8/openvpn-2.4.8/src/openvpn/openssl_compat.h	1970-01-01 01:00:00.000000000 +0100
+++ work/openvpn-2.4.8/openvpn-2.4.8/src/openvpn/openssl_compat.h	2018-01-05 12:00:11.000000000 +0100
@@ -693,6 +693,18 @@
 #define RSA_F_RSA_OSSL_PRIVATE_ENCRYPT       RSA_F_RSA_EAY_PRIVATE_ENCRYPT
 #endif
 
+#ifndef SSL_OP_NO_TLSv1
+#define SSL_OP_NO_TLSv1			0x04000000
+#endif
+
+#ifndef SSL_OP_NO_TLSv1_1
+#define SSL_OP_NO_TLSv1_1		0x10000000
+#endif
+
+#ifndef SSL_OP_NO_TLSv1_2
+#define SSL_OP_NO_TLSv1_2		0x08000000
+#endif
+
 #ifndef SSL_CTX_get_min_proto_version
 /** Return the min SSL protocol version currently enabled in the context.
  *  If no valid version >= TLS1.0 is found, return 0. */
