diff -urN D7000_V1.0.1.50_WW_src/Source/apps/nvram/Makefile D7000_V1.0.1.50_WW_src/Source/apps/nvram/Makefile
--- D7000_V1.0.1.50_WW_src/Source/apps/nvram/Makefile	2016-10-11 17:53:07.000000000 +0200
+++ D7000_V1.0.1.50_WW_src/Source/apps/nvram/Makefile	2017-05-02 12:33:28.031781668 +0200
@@ -1,7 +1,7 @@
 include ../../Rules.mak
 CC=$(CROSS)gcc
 
-CFLAGS  = -Os -s -Wall
+CFLAGS += -Os -s -Wall
 CFLAGS += -Werror
 CFLAGS += -DTEST
 
@@ -14,17 +14,17 @@
 
 BINS = libscnvram.so 
 
-#all: libnvram.so nvram_static
 all: $(BINS)
 
 libscnvram.so : nvram.o nvram_lock.o
-	$(CC) $(CFLAGS) -Os -s -shared -Wl,-soname,libscnvram.so -o libscnvram.so $^ 
+	$(CC) $(CFLAGS) -shared -Wl,-soname,libscnvram.so -o libscnvram.so $^ 
+	$(STRIP) libscnvram.so
 
 nvram.o : nvram.c
-	$(CC) $(CFLAGS) -Os -s -Wall -Werror -fPIC -g -c $^ -o $@
+	$(CC) $(CFLAGS) -fPIC -c $^ -o $@
 
 nvram_lock.o: nvram_lock.c
-	$(CC) $(CFLAGS) -Os -s -g -Wall -fPIC -c $^ -o $@
+	$(CC) $(CFLAGS) -fPIC -c $^ -o $@
 
 install :
 	cp -f libscnvram.so $(DESTLIB)
@@ -33,5 +33,3 @@
 	rm -rf *~ *.o *.so nvram $(BINS)
 	-rm -rf $(DESTLIB)/libscnvram.so
 
-
-
