diff -urN work/openssh-7.9p1/openssh-7.9p1/progressmeter.c work/openssh-7.9p1/openssh-7.9p1/progressmeter.c
--- work/openssh-7.9p1/openssh-7.9p1/progressmeter.c	1970-01-01 01:00:00.000000000 +0100
+++ work/openssh-7.9p1/openssh-7.9p1/progressmeter.c	2018-01-05 12:00:11.000000000 +0100
@@ -59,9 +59,6 @@ static void format_rate(char *, int, off_t);
 static void sig_winch(int);
 static void setscreensize(void);
 
-/* updates the progressmeter to reflect the current state of the transfer */
-void refresh_progress_meter(void);
-
 /* signal handler for updating the progress meter */
 static void sig_alarm(int);
 
@@ -120,7 +117,7 @@ format_size(char *buf, int size, off_t bytes)
 }
 
 void
-refresh_progress_meter(void)
+refresh_progress_meter(int force_update)
 {
 	char buf[MAX_WINSIZE + 1];
 	off_t transferred;
@@ -131,7 +128,7 @@ refresh_progress_meter(void)
 	int hours, minutes, seconds;
 	int file_len;
 
-	if ((!alarm_fired && !win_resized) || !can_output())
+	if ((!force_update && !alarm_fired && !win_resized) || !can_output())
 		return;
 	alarm_fired = 0;
 
@@ -254,7 +251,7 @@ start_progress_meter(const char *f, off_t filesize, off_t *ctr)
 	bytes_per_second = 0;
 
 	setscreensize();
-	refresh_progress_meter();
+	refresh_progress_meter(1);
 
 	signal(SIGALRM, sig_alarm);
 	signal(SIGWINCH, sig_winch);
@@ -271,7 +268,7 @@ stop_progress_meter(void)
 
 	/* Ensure we complete the progress */
 	if (cur_pos != end_pos)
-		refresh_progress_meter();
+		refresh_progress_meter(1);
 
 	atomicio(vwrite, STDOUT_FILENO, "\n", 1);
 }
diff -urN work/openssh-7.9p1/openssh-7.9p1/progressmeter.h work/openssh-7.9p1/openssh-7.9p1/progressmeter.h
--- work/openssh-7.9p1/openssh-7.9p1/progressmeter.h	1970-01-01 01:00:00.000000000 +0100
+++ work/openssh-7.9p1/openssh-7.9p1/progressmeter.h	2018-01-05 12:00:11.000000000 +0100
@@ -24,5 +24,5 @@
  */
 
 void	start_progress_meter(const char *, off_t, off_t *);
-void	refresh_progress_meter(void);
+void	refresh_progress_meter(int);
 void	stop_progress_meter(void);
diff -urN work/openssh-7.9p1/openssh-7.9p1/scp.c work/openssh-7.9p1/openssh-7.9p1/scp.c
--- work/openssh-7.9p1/openssh-7.9p1/scp.c	1970-01-01 01:00:00.000000000 +0100
+++ work/openssh-7.9p1/openssh-7.9p1/scp.c	2018-01-05 12:00:11.000000000 +0100
@@ -585,7 +585,7 @@ scpio(void *_cnt, size_t s)
 	off_t *cnt = (off_t *)_cnt;
 
 	*cnt += s;
-	refresh_progress_meter();
+	refresh_progress_meter(0);
 	if (limit_kbps > 0)
 		bandwidth_limit(&bwlimit, s);
 	return 0;
diff -urN work/openssh-7.9p1/openssh-7.9p1/sftp-client.c work/openssh-7.9p1/openssh-7.9p1/sftp-client.c
--- work/openssh-7.9p1/openssh-7.9p1/sftp-client.c	1970-01-01 01:00:00.000000000 +0100
+++ work/openssh-7.9p1/openssh-7.9p1/sftp-client.c	2018-01-05 12:00:11.000000000 +0100
@@ -101,7 +101,7 @@ sftpio(void *_bwlimit, size_t amount)
 {
 	struct bwlimit *bwlimit = (struct bwlimit *)_bwlimit;
 
-	refresh_progress_meter();
+	refresh_progress_meter(0);
 	if (bwlimit != NULL)
 		bandwidth_limit(bwlimit, amount);
 	return 0;
