--- work/transmission-3.00/transmission-3.00/web/javascript/common.js
+++ work/transmission-3.00/transmission-3.00/web/javascript/common.js
@@ -151,7 +151,7 @@ Prefs.setValue = function (key, val) {
 
     var date = new Date();
     date.setFullYear(date.getFullYear() + 1);
-    document.cookie = key + "=" + val + "; expires=" + date.toGMTString() + "; path=/";
+    document.cookie = key + "=" + val + "; expires=" + date.toGMTString() + "; path=/; SameSite=Lax";
 };
 
 /**
