diff -urN D7000_V1.0.1.74_WW_src/Source/apps/bftpd-2.3/Makefile D7000_V1.0.1.74_WW_src/Source/apps/bftpd-2.3/Makefile
--- D7000_V1.0.1.74_WW_src/Source/apps/bftpd-2.3/Makefile	2016-10-11 17:54:31.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/bftpd-2.3/Makefile	2017-05-01 22:16:00.683106402 +0200
@@ -3,7 +3,7 @@
 # Generated automatically from Makefile.in by configure.
 VERSION=2.2.1
 INSTALL=/usr/bin/install -c
-CFLAGS=-g -O2 -DHAVE_CONFIG_H -Wall -I. -DVERSION=\"$(VERSION)\" -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE 
+CFLAGS += -fcommon -Os -DHAVE_CONFIG_H -Wall -I. -DVERSION=\"$(VERSION)\" -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE 
 LIBS= -lcrypt
 prefix=/usr
 mandir=${prefix}/man
@@ -37,7 +37,7 @@
 bftpd: $(OBJS)
 	./mksources $(DIRPAX)
 	$(CC) $(OBJS2LINK) $(LDFLAGS) $(LIBS) -o bftpd
-	$(CROSS)strip $@
+	$(SSTRIP) $@
 
 $(OBJS): $(HEADERS) Makefile
 
@@ -48,9 +48,6 @@
 else
 	cp -f bftpd.fw.conf ../../target/usr/etc/bftpd.conf.sample
 endif
-#	cp -f ftpdpassword ../../target/usr/etc
-#	install -D bftpd.conf $(INSTALL_DIR)/etc/bftpd.conf
-#	install -D ftpdpassword $(INSTALL_DIR)/etc/ftpdpassword
 
 clean distclean:
 	rm -f *~ $(OBJS) bftpd mksources.finished config.cache
