--- D7000_V1.0.1.84_WW_src/Source/apps/utelnetd-0.1.9/rc_debug_mode
+++ D7000_V1.0.1.84_WW_src/Source/apps/utelnetd-0.1.9/rc_debug_mode
@@ -0,0 +1,72 @@
+#!/bin/sh
+#
+# ancistrus
+#
+# Netgear's D7000 (V1) Nighthawk Router Experience Distributed Project
+#
+# https://github.com/negan07/ancistrus
+#
+# License: GPLv2
+#
+#
+# Rc script for utelnetd daemon. Originally it linked '${APPDIR}/rc_apps' to run 'telnetenabled' .
+#
+# Usage: $0 < start restart stop authhostrenew >
+#
+# This script is invoked at every boot inside 'rc start' service.
+#
+
+stop() {
+	# Kill all clients
+	for PID in `pidof utelnetd`
+	do
+	[ -e /proc/${PID}/stat ] && while kill -9 $PID >/dev/null 2>&1; do :; done
+	done
+}
+
+start() {
+	# Checking if the service is already running
+	[ ! -z "`pidof utelnetd`" ] && exit 1
+
+	# Starting utelnetd
+	${UTELNETD} -l ${BINDIR}/login -d
+
+	# Sanity check
+	TELD_RC=$?
+	[ $TELD_RC -ne 0 ] && exit $TELD_RC
+}
+
+authhostrenew() {
+	# Update hostname
+	eval `nvram get device_name` >/dev/null 2>&1
+	[ ! -z "${device_name}" ] && hostname ${device_name} >/dev/null 2>&1 || hostname D7000 >/dev/null 2>&1
+
+	# Renew auth login
+	# rc_debug_mode is run every time password is changed from the web gui so renew /etc/passwd too
+	eval `nvram get http_password` >/dev/null 2>&1
+	for U in root nobody; do echo "${U}:${http_password}" | chpasswd >/dev/null 2>&1; done
+}
+
+# Main dirs
+BINDIR=/usr/sbin
+
+UTELNETD=${BINDIR}/utelnetd
+
+case $1 in
+start|restart)
+stop
+authhostrenew
+eval `nvram get telnet_enable` >/dev/null 2>&1
+eval `nvram get fw_remote` >/dev/null 2>&1
+[ "${telnet_enable}" != "0" ] && [ "${fw_remote}" = "0" ] && [ -x ${UTELNETD} ] && start
+;;
+stop)
+stop
+;;
+authhostrenew)
+authhostrenew
+;;
+*)
+echo "Usage: $0 < start restart stop authhostrenew >"
+;;
+esac
--- D7000_V1.0.1.84_WW_src/Source/apps/utelnetd-0.1.9/Makefile
+++ D7000_V1.0.1.84_WW_src/Source/apps/utelnetd-0.1.9/Makefile
@@ -60,6 +60,7 @@
 .PHONY: install
 install: $(PROGS)
 	cp utelnetd ../../target/usr/sbin
+	install -m 0755 rc_debug_mode ../../target/usr/sbin/rc_app
 
 .PHONY: clean
 clean:
