diff -urN D7000_V1.0.1.70_WW_src/Source/apps/utelnetd-0.1.9/Makefile D7000_V1.0.1.70_WW_src/Source/apps/utelnetd-0.1.9/Makefile
--- D7000_V1.0.1.70_WW_src/Source/apps/utelnetd-0.1.9/Makefile	2016-10-11 17:54:30.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Source/apps/utelnetd-0.1.9/Makefile	2017-05-01 22:09:25.765171623 +0200
@@ -36,10 +36,10 @@
 
 INSTALL   = install
 
-CFLAGS	 += -I. -pipe -DSHELLPATH=\"$(LOGIN)\" -Wall -Werror -Wno-unused-but-set-variable
+CFLAGS += -I. -pipe -DSHELLPATH=\"$(LOGIN)\" -Wall -Werror -Wno-unused-but-set-variable -Os
 
 ifneq ("","$(DEBUG)")
-CFLAGS   += -DDEBUG -g -Os
+CFLAGS   += -DDEBUG -g
 STRIP	  = \#
 else
 CFLAGS	 += -fomit-frame-pointer
@@ -55,7 +55,7 @@
 
 $(PROGS): $(OBJS)
 	$(CC) $(LDFLAGS) $^ $(LDLIBS) -o $@
-	$(STRIP) --remove-section=.comment --remove-section=.note $@
+	$(SSTRIP) $@
 
 .PHONY: install
 install: $(PROGS)
@@ -65,3 +65,4 @@
 clean:
 	rm -f $(PROGS) *.o $(CORE)
 	rm -f ../../target/usr/sbin/utelnetd 
+
