diff -urN work/libarchive-3.5.0/libarchive-3.5.0/libarchive/test/test_archive_read_support.c work/libarchive-3.5.0/libarchive-3.5.0/libarchive/test/test_archive_read_support.c
--- work/libarchive-3.5.0/libarchive-3.5.0/libarchive/test/test_archive_read_support.c	1970-01-01 01:00:00.000000000 +0100
+++ work/libarchive-3.5.0/libarchive-3.5.0/libarchive/test/test_archive_read_support.c	2017-03-23 20:35:44.381701657 +0100
@@ -126,7 +126,9 @@
 	    ARCHIVE_FORMAT_WARC,
 	    ARCHIVE_FORMAT_RAR_V5,
 	};
-	for (unsigned i = 0; i < sizeof(format_codes) / sizeof(int); i++) {
+	unsigned int i;
+
+	for (i = 0; i < sizeof(format_codes) / sizeof(int); i++) {
 		format_code = format_codes[i];
 		test_filter_or_format(format_code_enabler);
 		test_filter_or_format(format_code_setter);
