diff -urN work/curl-7.73.0/curl-7.73.0/Makefile.am work/curl-7.73.0/curl-7.73.0/Makefile.am
--- work/curl-7.73.0/curl-7.73.0/Makefile.am	2018-10-23 23:00:46.662292000 +0200
+++ work/curl-7.73.0/curl-7.73.0/Makefile.am	2018-10-23 23:01:08.566300143 +0200
@@ -195,7 +195,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS) packages include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -306,8 +306,8 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
+	#(cd docs && $(MAKE) install)
+	#(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
