diff -urN D7000_V1.0.1.70_WW_src/Source/apps/radvd-1.1/Makefile D7000_V1.0.1.70_WW_src/Source/apps/radvd-1.1/Makefile
--- D7000_V1.0.1.70_WW_src/Source/apps/radvd-1.1/Makefile	2016-10-11 17:54:35.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Source/apps/radvd-1.1/Makefile	2017-05-01 21:49:32.193798601 +0200
@@ -120,11 +120,9 @@
 AUTOHEADER = ${SHELL} /work/Platform/DG834GV_26/code/projects/DG834_V4/Neutral2/Source/apps/radvd-1.1/missing --run autoheader
 AUTOMAKE = ${SHELL} /work/Platform/DG834GV_26/code/projects/DG834_V4/Neutral2/Source/apps/radvd-1.1/missing --run automake-1.9
 AWK = gawk
-#CC = /work/Platform/DG834GV_26/code/projects/DG834_V4/Neutral2/Source/tools/mips-linux-uclibc/bin/mips-uclibc-gcc
 CC = $(CROSS)gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -g -O2 -Wall -Wpointer-arith  -Wcast-qual -Wcast-align -Wconversion \
-	-Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
+CFLAGS +=  -Os -Wall -Wpointer-arith  -Wcast-qual -Wcast-align -Wconversion -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
 ifeq ($(IP6_6RD_CE),1)
 CFLAGS+= -DIP6_6RD_CE
 CFLAGS += -I../nvram -L../nvram -lscnvram 
@@ -675,9 +673,8 @@
 	  test -z "$$dir" || $(mkdir_p) "$$dir"; \
 	done
 install: $(BUILT_SOURCES)
-	$(STRIP) radvd
+	$(SSTRIP) radvd
 	cp radvd ../../target/usr/sbin
-#	$(MAKE) $(AM_MAKEFLAGS) install-am
 install-exec: install-exec-am
 install-data: install-data-am
 uninstall: uninstall-am
