diff -urN D7000_V1.0.1.68_WW_src/Source/apps/totd-1.4/Makefile D7000_V1.0.1.68_WW_src/Source/apps/totd-1.4/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/totd-1.4/Makefile	2016-10-11 17:56:32.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/totd-1.4/Makefile	2017-05-01 22:08:23.442772852 +0200
@@ -12,7 +12,7 @@
 OBJS := $(patsubst %.c,%.o,$(SRC))
 
 
-CFLAGS := -Os -Wall  -DHAVE_CONFIG_H  -Wall -DTOTCONF=\"/etc/totd.conf\" -DSCOPED_REWRITE -DSTF -DUSE_INET4 -DUSE_INET6
+CFLAGS += -Os -Wall  -DHAVE_CONFIG_H  -Wall -DTOTCONF=\"/etc/totd.conf\" -DSCOPED_REWRITE -DSTF -DUSE_INET4 -DUSE_INET6
 LDFLAGS = -lnsl
 
 CFLAGS += -DIP6_ONLY
@@ -25,7 +25,7 @@
 
 totd: $(OBJS)
 	$(CC) $(LDFLAGS) -o $@ $^
-	$(STRIP) $@
+	$(SSTRIP) $@
 
 install: all
 	cp -a totd ../../target/usr/sbin/
