diff -urN work/libid3tag-0.15.1b/Makefile work/libid3tag-0.15.1b/Makefile
--- work/libid3tag-0.15.1b/Makefile	2018-10-23 23:00:46.662292000 +0200
+++ work/libid3tag-0.15.1b/Makefile	2018-10-23 23:01:08.566300143 +0200
@@ -0,0 +1,54 @@
+include ../Rules.mak
+
+LIBSNAME		:= libid3tag.so
+LIBSMAINVER		:= 0
+LIBSVER			:= $(LIBSMAINVER).3.0
+LIBSDIRNAME		:= $(ID3TAG)/.libs
+LIBS			:= $(LIBSNAME).$(LIBSVER)
+
+all: config
+	$(MAKE) -C $(ID3TAG)
+	$(STRIP) $(STRIPFLAGS) $(LIBSDIRNAME)/*.a
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(ID3TAG) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) CFLAGS="$(CFLAGS) -Os -s" CPPFLAGS="-I$(WORK_DIR)/$(ZLIB)/$(ZLIB)" LDFLAGS="-L$(WORK_DIR)/$(ZLIB)/$(ZLIB)" --enable-static --disable-shared --disable-debugging --disable-profiling
+
+install: control
+	#$(STRIP) $(LIBSDIRNAME)/$(LIBS)
+	#for D in $(LIBDIR); do install -m 0755 -d $$D; done
+	#install -m 0755 $(LIBSDIRNAME)/$(LIBS) $(LIBDIR)
+
+clean:
+	$(MAKE) -C $(ID3TAG) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(ID3TAG) distclean
+
+ifndef BUILD
+control:
+else
+POSTINST		:= $(CTRL_DIR)/postinst
+POSTRM			:= $(CTRL_DIR)/postrm
+
+control: postinst postrm
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "for L in $(LIBSNAME) $(LIBSNAME).$(LIBSMAINVER); do ln -sf $(LIBS) $(LDLIB)/\$$L; done" >> $(POSTINST)
+	@echo "sync" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+
+postrm:
+	@echo "#!/bin/sh" > $(POSTRM)
+	@echo "rm -f $(LDLIB)/$(LIBSNAME)*" >> $(POSTRM)
+	@echo "sync" >> $(POSTRM)
+	@echo "exit 0" >> $(POSTRM)
+endif
+
