diff -urN D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/chat/Makefile.linux D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/chat/Makefile.linux
--- D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/chat/Makefile.linux	2016-10-11 17:57:03.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/chat/Makefile.linux	2017-05-02 23:37:11.015568569 +0200
@@ -10,17 +10,15 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-CFLAGS=	-Os -pipe $(CDEFS)
-
 INSTALL= install
 
 all:	chat
 
 chat:	chat.o
-	$(CC) -o chat chat.o
+	$(CC) -o $(CFLAGS) $(MYFLAGS) $(CDEFS) chat chat.o
 
 chat.o:	chat.c
-	$(CC) -c $(CFLAGS) -o chat.o chat.c
+	$(CC) -c $(CFLAGS) $(MYFLAGS) $(CDEFS) -o chat.o chat.c
 
 install: chat
 	mkdir -p $(BINDIR)
@@ -29,3 +27,4 @@
 
 clean:
 	rm -f chat.o chat *~
+
diff -urN D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/linux/Makefile.top D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/linux/Makefile.top
--- D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/linux/Makefile.top	2016-10-11 17:57:03.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/linux/Makefile.top	2017-05-03 12:49:22.165672564 +0200
@@ -7,14 +7,14 @@
 STRIP=$(CROSS)strip
 
 all:
-	cd pppd; make -e -C plugins
-	make -e -C pppd
+	cd pppd; make -C plugins
+	make -C pppd
 ifeq ($(3G_FEATURE),1)
 	make -C chat
 endif
 
 install:# $(BINDIR) $(MANDIR)/man8 install-progs install-etcppp
-	$(STRIP) pppd/pppd		
+	$(SSTRIP) pppd/pppd		
 	$(STRIP) pppd/plugins/pppoe/libpppoe.so
 	cp -a pppd/pppd ../../target/usr/sbin/
 	cp -a pppd/plugins/pppoe/libpppoe.so ../../target/lib/
@@ -23,7 +23,7 @@
 	cp -a pppd/plugins/libpppoatm.so ../../target/lib/
 endif
 ifeq ($(3G_FEATURE),1)
-	$(STRIP) chat/chat
+	$(SSTRIP) chat/chat
 	cp -a chat/chat ../../target/usr/sbin
 endif
 
diff -urN D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/ipv6cp.c D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/ipv6cp.c
--- D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/ipv6cp.c	2016-10-11 17:57:04.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/ipv6cp.c	2017-05-03 18:10:31.985588949 +0200
@@ -106,6 +106,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 #include <netdb.h>
diff -urN D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux
--- D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux	2016-10-11 17:57:04.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux	2017-05-03 22:55:11.601874218 +0200
@@ -7,15 +7,10 @@
 BINDIR = /usr/sbin
 MANDIR = /usr/man
 
-PPPDSRCS = main.c magic.c fsm.c lcp.c ipcp.c upap.c chap.c md5.c ccp.c \
-	   ipxcp.c auth.c options.c sys-linux.c md4.c chap_ms.c cbcp.c \
-	   demand.c utils.c tty.c sha1.c
-HEADERS =  callout.h pathnames.h patchlevel.h chap.h md5.h chap_ms.h md4.h \
-	   ipxcp.h cbcp.h tdb.h sha1.h
+PPPDSRCS = main.c magic.c fsm.c lcp.c ipcp.c upap.c chap.c md5.c ccp.c ipxcp.c auth.c options.c sys-linux.c md4.c chap_ms.c cbcp.c demand.c utils.c tty.c sha1.c
+HEADERS =  callout.h pathnames.h patchlevel.h chap.h md5.h chap_ms.h md4.h ipxcp.h cbcp.h tdb.h sha1.h
 MANPAGES = pppd.8
-PPPDOBJS = main.o magic.o fsm.o lcp.o ipcp.o upap.o chap.o md5.o ccp.o \
-	   auth.o options.o demand.o utils.o sys-linux.o ipxcp.o \
-	   tty.o sha1.o
+PPPDOBJS = main.o magic.o fsm.o lcp.o ipcp.o upap.o chap.o md5.o ccp.o auth.o options.o demand.o utils.o sys-linux.o ipxcp.o tty.o sha1.o
 
 all: pppd
 
@@ -26,22 +21,16 @@
 endif
 
 include ../../../Rules.mak
-CC	= $(CROSS)gcc
+CC = $(CROSS)gcc
 #
-COPTS = -Os -pipe -Wall
-LIBS =
-
-ifneq ($(wildcard /usr/lib/libcrypt.*),)
-LIBS += -lcrypt
-endif
+COPTS = -Os -pipe -Wall -Wno-strict-aliasing
+LDFLAGS = -lcrypt
 
 # Uncomment the next 2 lines to include support for Microsoft's
 # MS-CHAP authentication protocol.
 CHAPMS=y
 USE_CRYPT=y
-ifneq ($(wildcard /usr/lib/libcrypt.*),)
 HAVE_CRYPT_H=y
-endif
 # Support MS_CHAPV2
 # porting from PPPD2.4.2
 MS_CHAPV2=y
@@ -60,86 +49,75 @@
 
 PLUGIN=y
 
-INCLUDE_DIRS= -I../include
-INCLUDE_DIRS+= -I$(ROOT)/shared
+INCLUDE_DIRS = -I. -I../include -I$(ROOT)/shared
 
-#COMPILE_FLAGS= -D_linux_=1 -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MULTILINK -DHAVE_MMAP
-#COMPILE_FLAGS= -D_linux_=1 -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP
-#COMPILE_FLAGS= -D_linux_=1 -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP #-D_DISABLE_SERIAL_
-COMPILE_FLAGS= -D_linux_=1 -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP
-
-CFLAGS= $(COPTS) $(COMPILE_FLAGS) $(INCLUDE_DIRS)
-#CFLAGS += -Werror -DMULTI_OFFER
-CFLAGS += -DMULTI_OFFER
+COMPILE_FLAGS = -D_linux_=1 -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP -DMULTI_OFFER
 
-CFLAGS += -Wno-strict-aliasing
 ifeq ($(TRAFFIC_METER),1)
-CFLAGS += -DTRAFFIC_METER
+COMPILE_FLAGS += -DTRAFFIC_METER
 endif
 
 ifdef CHAPMS
-CFLAGS   += -DCHAPMS=1
+COMPILE_FLAGS += -DCHAPMS=1
 ifndef USE_CRYPT
-LIBS     := -ldes $(LIBS)
+LDFLAGS     := -ldes $(LDFLAGS)
 else
-CFLAGS   += -DUSE_CRYPT=1
-ifneq ($(wildcard /usr/include/crypt.h),)
-CFLAGS   += -DHAVE_CRYPT_H=1
-endif
+COMPILE_FLAGS += -DUSE_CRYPT=1 -DHAVE_CRYPT_H=1
 endif
 PPPDOBJS += md4.o chap_ms.o
 ifdef MSLANMAN
-CFLAGS   += -DMSLANMAN=1
+COMPILE_FLAGS += -DMSLANMAN=1
 endif
 endif
 
 ifdef MS_CHAPV2
-CFLAGS += -DMS_CHAPV2=1
+COMPILE_FLAGS += -DMS_CHAPV2=1
 endif
 
 ifdef HAS_SHADOW
-CFLAGS   += -DHAS_SHADOW
-#LIBS     := -lshadow $(LIBS)
+COMPILE_FLAGS += -DHAS_SHADOW
 endif
 
 # For "Pluggable Authentication Modules", see ftp.redhat.com:/pub/pam/.
 ifdef USE_PAM
-CFLAGS   += -DUSE_PAM
-LIBS     := -lpam -ldl $(LIBS)
+COMPILE_FLAGS += -DUSE_PAM
+LDFLAGS     := -lpam -ldl $(LDFLAGS)
 endif
 
 # Lock library binary for Linux is included in 'linux' subdirectory.
 ifdef LOCKLIB
-LIBS     := -llock $(LIBS)
-CFLAGS   += -DLOCKLIB=1
+LDFLAGS     := -llock $(LDFLAGS)
+COMPILE_FLAGS += -DLOCKLIB=1
 endif
 
 # Removed the dependency on libdl.so
 ifdef PLUGIN
-CFLAGS	+= -DPLUGIN
-#LDFLAGS	+= -Wl,-E
-#LIBS	+= -ldl
+COMPILE_FLAGS += -DPLUGIN
 endif
 
 ifdef FILTER
-LIBS    += -lpcap
-CFLAGS  += -DPPP_FILTER -I/usr/include/pcap
+LDFLAGS    += -lpcap
+COMPILE_FLAGS += -DPPP_FILTER -I/usr/include/pcap
 endif
 
 ifdef HAVE_INET6
      PPPDSRCS += ipv6cp.c eui64.c
      HEADERS  += ipv6cp.h eui64.h
      PPPDOBJS += ipv6cp.o eui64.o
-     CFLAGS   += -DINET6=1
+     COMPILE_FLAGS += -DINET6=1
 endif
 
-LIBS +=  -L../../nvram -lscnvram
-LIBS +=  -L$(SC_LIBS_DIR)/liblink -llink
+LDFLAGS +=  -L../../nvram -lscnvram
+LDFLAGS +=  -L$(SC_LIBS_DIR)/liblink -llink
 
 ifeq ($(3G_FEATURE),1)
-LIBS += -L$(SC_LIBS_DIR)/libdongle_3g -ldongle_3g
+LDFLAGS += -L$(SC_LIBS_DIR)/libdongle_3g -ldongle_3g
 endif
 
+MYLDFLAGS += -Lplugins/pppoe -lpppoe -L../../$(ATMDIR)/src/lib/.libs -latm -Lplugins -lpppoatm
+
+MYFLAGS = $(COPTS) $(COMPILE_FLAGS) $(INCLUDE_DIRS)
+
 INSTALL= install -o beaulieu
 
 install: pppd
@@ -150,10 +128,14 @@
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
-	$(CC)  $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS) -Lplugins/pppoe -lpppoe -L../../linux-atm-2.4.0/src/lib/.libs -latm -Lplugins -lpppoatm
+	$(CC) $^ $(MYFLAGS) $(CFLAGS) $(LDFLAGS) $(MYLDFLAGS) -o $@
+
+%.o: %.c
+	$(CC) -c $< $(MYFLAGS) $(CFLAGS) $(LDFLAGS) $(MYLDFLAGS) -o $@
 
 clean:
-	rm -f $(PPPDOBJS) pppd *~ #* core
+	rm -f $(PPPDOBJS) pppd *~
 
 depend:
-	$(CPP) -M $(CFLAGS) $(PPPDSRCS) >.depend
+	$(CPP) -M $(CFLAGS) $(MYFLAGS) $(PPPDSRCS) >.depend
+
diff -urN D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/Makefile.linux D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/Makefile.linux
--- D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/Makefile.linux	2016-10-11 17:57:04.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/Makefile.linux	2017-05-03 23:05:17.123622840 +0200
@@ -1,35 +1,31 @@
 
 include ../../../../Rules.mak
-CC	= $(CROSS)gcc
-#CFLAGS	= -Os -I.. -I../../include -fPIC
-CFLAGS	+= -Os -I.. -I../../include -fPIC #-D_DISABLE_SERIAL_
-CFLAGS += -Wall -Werror
-CFLAGS += -DCONFIG_MIPS_BRCM
-LDFLAGS	= -shared
-LDFLAGS	+= -L../../../linux-atm-2.4.0/src/lib/.libs -latm
+CC = $(CROSS)gcc
+MYFLAGS = -Os -fPIC -Wall -Werror -DCONFIG_MIPS_BRCM
+CPPFLAGS = -I.. -I../../include
+LDFLAGS	= -shared -L../../../$(ATMDIR)/src/lib/.libs -latm
 INSTALL	= install
 
-all:	minconn.so passprompt.so pppoatm.so pppoe/pppoe.so pppoatm.d
-	$(MAKE) -C pppoe $(MFLAGS) all
+all:	minconn.so passprompt.so libpppoatm.so libpppoe.so pppoatm.d
 
-pppoe/pppoe.so:
-	$(MAKE) -C pppoe $(MFLAGS) pppoe.so
+libpppoe.so:
+	$(MAKE) -C pppoe
 
 minconn.so: minconn.c
-	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) minconn.c
-pppoatm.d:pppoatm.c
-	$(CC) -D_linux_=1 -o $@ $(LDFLAGS) $(CFLAGS) -M pppoatm.c
-
-pppoatm.so: pppoatm.c
-	$(CC) -D_linux_=1 -o $@ $(LDFLAGS) $(CFLAGS) pppoatm.c
-	cp pppoatm.so libpppoatm.so
+	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $(MYFLAGS) $(CPPFLAGS) minconn.c
+
+pppoatm.d: pppoatm.c
+	$(CC) -D_linux_=1 -o $@ $(LDFLAGS) $(CFLAGS) $(MYFLAGS) $(CPPFLAGS) -M pppoatm.c
+
+libpppoatm.so: pppoatm.c
+	$(CC) -D_linux_=1 -o $@ $(LDFLAGS) $(CFLAGS) $(MYFLAGS) $(CPPFLAGS) pppoatm.c
 
 passprompt.so: passprompt.c
-	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) passprompt.c
+	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $(MYFLAGS) $(CPPFLAGS) passprompt.c
 
 LIBDIR	= /home/beaulieu/work/mips/userland/filesystem/tiny/usr/lib/pppd
 
-install: minconn.so passprompt.so pppoe/pppoe.so
+install: minconn.so passprompt.so pppoe.so
 	version=`awk -F '"' '/VERSION/ { print $$2; }' ../patchlevel.h`; \
 	$(INSTALL) -d $(LIBDIR)/$$version; \
 	$(INSTALL) $? $(LIBDIR)/$$version
diff -urN D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/pppoe/Makefile.linux D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/pppoe/Makefile.linux
--- D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/pppoe/Makefile.linux	2016-10-11 17:57:04.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/apps/ppp-2.4.1.pppoe4.orig/pppd/plugins/pppoe/Makefile.linux	2017-05-03 23:48:12.791492244 +0200
@@ -1,9 +1,8 @@
 include ../../../../../Rules.mak
-CFLAGS	= -Os -Wall -I.. -I../.. -I../../../include -D_linux_=1 -fPIC -DMULTI_OFFER
-CFLAGS += -Werror -Wno-unused-but-set-variable
-
-CFLAGS += -Wno-strict-aliasing
-LDFLAGS	= -shared
+COPTS = -Os -fPIC -Wall -Werror -Wno-unused-but-set-variable -Wno-strict-aliasing
+DEFS = -D_linux_=1 -DMULTI_OFFER
+CPPFLAGS = -I.. -I../.. -I../../../include
+#LDFLAGS	= -shared
 
 #
 # Enable RECORD_SID to record the session id into flash, 
@@ -13,20 +12,16 @@
 #
 RECORD_SID=1
 ifeq ($(RECORD_SID),1)
-#CFLAGS  += -D_RECORD_SID_
-CFLAGS  += -I$(APPS_DIR)/nvram -I$(SHARED_DIR)
-LDFLAGS += -L$(APPS_DIR)/nvram -lscnvram
-CFLAGS += -I$(SC_LIBS_DIR)/include
-LDFLAGS += -L$(ROOT)/apps/sc_libs/liblink -llink
+CPPFLAGS += -I$(APPS_DIR)/nvram -I$(SHARED_DIR) -I$(SC_LIBS_DIR)/include
+LDFLAGS += -L$(APPS_DIR)/nvram -lscnvram -L$(ROOT)/apps/sc_libs/liblink -llink
 endif
 
 
-all: pppoe.so #pppoed
+all: libpppoe.so
 
 
 PLUGINDIR = $(LIBDIR)/plugins
-PLUGINSRCS= pppoe.c libpppoe.c utils.c pppoehash.c pppoe_client.c \
-		pppoe_relay.c pppoe_server.c pppd_utils.c
+PLUGINSRCS= pppoe.c libpppoe.c utils.c pppoehash.c pppoe_client.c pppoe_relay.c pppoe_server.c pppd_utils.c
 #
 # include dependancies if present and backup if as a header file
 ifeq (.depend,$(wildcard .depend))
@@ -36,28 +31,26 @@
 
 
 pppoefwd: pppoefwd.o libpppoe.a 
-	$(CC) -o $@ $^
+	$(CC) $^ $(DEFS) $(COPTS) $(CFLAGS) -fcommon $(CPPFLAGS) -o $@
 
 pppoed: pppoed.o pppd_utils.o libpppoe.a 
-	$(CC) -o $@ $^
+	$(CC) $^ $(DEFS) $(COPTS) $(CFLAGS) -fcommon $(CPPFLAGS) -o $@
+
+libpppoe.a: pppoehash.o pppoe_client.o pppoe_relay.o pppoe_server.o utils.o libpppoe.o 
+	$(AR) -rc $@ $^
+	$(STRIP) $(STRIPFLAGS) $@
 
-libpppoe.a: pppoehash.o pppoe_client.o pppoe_relay.o pppoe_server.o \
-	utils.o libpppoe.o 
-	ar -rc $@ $^
-
-pppoe.so: pppoe.o libpppoe.a
-	$(CC) -o $@ $(LDFLAGS) $^
-	cp pppoe.so libpppoe.so
+libpppoe.so: pppoe.o libpppoe.a
+	$(CC) $^ $(DEFS) $(CFLAGS) -fcommon $(CPPFLAGS) $(LDFLAGS) -shared -Wl,-soname,$@ -o $@
 
-%.so: %.c
-	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $^
+%.o: %.c
+	$(CC) -c $< $(DEFS) $(COPTS) $(CFLAGS) -fcommon $(CPPFLAGS) $(LDFLAGS) -o $@
 
 clean: 
 	rm -f *.o *.so *.a pppoefwd pppoed
 
-
 $(PLUGINDIR):
 	$(INSTALL) -d -m 755 $@
 
 depend:
-	$(CPP) -M $(CFLAGS) $(PLUGINSRCS) >.depend
+	$(CPP) $(DEFS) -M $(MYFLAGS) $(CFLAGS) -fcommon $(CPPFLAGS) $(PLUGINSRCS) >.depend
