diff -urN work/iproute2-4.19.0/tc/Makefile work/iproute2-4.19.0/tc/Makefile
--- work/iproute2-4.19.0/tc/Makefile	2016-12-13 00:07:42.000000000 +0100
+++ work/iproute2-4.19.0/tc/Makefile	2017-02-15 14:03:20.593464080 +0100
@@ -12,7 +12,6 @@
 TCMODULES += q_sfq.o
 TCMODULES += q_red.o
 TCMODULES += q_prio.o
-TCMODULES += q_skbprio.o
 TCMODULES += q_tbf.o
 TCMODULES += q_cbq.o
 TCMODULES += q_rr.o
@@ -25,10 +24,8 @@
 TCMODULES += f_route.o
 TCMODULES += f_fw.o
 TCMODULES += f_basic.o
-TCMODULES += f_bpf.o
 TCMODULES += f_flow.o
 TCMODULES += f_cgroup.o
-TCMODULES += f_flower.o
 TCMODULES += q_dsmark.o
 TCMODULES += q_gred.o
 TCMODULES += f_tcindex.o
@@ -41,16 +38,8 @@
 TCMODULES += m_mirred.o
 TCMODULES += m_nat.o
 TCMODULES += m_pedit.o
-TCMODULES += m_ife.o
 TCMODULES += m_skbedit.o
-TCMODULES += m_skbmod.o
 TCMODULES += m_csum.o
-TCMODULES += m_simple.o
-TCMODULES += m_vlan.o
-TCMODULES += m_connmark.o
-TCMODULES += m_bpf.o
-TCMODULES += m_tunnel_key.o
-TCMODULES += m_sample.o
 TCMODULES += p_ip.o
 TCMODULES += p_ip6.o
 TCMODULES += p_icmp.o
@@ -60,20 +49,10 @@
 TCMODULES += em_nbyte.o
 TCMODULES += em_cmp.o
 TCMODULES += em_u32.o
-TCMODULES += em_canid.o
 TCMODULES += em_meta.o
 TCMODULES += q_mqprio.o
 TCMODULES += q_codel.o
 TCMODULES += q_fq_codel.o
-TCMODULES += q_fq.o
-TCMODULES += q_pie.o
-TCMODULES += q_cake.o
-TCMODULES += q_hhf.o
-TCMODULES += q_clsact.o
-TCMODULES += e_bpf.o
-TCMODULES += f_matchall.o
-TCMODULES += q_cbs.o
-TCMODULES += q_etf.o
 
 TCSO :=
 ifeq ($(TC_CONFIG_ATM),y)
diff -urN work/iproute2-4.19.0/include/uapi/linux/pkt_sched.h work/iproute2-4.19.0/include/uapi/linux/pkt_sched.h
--- work/iproute2-4.19.0/include/uapi/linux/pkt_sched.h	2016-12-13 00:07:42.000000000 +0100
+++ work/iproute2-4.19.0/include/uapi/linux/pkt_sched.h	2017-02-15 14:04:20.324678961 +0100
@@ -124,21 +124,6 @@
 	__u32	limit;	/* Queue length: bytes for bfifo, packets for pfifo */
 };
 
-/* SKBPRIO section */
-
-/*
- * Priorities go from zero to (SKBPRIO_MAX_PRIORITY - 1).
- * SKBPRIO_MAX_PRIORITY should be at least 64 in order for skbprio to be able
- * to map one to one the DS field of IPV4 and IPV6 headers.
- * Memory allocation grows linearly with SKBPRIO_MAX_PRIORITY.
- */
-
-#define SKBPRIO_MAX_PRIORITY 64
-
-struct tc_skbprio_qopt {
-	__u32	limit;		/* Queue length in packets. */
-};
-
 /* PRIO section */
 
 #define TCQ_PRIO_BANDS	16
@@ -837,251 +822,4 @@
 	};
 };
 
-/* FQ */
-
-enum {
-	TCA_FQ_UNSPEC,
-
-	TCA_FQ_PLIMIT,		/* limit of total number of packets in queue */
-
-	TCA_FQ_FLOW_PLIMIT,	/* limit of packets per flow */
-
-	TCA_FQ_QUANTUM,		/* RR quantum */
-
-	TCA_FQ_INITIAL_QUANTUM,		/* RR quantum for new flow */
-
-	TCA_FQ_RATE_ENABLE,	/* enable/disable rate limiting */
-
-	TCA_FQ_FLOW_DEFAULT_RATE,/* obsolete, do not use */
-
-	TCA_FQ_FLOW_MAX_RATE,	/* per flow max rate */
-
-	TCA_FQ_BUCKETS_LOG,	/* log2(number of buckets) */
-
-	TCA_FQ_FLOW_REFILL_DELAY,	/* flow credit refill delay in usec */
-
-	TCA_FQ_ORPHAN_MASK,	/* mask applied to orphaned skb hashes */
-
-	TCA_FQ_LOW_RATE_THRESHOLD, /* per packet delay under this rate */
-
-	__TCA_FQ_MAX
-};
-
-#define TCA_FQ_MAX	(__TCA_FQ_MAX - 1)
-
-struct tc_fq_qd_stats {
-	__u64	gc_flows;
-	__u64	highprio_packets;
-	__u64	tcp_retrans;
-	__u64	throttled;
-	__u64	flows_plimit;
-	__u64	pkts_too_long;
-	__u64	allocation_errors;
-	__s64	time_next_delayed_flow;
-	__u32	flows;
-	__u32	inactive_flows;
-	__u32	throttled_flows;
-	__u32	unthrottle_latency_ns;
-};
-
-/* Heavy-Hitter Filter */
-
-enum {
-	TCA_HHF_UNSPEC,
-	TCA_HHF_BACKLOG_LIMIT,
-	TCA_HHF_QUANTUM,
-	TCA_HHF_HH_FLOWS_LIMIT,
-	TCA_HHF_RESET_TIMEOUT,
-	TCA_HHF_ADMIT_BYTES,
-	TCA_HHF_EVICT_TIMEOUT,
-	TCA_HHF_NON_HH_WEIGHT,
-	__TCA_HHF_MAX
-};
-
-#define TCA_HHF_MAX	(__TCA_HHF_MAX - 1)
-
-struct tc_hhf_xstats {
-	__u32	drop_overlimit; /* number of times max qdisc packet limit
-				 * was hit
-				 */
-	__u32	hh_overlimit;   /* number of times max heavy-hitters was hit */
-	__u32	hh_tot_count;   /* number of captured heavy-hitters so far */
-	__u32	hh_cur_count;   /* number of current heavy-hitters */
-};
-
-/* PIE */
-enum {
-	TCA_PIE_UNSPEC,
-	TCA_PIE_TARGET,
-	TCA_PIE_LIMIT,
-	TCA_PIE_TUPDATE,
-	TCA_PIE_ALPHA,
-	TCA_PIE_BETA,
-	TCA_PIE_ECN,
-	TCA_PIE_BYTEMODE,
-	__TCA_PIE_MAX
-};
-#define TCA_PIE_MAX   (__TCA_PIE_MAX - 1)
-
-struct tc_pie_xstats {
-	__u32 prob;             /* current probability */
-	__u32 delay;            /* current delay in ms */
-	__u32 avg_dq_rate;      /* current average dq_rate in bits/pie_time */
-	__u32 packets_in;       /* total number of packets enqueued */
-	__u32 dropped;          /* packets dropped due to pie_action */
-	__u32 overlimit;        /* dropped due to lack of space in queue */
-	__u32 maxq;             /* maximum queue size */
-	__u32 ecn_mark;         /* packets marked with ecn*/
-};
-
-/* CBS */
-struct tc_cbs_qopt {
-	__u8 offload;
-	__u8 _pad[3];
-	__s32 hicredit;
-	__s32 locredit;
-	__s32 idleslope;
-	__s32 sendslope;
-};
-
-enum {
-	TCA_CBS_UNSPEC,
-	TCA_CBS_PARMS,
-	__TCA_CBS_MAX,
-};
-
-#define TCA_CBS_MAX (__TCA_CBS_MAX - 1)
-
-
-/* ETF */
-struct tc_etf_qopt {
-	__s32 delta;
-	__s32 clockid;
-	__u32 flags;
-#define TC_ETF_DEADLINE_MODE_ON	BIT(0)
-#define TC_ETF_OFFLOAD_ON	BIT(1)
-};
-
-enum {
-	TCA_ETF_UNSPEC,
-	TCA_ETF_PARMS,
-	__TCA_ETF_MAX,
-};
-
-#define TCA_ETF_MAX (__TCA_ETF_MAX - 1)
-
-
-/* CAKE */
-enum {
-	TCA_CAKE_UNSPEC,
-	TCA_CAKE_PAD,
-	TCA_CAKE_BASE_RATE64,
-	TCA_CAKE_DIFFSERV_MODE,
-	TCA_CAKE_ATM,
-	TCA_CAKE_FLOW_MODE,
-	TCA_CAKE_OVERHEAD,
-	TCA_CAKE_RTT,
-	TCA_CAKE_TARGET,
-	TCA_CAKE_AUTORATE,
-	TCA_CAKE_MEMORY,
-	TCA_CAKE_NAT,
-	TCA_CAKE_RAW,
-	TCA_CAKE_WASH,
-	TCA_CAKE_MPU,
-	TCA_CAKE_INGRESS,
-	TCA_CAKE_ACK_FILTER,
-	TCA_CAKE_SPLIT_GSO,
-	__TCA_CAKE_MAX
-};
-#define TCA_CAKE_MAX	(__TCA_CAKE_MAX - 1)
-
-enum {
-	__TCA_CAKE_STATS_INVALID,
-	TCA_CAKE_STATS_PAD,
-	TCA_CAKE_STATS_CAPACITY_ESTIMATE64,
-	TCA_CAKE_STATS_MEMORY_LIMIT,
-	TCA_CAKE_STATS_MEMORY_USED,
-	TCA_CAKE_STATS_AVG_NETOFF,
-	TCA_CAKE_STATS_MIN_NETLEN,
-	TCA_CAKE_STATS_MAX_NETLEN,
-	TCA_CAKE_STATS_MIN_ADJLEN,
-	TCA_CAKE_STATS_MAX_ADJLEN,
-	TCA_CAKE_STATS_TIN_STATS,
-	TCA_CAKE_STATS_DEFICIT,
-	TCA_CAKE_STATS_COBALT_COUNT,
-	TCA_CAKE_STATS_DROPPING,
-	TCA_CAKE_STATS_DROP_NEXT_US,
-	TCA_CAKE_STATS_P_DROP,
-	TCA_CAKE_STATS_BLUE_TIMER_US,
-	__TCA_CAKE_STATS_MAX
-};
-#define TCA_CAKE_STATS_MAX (__TCA_CAKE_STATS_MAX - 1)
-
-enum {
-	__TCA_CAKE_TIN_STATS_INVALID,
-	TCA_CAKE_TIN_STATS_PAD,
-	TCA_CAKE_TIN_STATS_SENT_PACKETS,
-	TCA_CAKE_TIN_STATS_SENT_BYTES64,
-	TCA_CAKE_TIN_STATS_DROPPED_PACKETS,
-	TCA_CAKE_TIN_STATS_DROPPED_BYTES64,
-	TCA_CAKE_TIN_STATS_ACKS_DROPPED_PACKETS,
-	TCA_CAKE_TIN_STATS_ACKS_DROPPED_BYTES64,
-	TCA_CAKE_TIN_STATS_ECN_MARKED_PACKETS,
-	TCA_CAKE_TIN_STATS_ECN_MARKED_BYTES64,
-	TCA_CAKE_TIN_STATS_BACKLOG_PACKETS,
-	TCA_CAKE_TIN_STATS_BACKLOG_BYTES,
-	TCA_CAKE_TIN_STATS_THRESHOLD_RATE64,
-	TCA_CAKE_TIN_STATS_TARGET_US,
-	TCA_CAKE_TIN_STATS_INTERVAL_US,
-	TCA_CAKE_TIN_STATS_WAY_INDIRECT_HITS,
-	TCA_CAKE_TIN_STATS_WAY_MISSES,
-	TCA_CAKE_TIN_STATS_WAY_COLLISIONS,
-	TCA_CAKE_TIN_STATS_PEAK_DELAY_US,
-	TCA_CAKE_TIN_STATS_AVG_DELAY_US,
-	TCA_CAKE_TIN_STATS_BASE_DELAY_US,
-	TCA_CAKE_TIN_STATS_SPARSE_FLOWS,
-	TCA_CAKE_TIN_STATS_BULK_FLOWS,
-	TCA_CAKE_TIN_STATS_UNRESPONSIVE_FLOWS,
-	TCA_CAKE_TIN_STATS_MAX_SKBLEN,
-	TCA_CAKE_TIN_STATS_FLOW_QUANTUM,
-	__TCA_CAKE_TIN_STATS_MAX
-};
-#define TCA_CAKE_TIN_STATS_MAX (__TCA_CAKE_TIN_STATS_MAX - 1)
-#define TC_CAKE_MAX_TINS (8)
-
-enum {
-	CAKE_FLOW_NONE = 0,
-	CAKE_FLOW_SRC_IP,
-	CAKE_FLOW_DST_IP,
-	CAKE_FLOW_HOSTS,    /* = CAKE_FLOW_SRC_IP | CAKE_FLOW_DST_IP */
-	CAKE_FLOW_FLOWS,
-	CAKE_FLOW_DUAL_SRC, /* = CAKE_FLOW_SRC_IP | CAKE_FLOW_FLOWS */
-	CAKE_FLOW_DUAL_DST, /* = CAKE_FLOW_DST_IP | CAKE_FLOW_FLOWS */
-	CAKE_FLOW_TRIPLE,   /* = CAKE_FLOW_HOSTS  | CAKE_FLOW_FLOWS */
-	CAKE_FLOW_MAX,
-};
-
-enum {
-	CAKE_DIFFSERV_DIFFSERV3 = 0,
-	CAKE_DIFFSERV_DIFFSERV4,
-	CAKE_DIFFSERV_DIFFSERV8,
-	CAKE_DIFFSERV_BESTEFFORT,
-	CAKE_DIFFSERV_PRECEDENCE,
-	CAKE_DIFFSERV_MAX
-};
-
-enum {
-	CAKE_ACK_NONE = 0,
-	CAKE_ACK_FILTER,
-	CAKE_ACK_AGGRESSIVE,
-	CAKE_ACK_MAX
-};
-
-enum {
-	CAKE_ATM_NONE = 0,
-	CAKE_ATM_ATM,
-	CAKE_ATM_PTM,
-	CAKE_ATM_MAX
-};
-
 #endif
