diff -urN D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/bcmdrivers/Makefile D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/bcmdrivers/Makefile
--- D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/bcmdrivers/Makefile	2016-10-11 17:52:38.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/bcmdrivers/Makefile	2017-01-03 14:19:41.660874198 +0100
@@ -186,26 +186,26 @@
 		rm -f $$dir; \
 	done
 ifneq ($(CONFIG_BCM_RDPA),)
-	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/rdp_cpu_ring.c
+#	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/rdp_cpu_ring.c
 	@rm -rf $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/rdd
 ifeq ($(CONFIG_BCM968500),y)
-	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_$(RDPA_PLATFORM)/*
-	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_$(RDPA_PLATFORM)
+	@rm -f $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_$(RDPA_PLATFORM)/*
+#	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_$(RDPA_PLATFORM)
 endif
 ifeq ($(CONFIG_BCM96838),y)
-	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/platform/$(RDPA_PLATFORM)/oren_data_path_init.c
-	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_oren/*
-	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_oren
+#	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/platform/$(RDPA_PLATFORM)/oren_data_path_init.c
+	@rm -f $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_oren/*
+#	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_oren
 endif
 ifeq ($(CONFIG_BCM963138),y)
-	-@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/platform/$(RDPA_PLATFORM)/bcm63138_data_path_init.c
-	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63138$(IS_REV_A)/*
-	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63138$(IS_REV_A)
+#	-@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/platform/$(RDPA_PLATFORM)/bcm63138_data_path_init.c
+	@rm -f $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63138$(IS_REV_A)/*
+#	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63138$(IS_REV_A)
 endif
 ifeq ($(CONFIG_BCM963148),y)
-	-@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/platform/$(RDPA_PLATFORM)/bcm63138_data_path_init.c
-	@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63148/*
-	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63148
+#	-@rm $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/platform/$(RDPA_PLATFORM)/bcm63138_data_path_init.c
+	@rm -f $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63148/*
+#	@rmdir $(BRCMDRIVERS_DIR)/broadcom/char/rdpa/impl$(CONFIG_BCM_RDPA_IMPL)/rdp/firmware_dsl_63148
 endif
 endif

diff -urN D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/hostTools/scripts/defconfig-bcm.template D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/hostTools/scripts/defconfig-bcm.template
--- D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/hostTools/scripts/defconfig-bcm.template	2016-10-11 17:47:55.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/hostTools/scripts/defconfig-bcm.template	2017-01-03 19:59:56.621028892 +0100
@@ -979,7 +979,7 @@
 #
 CONFIG_PCIEPORTBUS=y
 CONFIG_PCIEASPM=y
-CONFIG_PCIEASPM_POWERSAVE=y
+# CONFIG_PCIEASPM_POWERSAVE is not set
 # CONFIG_PCIEASPM_DEBUG is not set
 # CONFIG_PCIEAER is not set
 
@@ -2688,6 +2688,7 @@
 # CONFIG_SECCOMP is not set
 # CONFIG_CLASSIC_RCU is not set
 # CONFIG_TREE_RCU is not set
+# CONFIG_TREE_PREEMPT_RCU is not set
 # CONFIG_PREEMPT_RCU is not set
 # CONFIG_TREE_RCU_TRACE is not set
 # CONFIG_PREEMPT_RCU_TRACE is not set
@@ -3247,7 +3248,7 @@
 # CONFIG_NF_CONNTRACK_TIMEOUT is not set
 # CONFIG_NF_CT_NETLINK_TIMEOUT is not set
 # CONFIG_NETFILTER_XT_TARGET_CT is not set
-# CONFIG_NETFILTER_XT_TARGET_LOG is not set
+CONFIG_NETFILTER_XT_TARGET_LOG=m
 # CONFIG_NETFILTER_XT_MATCH_ECN is not set
 # CONFIG_NETFILTER_XT_MATCH_NFACCT is not set
 # CONFIG_IP_NF_MATCH_RPFILTER is not set
diff -urN D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/arch/arm/mm/Kconfig D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/arch/arm/mm/Kconfig
--- D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/arch/arm/mm/Kconfig	2016-10-11 17:47:58.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/arch/arm/mm/Kconfig	2017-01-07 21:17:07.283396000 +0100
@@ -696,7 +696,7 @@
 	  If unsure, say Y.
 
 config CPU_BIG_ENDIAN
-	bool "Build big-endian kernel"
+	bool
 	depends on ARCH_SUPPORTS_BIG_ENDIAN
 	help
 	  Say Y if you plan on running a kernel in big-endian mode.
diff -urN D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/kernel/timeconst.pl D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/kernel/timeconst.pl
--- D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/kernel/timeconst.pl	2016-10-11 17:47:37.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/kernel/timeconst.pl	2017-01-02 23:47:03.578053521 +0100
@@ -369,10 +369,8 @@
 		die "Usage: $0 HZ\n";
 	}
 
-	@val = @{$canned_values{$hz}};
-	if (!defined(@val)) {
-		@val = compute_values($hz);
-	}
+	$cv = $canned_values{$hz};
+	@val = defined($cv) ? @$cv : compute_values($hz);
 	output($hz, @val);
 }
 exit 0;
diff -urN D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/Makefile D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/Makefile
--- D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/Makefile	2016-10-11 17:52:44.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Kernel/bcm963xx/Makefile	2017-01-07 18:14:59.265907320 +0100
@@ -244,7 +244,7 @@
 	fi
 	$(GENDEFCONFIG_CMD) $(PROFILE_PATH) ${MAKEFLAGS}
 	cd $(KERNEL_DIR); \
-	cp -f $(KERNEL_DIR)/arch/$(ARCH)/defconfig $(KERNEL_DIR)/.config; \
+#	cp -f $(KERNEL_DIR)/arch/$(ARCH)/defconfig $(KERNEL_DIR)/.config; \
 	$(MAKE) oldconfig;
 endef
 
@@ -959,7 +959,8 @@
 	rm -fr $(FSSRC_DIR)/etc/certs
 
 kernel_clean: sanity_check
-	-$(MAKE) -C $(KERNEL_DIR) mrproper
+#	-$(MAKE) -C $(KERNEL_DIR) mrproper
+	-$(MAKE) -C $(KERNEL_DIR) clean
 	rm -f $(KERNEL_DIR)/arch/mips/defconfig
 	rm -f $(KERNEL_DIR)/arch/arm/defconfig
 	rm -f $(HOSTTOOLS_DIR)/lzma/decompress/*.o
