diff -urN D7000_V1.0.1.74_WW_src/Kernel/bcm963xx/sc.mak D7000_V1.0.1.74_WW_src/Kernel/bcm963xx/sc.mak
--- D7000_V1.0.1.74_WW_src/Kernel/bcm963xx/sc.mak	2016-10-11 17:52:43.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Kernel/bcm963xx/sc.mak	2017-01-03 18:14:59.265907320 +0100
@@ -190,6 +190,7 @@
       create_install kernelbuild modbuild kernelbuildlite hosttools vmlinux.lz_gen
 
 vmlinux.lz_gen:
+	@echo "Removing old vmlinux..."
 	@rm -f ./vmlinux
 	@rm -f ./vmlinux.bin
 	@rm -f ./vmlinux.lz
@@ -197,6 +198,8 @@
 	$(STRIP) --remove-section=.note --remove-section=.comment vmlinux; \
 	$(OBJCOPY) -O binary vmlinux vmlinux.bin; \
 	$(HOSTTOOLS_DIR)/cmplzma -k -1 vmlinux vmlinux.bin vmlinux.lz
+	@rm -f ./vmlinux
+	@rm -f ./vmlinux.bin
 
 sc_prepare:
 	if [ -d $(BCM_DIR) ]; then\
