diff -urN D7000_V1.0.1.48_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/net/sched/sch_htb.c D7000_V1.0.1.48_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/net/sched/sch_htb.c
--- D7000_V1.0.1.48_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/net/sched/sch_htb.c	2016-10-11 17:48:33.000000000 +0200
+++ D7000_V1.0.1.48_WW_src/Kernel/bcm963xx/kernel/linux-3.4rt/net/sched/sch_htb.c	2017-02-20 20:38:27.110646644 +0100
@@ -676,7 +676,7 @@
 	 * 1 to simplify things when jiffy is going to be incremented
 	 * too soon
 	 */
-	unsigned long stop_at = start + 2;
+	unsigned long stop_at = start + 4;
 	while (time_before(jiffies, stop_at)) {
 		struct htb_class *cl;
 		long diff;
