diff -urN D7000_V1.0.1.44_WW_src/Kernel/bcm963xx/sc.mak D7000_V1.0.1.44_WW_src/Kernel/bcm963xx/sc.mak
--- D7000_V1.0.1.44_WW_src/Kernel/bcm963xx/sc.mak	2017-02-27 20:33:25.998891029 +0100
+++ D7000_V1.0.1.44_WW_src/Kernel/bcm963xx/sc.mak	2017-02-27 20:43:32.021156543 +0100
@@ -184,6 +184,9 @@
 
 	cp -af $(BCM_DIR)/netfilter/*.ko $(FS_DIR)/lib/modules/
 	cp -af $(BCM_DIR)/bcmvlan.ko $(FS_DIR)/lib/modules/
+	cp -af $(BCM_DIR)/act_ipt.ko $(FS_DIR)/lib/modules/
+	cp -af $(BCM_DIR)/cls_flow.ko $(FS_DIR)/lib/modules/
+	cp -af $(BCM_DIR)/*_codel.ko $(FS_DIR)/lib/modules/
 	@echo -e "\\e[36m -- sc_driver --\e[0m"	
 
 vmlinux.lz_only : prebuild_checks profile_saved_check sanity_check \
@@ -270,6 +273,7 @@
 	cp -af $(KERNEL_DIR)/net/ipv4/netfilter/*.ko $(BCM_DIR)/netfilter;\
 	cp -af $(KERNEL_DIR)/net/ipv6/netfilter/*.ko $(BCM_DIR)/netfilter;\
 # qos
+	cp -af $(KERNEL_DIR)/net/sched/*.ko $(BCM_DIR)
 # voice
 ifneq ($(strip $(BUILD_VODSL)),)
 	cp -af $(BRCMDRIVERS_DIR)/broadcom/char/endpoint/bcm9$(BRCM_CHIP)/endpointdd.ko $(BCM_DIR)
