diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/ca.crt work/easy-rsa-3.0.5/easyrsa3/keys/ca.crt
--- work/easy-rsa-3.0.5/easyrsa3/keys/ca.crt	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/ca.crt	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,37 @@
+-----BEGIN CERTIFICATE-----
+MIIGfjCCBGagAwIBAgIJAPlFg9VlfE9BMA0GCSqGSIb3DQEBCwUAMIGBMQswCQYD
+VQQGEwJDTjELMAkGA1UECAwCSlMxCzAJBgNVBAcMAlNaMRAwDgYDVQQKDAduZXRn
+ZWFyMRAwDgYDVQQLDAduZXRnZWFyMRMwEQYDVQQDDApuZXRnZWFyIENBMR8wHQYJ
+KoZIhvcNAQkBFhBtYWlsQG5ldGdlYXIuY29tMB4XDTE5MDEyMDEzMjIzN1oXDTI5
+MDExNzEzMjIzN1owgYExCzAJBgNVBAYTAkNOMQswCQYDVQQIDAJKUzELMAkGA1UE
+BwwCU1oxEDAOBgNVBAoMB25ldGdlYXIxEDAOBgNVBAsMB25ldGdlYXIxEzARBgNV
+BAMMCm5ldGdlYXIgQ0ExHzAdBgkqhkiG9w0BCQEWEG1haWxAbmV0Z2Vhci5jb20w
+ggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDYYzAGklTL429wwdGSYSS7
+Wkn88yml102mjliaeEaf9+ctM7NQ688NMK2olB4Lg9jvxOLXf1D+whUOLSJc37uZ
+zksUzon89UkgwwooSSvtJJPQfSp1JDsykSdJiGkjwuW7rbZoFv0Owe2W8tyQVJ6o
+13vMJugt+9jUHSwtgPkzmPJ3IJWlTbEhfEE5teKdY5SBTCjpTg0pyEqGsFnULH1A
+79qrvcIt6M7lAUpT7i+w67vUfu2p5FewwXUR+oSfFu1MzOtZ1H0Cz44uiaDwL+az
+vIKagNCheYmInQ4hZ3IPiaZli0Pr5FJIyQq0fXa2JkRHo2Y+zNmJbEwitXxdRtA8
+RbNGkr0wnJHCcICGn3n3bo1gxzlFg19uKcBMQoer8c6oK8UyqjfiDpFJeh6G0DAZ
+dbnlvjUf7es5b1yMN6NFUBDl3Ni27FgeMq+HEoI1m1Zjl8CVLH02bnfBLHLe8RxQ
+Ux7gr3fjrCGmpf2R49hHMrH/XiumMhw/kLWIFjLHtkZsMCwsJMCn++DGsMLVSK5r
+/l0xEh/kvkZDCLVlRfeWG7F1ZdqQG4xebk2cvOYbXgO/w7568O4YYstpWls3nmln
+9DSB2HIk7bcxQcZvFV+UePS/62gHEw9mS0Djy2+SOrbVxjLHL6VpAEHcVr3sGwXG
+L7R0/NrC50GoG1oVU11EhQIDAQABo4H2MIHzMB0GA1UdDgQWBBSmuOQS2bzygPb7
+GUsC9UQzB3kg6DCBtgYDVR0jBIGuMIGrgBSmuOQS2bzygPb7GUsC9UQzB3kg6KGB
+h6SBhDCBgTELMAkGA1UEBhMCQ04xCzAJBgNVBAgMAkpTMQswCQYDVQQHDAJTWjEQ
+MA4GA1UECgwHbmV0Z2VhcjEQMA4GA1UECwwHbmV0Z2VhcjETMBEGA1UEAwwKbmV0
+Z2VhciBDQTEfMB0GCSqGSIb3DQEJARYQbWFpbEBuZXRnZWFyLmNvbYIJAPlFg9Vl
+fE9BMAwGA1UdEwQFMAMBAf8wCwYDVR0PBAQDAgEGMA0GCSqGSIb3DQEBCwUAA4IC
+AQAFiMruoLfp0IBmzCkaWMSknvzecg1lS69o2apXuUiiMQq7dnskzzDsmLHcaiAm
+TS0OPgXCobpixvpCl09v+K4CHj2tKnH6+FNh0uuCas6KMtqy3OgXYMO9LlvhDSjD
+22Ijgp0R1+6nDOwFaQttn/PjXAdrkFMNjgDV0ADaUvzT0dtKUIfo/jSNouN7IrQ+
+1bV1+WTP89j6Fzr8TInUG3FZljmu4b2+9Xi1wr+conJ5quM7toAPpfTMVcuLJIWi
+qIAOnDZFvJG3p6pLL0b462FX4CjOgOfb+9e8hDRHVfsm1yEo4m4QYYFwWruA6t9G
+Ge/BjEGZbahbQzhKjKNPQ3BeUgzzJskapAF4hk25d8b7kG4yJ5tG7XEw+swYKhkd
+I/Hhej74LGX1JcDwYwRZig6l38JHnIaKIi8M7l6fh/WZrk82D14ZDveZNBwgxsOa
+jvI61431g7SrSwkVBre+2FYrOH2JsqO4+gRbNHZcnrf3faeDHIK4xcZtDE6T2guG
+82i02KvPNenrxaG9yydjKNasQ3QUWCZFbVsKeyJG5qS8KkQ9c4TyJ2awpcAdYx6J
+LqZhctZWg4V4gr6ILeND7fwKC4aXMv+RmWamZ1JC81cQZcwW7Z+BxxabpxovwaLJ
+8DIVMTpG00LlURiIq1sjZCqYhhOaO9IplZv7I0VUFYPpqA==
+-----END CERTIFICATE-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/server.crt work/easy-rsa-3.0.5/easyrsa3/keys/server.crt
--- work/easy-rsa-3.0.5/easyrsa3/keys/server.crt	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/server.crt	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,135 @@
+Certificate:
+    Data:
+        Version: 3 (0x2)
+        Serial Number:
+            c5:46:84:f2:c2:47:df:a2:0f:ae:49:a4:0e:40:8b:80
+    Signature Algorithm: sha256WithRSAEncryption
+        Issuer: C=CN, ST=JS, L=SZ, O=netgear, OU=netgear, CN=netgear CA/emailAddress=mail@netgear.com
+        Validity
+            Not Before: Jan 20 13:34:58 2019 GMT
+            Not After : Jan 17 13:34:58 2029 GMT
+        Subject: C=CN, ST=JS, L=SZ, O=netgear, OU=netgear, CN=server/emailAddress=mail@netgear.com
+        Subject Public Key Info:
+            Public Key Algorithm: rsaEncryption
+                Public-Key: (4096 bit)
+                Modulus:
+                    00:cb:d7:9e:4a:5f:65:70:9d:5b:50:d7:d6:5d:16:
+                    8d:e6:2d:ea:e6:93:1c:1b:40:4c:e4:1c:59:e2:01:
+                    01:55:ae:8a:38:92:f5:27:f6:1e:46:e0:7c:0a:cc:
+                    17:e1:7e:e9:95:5d:aa:6b:94:6f:84:e0:70:66:3d:
+                    13:2a:90:45:8a:b7:1e:a4:b2:ae:46:a7:7e:52:7e:
+                    31:da:94:be:8d:40:31:6f:83:7e:b4:c6:30:c3:31:
+                    76:f0:ec:da:a6:61:0c:ec:9c:44:fb:5a:50:79:1b:
+                    c8:22:96:39:c1:3d:1b:e9:59:2e:09:aa:ad:ad:b4:
+                    bf:5e:4b:fd:40:2b:0b:f0:65:a7:36:4c:12:7a:17:
+                    6c:1e:3e:0e:56:93:9f:48:fc:c3:c2:02:b2:0b:5a:
+                    fd:7c:06:18:31:3e:df:68:2e:3e:9b:b6:2a:1c:66:
+                    98:f1:9c:f8:01:52:fe:23:49:4f:cb:c5:a7:a3:4a:
+                    1c:4d:78:87:ad:a2:b1:6d:e0:61:49:9f:77:aa:bf:
+                    c8:b8:4b:39:a8:33:f2:e9:9e:67:96:16:e9:70:7b:
+                    93:4f:1c:bc:91:9b:04:0b:01:aa:a7:57:6c:1e:c1:
+                    6c:93:2e:40:28:54:1b:bf:66:6b:1f:dd:50:e2:c2:
+                    c9:68:4a:ce:bc:4a:72:65:e0:0b:33:c7:8d:a2:d1:
+                    dc:0e:e0:86:9c:90:b6:a6:d0:38:0e:29:f6:fb:a1:
+                    4b:1b:d8:ec:23:03:25:51:47:f4:51:43:6d:ff:d1:
+                    d3:d9:51:a1:25:34:bc:41:9d:f7:18:7a:c7:28:27:
+                    ea:a8:10:aa:16:c7:9c:6d:f3:d0:b0:ab:38:f5:17:
+                    20:72:f6:0a:7f:ee:fb:66:25:d9:06:f8:85:7e:fa:
+                    54:b7:fa:51:31:ba:fc:95:b5:f7:d2:b8:01:d8:ca:
+                    99:b2:86:87:1d:49:58:01:95:b7:be:ac:dc:e6:c4:
+                    a0:70:df:01:6a:43:76:8b:60:2d:6a:87:1a:50:1f:
+                    fc:c7:1f:2a:23:32:57:9e:66:34:d3:98:a4:29:9f:
+                    46:da:6f:14:5f:52:79:ff:61:47:14:eb:e1:db:b3:
+                    0b:8b:b5:6b:1c:06:5e:d2:55:86:43:44:09:b4:30:
+                    6b:65:89:2a:b3:5a:91:c2:c8:45:00:c0:3d:c9:78:
+                    68:39:cf:f9:e8:02:d7:dc:8f:38:db:8c:e4:dd:a5:
+                    fb:dc:a0:ed:b0:7c:69:ea:b8:c0:0e:b1:28:ad:5c:
+                    48:76:25:8d:82:8c:4b:97:d1:ce:9d:42:c6:99:e3:
+                    6b:75:61:46:3c:b2:00:b4:f4:36:53:7b:8c:8d:58:
+                    2e:08:12:4e:95:a9:3f:2f:05:ac:65:38:36:91:6d:
+                    9a:25:73
+                Exponent: 65537 (0x10001)
+        X509v3 extensions:
+            X509v3 Basic Constraints: 
+                CA:FALSE
+            X509v3 Subject Key Identifier: 
+                32:1F:A9:66:8A:1B:0D:1E:B4:E3:6A:C3:A0:1F:0F:23:82:CF:3D:0E
+            X509v3 Authority Key Identifier: 
+                keyid:A6:B8:E4:12:D9:BC:F2:80:F6:FB:19:4B:02:F5:44:33:07:79:20:E8
+                DirName:/C=CN/ST=JS/L=SZ/O=netgear/OU=netgear/CN=netgear CA/emailAddress=mail@netgear.com
+                serial:F9:45:83:D5:65:7C:4F:41
+
+            X509v3 Extended Key Usage: 
+                TLS Web Server Authentication
+            X509v3 Key Usage: 
+                Digital Signature, Key Encipherment
+            X509v3 Subject Alternative Name: 
+                DNS:server
+    Signature Algorithm: sha256WithRSAEncryption
+         1d:de:ef:c8:4a:e3:6f:09:1d:fa:97:31:e7:7f:3d:98:57:ed:
+         94:cb:a1:06:37:fd:a5:f2:aa:cf:8a:31:c1:60:93:49:31:bd:
+         d1:d1:60:b4:5b:12:91:f4:7e:f6:8c:a2:9d:15:fb:a7:17:ad:
+         97:65:81:3f:ca:78:7d:4f:03:a5:f0:b8:c1:65:5b:89:20:03:
+         0d:e4:57:61:ba:57:da:14:ca:62:12:d6:bf:a1:71:ef:9e:bb:
+         58:4c:83:08:4f:0f:73:4b:8b:4b:6c:c4:e1:13:ec:56:18:df:
+         ae:ea:f9:5f:63:59:95:4d:7e:97:85:60:ee:c3:84:92:30:64:
+         67:52:a9:33:08:48:3a:5e:22:43:e1:d8:c7:61:c8:ab:e3:4f:
+         d2:f5:89:47:99:46:4c:2d:32:fe:e5:95:86:b2:ac:bf:b8:21:
+         6b:29:9f:a1:01:1a:6d:d9:04:50:84:8c:5a:de:cc:39:10:4f:
+         3f:c5:52:05:06:af:89:fa:4e:f1:f7:3e:95:7d:d8:ce:69:f4:
+         37:65:8f:ff:3e:68:23:d3:32:de:03:54:be:68:ad:e4:81:4b:
+         db:23:1c:61:4d:d0:bd:84:76:68:2f:eb:53:33:c4:bd:ce:eb:
+         d2:c2:87:bc:b7:2b:b9:3a:6f:76:9d:00:61:1d:22:d2:06:c0:
+         bf:cb:cd:b9:3e:b6:a3:79:cf:95:10:1b:04:0c:71:fe:26:fb:
+         9f:8d:54:72:21:aa:53:90:04:57:a4:b3:84:98:d8:59:d4:2d:
+         98:18:89:c9:65:99:2c:85:af:1d:83:96:f6:8e:c6:be:01:ab:
+         3a:63:74:28:67:6a:c5:c7:fa:ac:7d:30:ae:a5:15:be:78:fc:
+         38:60:e6:ec:31:4a:b6:af:de:06:68:36:f2:22:33:5d:a2:fe:
+         73:fd:4c:eb:af:71:d7:18:89:ce:d0:36:8b:a3:d6:ca:50:27:
+         5d:0e:8c:42:3e:4a:a0:b2:c0:39:42:83:b0:c1:e6:81:aa:84:
+         4a:8a:e2:ce:af:11:1f:82:85:fd:e8:a2:4f:d8:1f:1d:f4:71:
+         48:71:32:cd:93:25:61:99:e3:62:cd:4f:af:41:92:b2:49:70:
+         62:aa:d5:20:d7:95:77:b8:36:bc:c8:99:b4:f1:e1:13:7b:fd:
+         13:41:7d:26:be:ed:b5:d7:4d:2d:cd:2c:99:60:b1:48:24:f7:
+         3c:f1:c2:33:26:37:b2:9f:e0:33:66:47:01:9a:27:45:67:96:
+         6d:5c:43:81:81:6e:78:9a:48:26:5b:bc:c1:7d:51:15:29:f9:
+         f0:d5:07:c7:40:cb:88:1f:85:b0:74:29:73:b7:20:6a:51:45:
+         dc:0f:36:16:20:31:97:82
+-----BEGIN CERTIFICATE-----
+MIIGqDCCBJCgAwIBAgIRAMVGhPLCR9+iD65JpA5Ai4AwDQYJKoZIhvcNAQELBQAw
+gYExCzAJBgNVBAYTAkNOMQswCQYDVQQIDAJKUzELMAkGA1UEBwwCU1oxEDAOBgNV
+BAoMB25ldGdlYXIxEDAOBgNVBAsMB25ldGdlYXIxEzARBgNVBAMMCm5ldGdlYXIg
+Q0ExHzAdBgkqhkiG9w0BCQEWEG1haWxAbmV0Z2Vhci5jb20wHhcNMTkwMTIwMTMz
+NDU4WhcNMjkwMTE3MTMzNDU4WjB9MQswCQYDVQQGEwJDTjELMAkGA1UECAwCSlMx
+CzAJBgNVBAcMAlNaMRAwDgYDVQQKDAduZXRnZWFyMRAwDgYDVQQLDAduZXRnZWFy
+MQ8wDQYDVQQDDAZzZXJ2ZXIxHzAdBgkqhkiG9w0BCQEWEG1haWxAbmV0Z2Vhci5j
+b20wggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDL155KX2VwnVtQ19Zd
+Fo3mLermkxwbQEzkHFniAQFVroo4kvUn9h5G4HwKzBfhfumVXaprlG+E4HBmPRMq
+kEWKtx6ksq5Gp35SfjHalL6NQDFvg360xjDDMXbw7NqmYQzsnET7WlB5G8giljnB
+PRvpWS4Jqq2ttL9eS/1AKwvwZac2TBJ6F2wePg5Wk59I/MPCArILWv18BhgxPt9o
+Lj6btiocZpjxnPgBUv4jSU/LxaejShxNeIetorFt4GFJn3eqv8i4SzmoM/LpnmeW
+Fulwe5NPHLyRmwQLAaqnV2wewWyTLkAoVBu/Zmsf3VDiwsloSs68SnJl4Aszx42i
+0dwO4IackLam0DgOKfb7oUsb2OwjAyVRR/RRQ23/0dPZUaElNLxBnfcYescoJ+qo
+EKoWx5xt89Cwqzj1FyBy9gp/7vtmJdkG+IV++lS3+lExuvyVtffSuAHYypmyhocd
+SVgBlbe+rNzmxKBw3wFqQ3aLYC1qhxpQH/zHHyojMleeZjTTmKQpn0babxRfUnn/
+YUcU6+HbswuLtWscBl7SVYZDRAm0MGtliSqzWpHCyEUAwD3JeGg5z/noAtfcjzjb
+jOTdpfvcoO2wfGnquMAOsSitXEh2JY2CjEuX0c6dQsaZ42t1YUY8sgC09DZTe4yN
+WC4IEk6VqT8vBaxlODaRbZolcwIDAQABo4IBHDCCARgwCQYDVR0TBAIwADAdBgNV
+HQ4EFgQUMh+pZoobDR6042rDoB8PI4LPPQ4wgbYGA1UdIwSBrjCBq4AUprjkEtm8
+8oD2+xlLAvVEMwd5IOihgYekgYQwgYExCzAJBgNVBAYTAkNOMQswCQYDVQQIDAJK
+UzELMAkGA1UEBwwCU1oxEDAOBgNVBAoMB25ldGdlYXIxEDAOBgNVBAsMB25ldGdl
+YXIxEzARBgNVBAMMCm5ldGdlYXIgQ0ExHzAdBgkqhkiG9w0BCQEWEG1haWxAbmV0
+Z2Vhci5jb22CCQD5RYPVZXxPQTATBgNVHSUEDDAKBggrBgEFBQcDATALBgNVHQ8E
+BAMCBaAwEQYDVR0RBAowCIIGc2VydmVyMA0GCSqGSIb3DQEBCwUAA4ICAQAd3u/I
+SuNvCR36lzHnfz2YV+2Uy6EGN/2l8qrPijHBYJNJMb3R0WC0WxKR9H72jKKdFfun
+F62XZYE/ynh9TwOl8LjBZVuJIAMN5FdhulfaFMpiEta/oXHvnrtYTIMITw9zS4tL
+bMThE+xWGN+u6vlfY1mVTX6XhWDuw4SSMGRnUqkzCEg6XiJD4djHYcir40/S9YlH
+mUZMLTL+5ZWGsqy/uCFrKZ+hARpt2QRQhIxa3sw5EE8/xVIFBq+J+k7x9z6VfdjO
+afQ3ZY//Pmgj0zLeA1S+aK3kgUvbIxxhTdC9hHZoL+tTM8S9zuvSwoe8tyu5Om92
+nQBhHSLSBsC/y825Prajec+VEBsEDHH+JvufjVRyIapTkARXpLOEmNhZ1C2YGInJ
+ZZksha8dg5b2jsa+Aas6Y3QoZ2rFx/qsfTCupRW+ePw4YObsMUq2r94GaDbyIjNd
+ov5z/Uzrr3HXGInO0DaLo9bKUCddDoxCPkqgssA5QoOwweaBqoRKiuLOrxEfgoX9
+6KJP2B8d9HFIcTLNkyVhmeNizU+vQZKySXBiqtUg15V3uDa8yJm08eETe/0TQX0m
+vu21100tzSyZYLFIJPc88cIzJjeyn+AzZkcBmidFZ5ZtXEOBgW54mkgmW7zBfVEV
+Kfnw1QfHQMuIH4WwdClztyBqUUXcDzYWIDGXgg==
+-----END CERTIFICATE-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/client.crt work/easy-rsa-3.0.5/easyrsa3/keys/client.crt
--- work/easy-rsa-3.0.5/easyrsa3/keys/client.crt	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/client.crt	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,133 @@
+Certificate:
+    Data:
+        Version: 3 (0x2)
+        Serial Number:
+            31:85:09:27:b9:ae:15:e3:2e:36:0e:2d:8f:45:bd:73
+    Signature Algorithm: sha256WithRSAEncryption
+        Issuer: C=CN, ST=JS, L=SZ, O=netgear, OU=netgear, CN=netgear CA/emailAddress=mail@netgear.com
+        Validity
+            Not Before: Jan 20 13:35:33 2019 GMT
+            Not After : Jan 17 13:35:33 2029 GMT
+        Subject: C=CN, ST=JS, L=SZ, O=netgear, OU=netgear, CN=client/emailAddress=mail@netgear.com
+        Subject Public Key Info:
+            Public Key Algorithm: rsaEncryption
+                Public-Key: (4096 bit)
+                Modulus:
+                    00:b2:3d:a1:cb:83:e8:bb:86:99:fb:6d:d1:17:14:
+                    a7:b0:31:b4:f5:38:e1:1c:8c:ac:9a:e9:df:f7:a3:
+                    8c:99:60:11:4e:49:8b:9f:3b:b4:9d:6a:b7:3f:e5:
+                    0f:ed:8b:3d:11:d4:70:67:e9:9d:e4:3b:2d:f9:68:
+                    3c:fb:c1:b2:69:77:e4:35:73:f8:d6:db:44:ae:5c:
+                    6d:c5:b6:5a:66:88:90:37:74:26:53:31:40:8f:be:
+                    89:74:3b:82:d1:4f:52:ee:df:c2:33:0a:a9:11:e6:
+                    65:b7:28:5b:24:9b:e5:30:09:9e:eb:47:78:04:9e:
+                    cc:77:f2:6f:9c:e0:16:26:86:d9:e5:67:bd:cd:5d:
+                    9d:8f:0b:21:16:eb:3f:1c:24:be:b4:ed:31:f9:b0:
+                    0e:96:e2:fe:a0:a5:b2:b4:8f:f8:5e:b6:eb:8f:a2:
+                    c3:76:3d:da:9d:36:bd:bc:af:a4:85:d1:f0:e0:0b:
+                    05:61:4f:5d:65:f5:8d:7f:0a:df:a0:8f:49:a5:ca:
+                    83:35:fb:85:eb:9c:62:71:b3:57:e9:a8:08:45:1d:
+                    a2:16:d1:5f:93:18:08:62:e4:20:04:f6:26:38:c2:
+                    56:8d:18:41:4b:62:b1:31:c1:51:04:77:93:63:b5:
+                    d3:73:d5:46:30:a9:8d:cb:2f:fe:4c:dd:35:6f:11:
+                    dc:b6:e6:29:0b:b8:f9:c1:e0:1e:64:8a:dc:b0:6b:
+                    5b:f6:e2:10:a9:bb:ce:ff:f1:39:67:01:08:f9:14:
+                    79:30:8d:93:c9:7e:6f:5b:fa:4c:a5:de:8c:7d:f6:
+                    2a:26:70:48:52:9f:d8:69:1f:fb:4a:0e:14:63:1f:
+                    c8:a6:77:aa:b3:49:18:03:76:2e:d5:f7:5b:72:d9:
+                    40:35:b9:f1:2e:02:87:45:89:67:52:8f:78:7b:75:
+                    41:b9:21:95:de:c9:dc:3f:5b:b4:d8:93:a4:60:12:
+                    00:2d:ba:0e:d2:52:10:a1:50:40:8f:57:b7:73:bb:
+                    34:d9:8c:01:fd:a8:d9:23:c5:09:7d:b4:30:8d:c0:
+                    f1:37:d7:51:12:ad:69:bd:98:1e:08:c7:77:c9:d7:
+                    7a:10:ea:0e:93:62:84:2e:e3:1e:b1:81:a7:c1:e5:
+                    9a:95:f3:93:f7:b7:ff:79:5a:35:4a:d3:1b:da:ac:
+                    18:5d:e5:86:2e:3f:ce:18:fa:17:6e:0f:1d:97:8a:
+                    36:29:1b:22:36:e6:ca:b8:ea:2e:f6:b0:3b:78:db:
+                    e8:26:4a:24:a2:37:59:e4:fe:7d:1f:d2:e9:1d:46:
+                    78:3d:5a:a9:23:91:d7:14:22:fe:44:4b:94:48:19:
+                    aa:f1:e7:3f:a9:ef:d7:bd:3c:cb:93:0b:3e:22:dc:
+                    7f:26:d3
+                Exponent: 65537 (0x10001)
+        X509v3 extensions:
+            X509v3 Basic Constraints: 
+                CA:FALSE
+            X509v3 Subject Key Identifier: 
+                5D:1B:2B:AA:3C:D4:96:5E:31:63:D0:79:CF:89:5D:EB:E0:48:9A:32
+            X509v3 Authority Key Identifier: 
+                keyid:A6:B8:E4:12:D9:BC:F2:80:F6:FB:19:4B:02:F5:44:33:07:79:20:E8
+                DirName:/C=CN/ST=JS/L=SZ/O=netgear/OU=netgear/CN=netgear CA/emailAddress=mail@netgear.com
+                serial:F9:45:83:D5:65:7C:4F:41
+
+            X509v3 Extended Key Usage: 
+                TLS Web Client Authentication
+            X509v3 Key Usage: 
+                Digital Signature
+    Signature Algorithm: sha256WithRSAEncryption
+         28:85:8b:7c:e1:e4:37:02:f9:38:7c:d1:5b:b9:77:4a:2a:2a:
+         2f:74:a8:1a:23:4f:04:1b:92:9b:03:2e:b9:3f:f4:4e:54:d8:
+         d0:5d:b7:06:c2:15:8e:d4:db:6c:88:9b:31:bb:72:84:a3:09:
+         8d:05:50:ba:c2:36:a2:c2:e9:cf:e6:4b:43:e1:bb:90:5b:5f:
+         ca:e1:e0:2d:b3:c7:1d:18:1e:5e:b3:02:01:9b:aa:cd:a7:ee:
+         c7:74:78:ea:fc:50:e5:68:94:bd:8a:bd:64:7b:60:4a:64:65:
+         86:f6:f3:58:0c:6a:b7:6f:3b:9c:da:35:41:a0:fb:e9:e4:1e:
+         ef:c0:a1:88:2e:98:e3:d9:cb:f2:ca:ba:30:89:73:29:29:8e:
+         c4:9b:d1:ef:c9:b5:b5:41:0c:f4:2d:90:af:a0:82:4f:ea:b2:
+         c8:be:60:a5:87:4b:1e:87:6f:99:8c:dc:4c:37:bb:a3:00:40:
+         46:2f:d8:ed:73:9f:15:db:30:b2:5e:9e:25:c3:b1:55:1d:ce:
+         87:33:68:54:ac:45:0e:a2:eb:75:cd:d8:4b:a8:7c:20:37:17:
+         3d:ce:e0:38:85:27:65:a9:d2:57:94:ad:7a:32:ef:c7:5a:54:
+         c0:c5:bd:ff:55:c6:ae:2b:f4:7a:dc:47:be:bd:4c:d8:dd:92:
+         3b:23:f7:ad:7f:fe:bf:98:ec:fd:7a:9a:b9:01:e2:5e:08:bf:
+         41:40:0b:a1:77:a2:17:91:d8:ed:0f:76:f5:65:69:8f:47:38:
+         e1:61:ee:5e:c6:6b:de:3f:af:f1:01:8e:a3:de:9a:bc:5d:78:
+         bd:fe:f1:0f:54:39:fc:74:81:ce:77:94:0d:0b:55:26:67:05:
+         b8:6f:4f:c0:ea:83:b5:d1:b0:b3:05:fa:9b:91:34:83:fc:3e:
+         1e:cc:3c:67:c0:f7:67:2f:4f:3e:81:41:ae:cc:aa:53:a7:83:
+         a9:91:fe:f0:eb:03:65:5c:0b:40:0a:4e:12:47:75:76:d2:bb:
+         0e:e2:a6:2a:d4:44:8a:b7:3b:4d:0a:54:64:1e:22:05:b0:e7:
+         4f:c0:3e:6c:33:14:74:3d:95:78:fe:e6:f9:e8:42:c8:8f:58:
+         71:ea:4b:fc:b7:ac:bc:17:3c:31:74:a1:7a:17:9e:86:f3:97:
+         11:71:b8:87:2f:12:a1:a2:c9:9c:37:59:36:b4:1d:fe:da:f6:
+         9b:44:f3:4e:40:49:c6:f5:c2:ae:36:a7:bd:14:03:fa:71:e0:
+         9b:e6:e6:a1:f7:89:ed:f9:a4:d7:94:9c:70:d8:40:a3:4d:41:
+         a8:8f:09:66:58:2b:c5:44:de:ea:a9:21:ae:a5:ec:31:26:f5:
+         f3:f7:5a:7c:ca:0e:89:a1
+-----BEGIN CERTIFICATE-----
+MIIGlDCCBHygAwIBAgIQMYUJJ7muFeMuNg4tj0W9czANBgkqhkiG9w0BAQsFADCB
+gTELMAkGA1UEBhMCQ04xCzAJBgNVBAgMAkpTMQswCQYDVQQHDAJTWjEQMA4GA1UE
+CgwHbmV0Z2VhcjEQMA4GA1UECwwHbmV0Z2VhcjETMBEGA1UEAwwKbmV0Z2VhciBD
+QTEfMB0GCSqGSIb3DQEJARYQbWFpbEBuZXRnZWFyLmNvbTAeFw0xOTAxMjAxMzM1
+MzNaFw0yOTAxMTcxMzM1MzNaMH0xCzAJBgNVBAYTAkNOMQswCQYDVQQIDAJKUzEL
+MAkGA1UEBwwCU1oxEDAOBgNVBAoMB25ldGdlYXIxEDAOBgNVBAsMB25ldGdlYXIx
+DzANBgNVBAMMBmNsaWVudDEfMB0GCSqGSIb3DQEJARYQbWFpbEBuZXRnZWFyLmNv
+bTCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBALI9ocuD6LuGmftt0RcU
+p7AxtPU44RyMrJrp3/ejjJlgEU5Ji587tJ1qtz/lD+2LPRHUcGfpneQ7LfloPPvB
+sml35DVz+NbbRK5cbcW2WmaIkDd0JlMxQI++iXQ7gtFPUu7fwjMKqRHmZbcoWySb
+5TAJnutHeASezHfyb5zgFiaG2eVnvc1dnY8LIRbrPxwkvrTtMfmwDpbi/qClsrSP
++F6264+iw3Y92p02vbyvpIXR8OALBWFPXWX1jX8K36CPSaXKgzX7heucYnGzV+mo
+CEUdohbRX5MYCGLkIAT2JjjCVo0YQUtisTHBUQR3k2O103PVRjCpjcsv/kzdNW8R
+3LbmKQu4+cHgHmSK3LBrW/biEKm7zv/xOWcBCPkUeTCNk8l+b1v6TKXejH32KiZw
+SFKf2Gkf+0oOFGMfyKZ3qrNJGAN2LtX3W3LZQDW58S4Ch0WJZ1KPeHt1Qbkhld7J
+3D9btNiTpGASAC26DtJSEKFQQI9Xt3O7NNmMAf2o2SPFCX20MI3A8TfXURKtab2Y
+HgjHd8nXehDqDpNihC7jHrGBp8HlmpXzk/e3/3laNUrTG9qsGF3lhi4/zhj6F24P
+HZeKNikbIjbmyrjqLvawO3jb6CZKJKI3WeT+fR/S6R1GeD1aqSOR1xQi/kRLlEgZ
+qvHnP6nv1708y5MLPiLcfybTAgMBAAGjggEJMIIBBTAJBgNVHRMEAjAAMB0GA1Ud
+DgQWBBRdGyuqPNSWXjFj0HnPiV3r4EiaMjCBtgYDVR0jBIGuMIGrgBSmuOQS2bzy
+gPb7GUsC9UQzB3kg6KGBh6SBhDCBgTELMAkGA1UEBhMCQ04xCzAJBgNVBAgMAkpT
+MQswCQYDVQQHDAJTWjEQMA4GA1UECgwHbmV0Z2VhcjEQMA4GA1UECwwHbmV0Z2Vh
+cjETMBEGA1UEAwwKbmV0Z2VhciBDQTEfMB0GCSqGSIb3DQEJARYQbWFpbEBuZXRn
+ZWFyLmNvbYIJAPlFg9VlfE9BMBMGA1UdJQQMMAoGCCsGAQUFBwMCMAsGA1UdDwQE
+AwIHgDANBgkqhkiG9w0BAQsFAAOCAgEAKIWLfOHkNwL5OHzRW7l3SioqL3SoGiNP
+BBuSmwMuuT/0TlTY0F23BsIVjtTbbIibMbtyhKMJjQVQusI2osLpz+ZLQ+G7kFtf
+yuHgLbPHHRgeXrMCAZuqzafux3R46vxQ5WiUvYq9ZHtgSmRlhvbzWAxqt287nNo1
+QaD76eQe78ChiC6Y49nL8sq6MIlzKSmOxJvR78m1tUEM9C2Qr6CCT+qyyL5gpYdL
+HodvmYzcTDe7owBARi/Y7XOfFdswsl6eJcOxVR3OhzNoVKxFDqLrdc3YS6h8IDcX
+Pc7gOIUnZanSV5StejLvx1pUwMW9/1XGriv0etxHvr1M2N2SOyP3rX/+v5js/Xqa
+uQHiXgi/QUALoXeiF5HY7Q929WVpj0c44WHuXsZr3j+v8QGOo96avF14vf7xD1Q5
+/HSBzneUDQtVJmcFuG9PwOqDtdGwswX6m5E0g/w+Hsw8Z8D3Zy9PPoFBrsyqU6eD
+qZH+8OsDZVwLQApOEkd1dtK7DuKmKtREirc7TQpUZB4iBbDnT8A+bDMUdD2VeP7m
++ehCyI9YcepL/LesvBc8MXSheheehvOXEXG4hy8SoaLJnDdZNrQd/tr2m0TzTkBJ
+xvXCrjanvRQD+nHgm+bmofeJ7fmk15SccNhAo01BqI8JZlgrxUTe6qkhrqXsMSb1
+8/dafMoOiaE=
+-----END CERTIFICATE-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/ca.key work/easy-rsa-3.0.5/easyrsa3/keys/ca.key
--- work/easy-rsa-3.0.5/easyrsa3/keys/ca.key	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/ca.key	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
+MIIJKAIBAAKCAgEA2GMwBpJUy+NvcMHRkmEku1pJ/PMppddNpo5YmnhGn/fnLTOz
+UOvPDTCtqJQeC4PY78Ti139Q/sIVDi0iXN+7mc5LFM6J/PVJIMMKKEkr7SST0H0q
+dSQ7MpEnSYhpI8Llu622aBb9DsHtlvLckFSeqNd7zCboLfvY1B0sLYD5M5jydyCV
+pU2xIXxBObXinWOUgUwo6U4NKchKhrBZ1Cx9QO/aq73CLejO5QFKU+4vsOu71H7t
+qeRXsMF1EfqEnxbtTMzrWdR9As+OLomg8C/ms7yCmoDQoXmJiJ0OIWdyD4mmZYtD
+6+RSSMkKtH12tiZER6NmPszZiWxMIrV8XUbQPEWzRpK9MJyRwnCAhp95926NYMc5
+RYNfbinATEKHq/HOqCvFMqo34g6RSXoehtAwGXW55b41H+3rOW9cjDejRVAQ5dzY
+tuxYHjKvhxKCNZtWY5fAlSx9Nm53wSxy3vEcUFMe4K9346whpqX9kePYRzKx/14r
+pjIcP5C1iBYyx7ZGbDAsLCTAp/vgxrDC1Uiua/5dMRIf5L5GQwi1ZUX3lhuxdWXa
+kBuMXm5NnLzmG14Dv8O+evDuGGLLaVpbN55pZ/Q0gdhyJO23MUHGbxVflHj0v+to
+BxMPZktA48tvkjq21cYyxy+laQBB3Fa97BsFxi+0dPzawudBqBtaFVNdRIUCAwEA
+AQKCAgBbCT/HRAEQ1lqoIaK95sjBWbcPCG5ZKz+r9u8n7booyu8jgUDLpqmxbu1g
+oxAQ4COAnGCjMRkuVRDuqeZvFGGkz7mew1Yiaj+d3PCFCmPEnULi0AXwypFtMMxK
+j21Mck6cqvS2zhUhBE2k2AAbaL5nYmDe/u9niTEr7CoN6tjs8Efx/m924DzJTbFT
+SkZQ2Pg9fYEoxoGbNnzENaVVMnJvw3lID8mF7xA8IFLZBLhjwol/WPc8JqkN4BYU
+vkTVeZvUV9IcNN1zNUcoJRbkdgKlU4XO4qrjhHJsW0b6nr7J8qtixTiKIcDMhmhO
+jwpGj8riKMh3W0+aFUniwF/WDljEfLMd8SAml8JyYdJdRZaozOY72qKEbZ5EzWAc
+5ZzAkyW7EQCF3zBH76vps8GY8SIm/DYPA/uSixzHCZA+dJhOfsrGdqc00B9kR1Rh
+KCZn0Hxs/fo+179/XCOzRCLNvySUnA19S3Ibn+q9IbZgLWosLaGyhbEzNyGmA3/o
+ecSRhYXpJ2bqAfDrL/F6lBZIvx4MFVCeSG1YmkJBqkWUxsWIxYnkygPU20UKC5L7
+q+I69jV4eu70QHutvayYsNbvqLFIxBr+ioljpknwGawcq4UUBVLiireFdPFn2DEs
+Np2eDhI4lQa/uUDgbYo/qP+qBcAnzbwdzmzaws/jdV0BmbpV8QKCAQEA7MpH9Jev
+7O9fq3Lpg8u1K5KIku+LjpCInhCpVWIeG1uBNRdOHI0266GPptch5EnGiDf/mI9u
+ZvuAPCopbl7hcNiIL9jhHF8aLpTEgj9/L/zWVyXnd3NM9lbkjVilxsFFEtr28l6p
+ZWTETS8rzdwIvgmegFrxisQzXSbvWBDIQRcVEbmHQ7zHHZm0Dob/E7xb+7RT9Pzp
+Ycen9rWy5zveX3K1JYvh/4NW8XspotxSkHylUgBac9kCHPV3h6tjSppbPOpfCnQr
+qJ5DFTHgnG2tjNs2zKgekysh3Pa85abjBM9bbXrVxP2qCxL3cVyz+NumNLBQE/V+
+9N7GaFX55eIhMwKCAQEA6fEthbCEXRn9tjDK+bL0DP0Q3tiSeePIlJdKCVHqPaNW
+lnSifE1+m2rKzQvHiaNS5+PEjejt0OyCb2fwuWISnrKdw8AI/xAv0J+jqKpBvQ28
+saAq4nmsAeLYUJu1v8gpN2gcdkeff6/zNDEPVLEFvv4rGDzYrPGk+yl3j9kGrjLh
+fLTQGdkVaYGLIrojoP6V+iXzuMpIucxzNxIW5wMteZfAeZHzFdDAUsViNsQoxCon
+/w/ztDwP+9U3jpO/YmVNEK+J0HsmP5YHp7tO20XqPhNyML4yvINmsgWgYbM/uMep
+hS4mYIK9YB53rUTRIi50rWVqcp6wp/jUDr7QrahzZwKCAQAryXAq683te4pucu55
+n7bd2tgEi+V7tIB+8kfN8TnuByrKfYfFQdvREOw2hvvOclITKH/MX2GkC4f4GDfX
+IBaCPphYeLuP+ool6P65szI61n9wWCMIdLrbps/Wz3jAW0cBqJoX0adzh4HBYCKy
+i6jWM4Mq6yZVadixthx2ZYwSdUlBINTBkuwjLiQct5v9s0wzLHKGECq8GSaXOZxe
+pDSH4/dZfReKn2f5ebQGWMWv0Ezl8AjJ+WwpnL7AyNGwnwCWMCTpEvhr0vBGwBVv
+99wq+vKUP9Z/km9NpBbkRMsgPZQDl75mEh5as+QsA1+Iel2uypfUSQZ0E6oO6y8t
+8nX/AoIBABOCChLA4njFBxQfQ33I/3Lp03NRyf0iEoLZjtHyI79aFIldVoNZDvq0
+0A6DMfN3TB2ML8LJrrMiVcomd7ttwzG6oU9GfzmNxv71oPcao3q5X4fMXGcgUEij
++vfV1DUW4uD87IqsdE94g6ZRhP7B+EGx/jP6NvNM/z32ekLjjRwPoB//D+9n29I9
+fDW8IeZ19Fm0unspWyErPFxWRNSkvjEKh4Fbu1KRYshEL2lzuBaLkWRCl0brT0Xn
+xMc96NKysdd3W92nt0vMHf1NsEi2rs7YXkhUQ/TWXETyFSqwgNrOw4MYkxivLk2K
+nN41A99+0jn3YAYI16BaLozAOrSQfHcCggEBALv/QgZ1EtBTevvaAQddwSfvL3mD
+a8OcoReqqHCN3Va/aYFeNoUXG1+rLSDc90XiMaXYrmv62q1Tmcq945sqiBL09UPx
+wNh/pxtGburIiswN9hQmXzClN3dBy20rMPrrYcaJVB4CF+g3i1iOq3ZBCqsWFTyv
+r5/AfGXSCxq5Lzo5wmEODU9w5q/9RvpssTrByKqvelvKLwwwqVwczUe4bBU5UBYb
+LtxgPimtgXj0wbWhwLEusavqNOIWy6/Okwq1CHZXNpsxbd1ghzS/zzpojr9rfiRR
+/0mcKEpNg3njWx4iOk3l5A7SPvH0yjjNy95Cz2F+KOH/r/jXNHyNGauD7P8=
+-----END RSA PRIVATE KEY-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/server.key work/easy-rsa-3.0.5/easyrsa3/keys/server.key
--- work/easy-rsa-3.0.5/easyrsa3/keys/server.key	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/server.key	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
+MIIJQgIBADANBgkqhkiG9w0BAQEFAASCCSwwggkoAgEAAoICAQDL155KX2VwnVtQ
+19ZdFo3mLermkxwbQEzkHFniAQFVroo4kvUn9h5G4HwKzBfhfumVXaprlG+E4HBm
+PRMqkEWKtx6ksq5Gp35SfjHalL6NQDFvg360xjDDMXbw7NqmYQzsnET7WlB5G8gi
+ljnBPRvpWS4Jqq2ttL9eS/1AKwvwZac2TBJ6F2wePg5Wk59I/MPCArILWv18Bhgx
+Pt9oLj6btiocZpjxnPgBUv4jSU/LxaejShxNeIetorFt4GFJn3eqv8i4SzmoM/Lp
+nmeWFulwe5NPHLyRmwQLAaqnV2wewWyTLkAoVBu/Zmsf3VDiwsloSs68SnJl4Asz
+x42i0dwO4IackLam0DgOKfb7oUsb2OwjAyVRR/RRQ23/0dPZUaElNLxBnfcYesco
+J+qoEKoWx5xt89Cwqzj1FyBy9gp/7vtmJdkG+IV++lS3+lExuvyVtffSuAHYypmy
+hocdSVgBlbe+rNzmxKBw3wFqQ3aLYC1qhxpQH/zHHyojMleeZjTTmKQpn0babxRf
+Unn/YUcU6+HbswuLtWscBl7SVYZDRAm0MGtliSqzWpHCyEUAwD3JeGg5z/noAtfc
+jzjbjOTdpfvcoO2wfGnquMAOsSitXEh2JY2CjEuX0c6dQsaZ42t1YUY8sgC09DZT
+e4yNWC4IEk6VqT8vBaxlODaRbZolcwIDAQABAoICAQCmBtnrk6WowtH0l8gGi+9J
+ss6nvNxgqc+24Aypks4GfqIgEDpTsRfZ/M+rb65nwRn5yY+cDguC+9cwvsAUo+uw
+XxQ7azmdXKD8CcMLWKmXnNhnDoAuN1rSMUDCXPh+GYAQW/unilkPO3yYBynFZcON
+HVDtGLJbUGOqJ1R/ZlBUHkfCZtWJ0fgC7eJ5y8UqCo5bQBzKOPNLiB8+RQpr7Xab
+FxVXio06BfLOCU3rabPvse6nhh5lTy2bVG8VonM7JwKt9YVD7ilpAd7xP075pXG/
+ZzeFcU95fPwfBt17bS4oIr0PZ9Vjw5XSIFZSXs2Z1LN9GaknHa/toD0fkIKhQjLs
+CLHvXHlKZTAFGn4hGIFEJQHu9ZrsIENyI9AGF9wqoeBV1XtVacTuqy4ddYQob6g1
+0YqnqCHUkOkjCVpls9XQmwmAsZ9JLtD5jgJjkLo3oWHIyL2aDYU042hSxd0azfE0
+pf9xDNLlJDQzNzqG/Kb18WRwtKd+4OERre2LEGYDzwJfZt0GG8/yJ68zaV1MgjBw
+RHPFLoTNKWhJT6Se47bX8ol25WPw9jkhAscdkGleYeqw99iO+XPpmYdSquZ8uUPe
+RQ3+P/Cz3LKtZ7s5TMwZLrGPi4sMyAva6B8CFT18Z4q8++pnHKw85qMo4NP6O6sQ
+b27Fw7ZtgCl3yHV9AE/rAQKCAQEA/0f43kQi/IovBpxCsjOTSjSVedml9w1dX8Il
+1BaJOty2654QWCp8RmwI0YZir2zh4QBl3cwOyuMW2zBYCjkH8GIOGMX9STXLbbI3
+/uGmn3um/P24BAC0A7eVPMSNW10r0/kEe5K5UBuG7Tuorr1i6OD6YMfT3q0Nri2U
+YJniMyMclL4PzrZf+p4ucCF8sUuRcSXuKn4NTNDNgnzYpfAIPCIN8bAKU9ApLQsz
+ckiKM3+1126S/VZFXM8RZvuwjlHJbhaXrJ42zSZpGjN1QXtYV5Anpm7SknQdbN8X
+3iDgq3YsOAeT3Kn5V3iQU2FqgGtc+oq6B61G+GEn/inw98PAgQKCAQEAzGqQlCOS
+BAAofl6m9yqeMzYC2w0/RLj/Gw7rtmnqKV7obaW/uEQXzIfa/jJxIP19rMYHPPfP
+PygNnX5v+uJri4uEyCIssvq8TBFjSHHFkkC1vDjQ/8/ZeT8O1lLM4WGG0MA+GjFo
+LPQHI8fOyvNgmuRRmqymgVFS0tm5sKign3v695S4LkG18yOCYMk72YjLK0GhKYbm
+pXcdo/thpRXJ4SljsXDZswZbN4ZCy7JpImD5e9t1IyX+Bcoesh7BEtHoTERsV+SC
+OCoo1zDEdwNMcIhp55ONW9VJXurNCa5QjYkPIyNCq74J1+6QGFe+Dw4/DuMKNhkM
+hZla3aBZAxTr8wKCAQAfugErForGmiHoD2mWniRTF5/wKeWfg7C0IvvylO5vO1kc
+hWOAoZ8aXXjnGW2QYdbc0ODYJPE9s+V2TlbSxm2/KWCZIZouDBqBQqIE8mRz87k4
+Pgwtqyy0PY+1orPyw+ybhppRBVY0p4coWr7HINB3IubbOlqRlS5bFF0ZeGaNdnKp
+5ckvJqldC2W4F+FYY4zq5AzqiQn8HwM7kWqXjPJpDs5u2DMgYYGMkJTPQkayyc9+
+u8ZW5+7s7us5wlQaC/4dyX8GPA+RhZGQaQ7MmZjMISVcYO+YtgemLT/ZInrptGEg
+gYiV5yYPAfPl+Ph0h2Nh2/Xs2FM2sVR4GRxSnqiBAoIBAEjv3RZ+eqMiy1ALxdd7
+RWdQZkRyvZhvydaFxGUZ9Ys9agF1mqrM4REBiIoaxXLX//ed0QdYBIKsb+86JgqG
+r4jjB0XBq+FQxc7+hUZcEfrVTAMU4r3kE9zHOywN7zyb59akQYRnDJy9la6WpJqV
+rGqX/OYLpH2P3tWNCkyWXnmJa0rbrbwThjBzAxo1sz0zKgsNJ8wBblqr3c+c6oJU
+++tzMTcm5QaTxghRx5OW3Rx/LYWcEHeyHqlCLSGxB4MHkrNj1qSlzDDuGhRuOiuc
+6bFdeyw94bpaqAtODErf/64AsX4AHtHYxPpfDNsis5X7p1RQflw/I9Cnn1xk0l3f
+ITECggEAAQhlcJsGECzxKoBuYSJe1hdMfrS1tYn7CQQtVuat19FmO/fVfN06ner9
+ch69aXeWdPeHp90BN/gftCJ/sXjA/nTmu6O2FXa1+Ik8b8m1nJs145aBs0nBChTc
+JY9fQQoOUQUbIjHz5uHcoYvuHS7XpIAvEprz7sgshd/AzOm8wSl+MbOW5mYemdGV
+MOh5sBXx0AWTWwKNuzgHIIGRJ9DNCQETkfTSACF1cUJnCKJKjXb/0gIUmTooyEZY
++7LWJqHAGWcauhEQxomWGGQwAgxC/mXniYfqB1pHHgY39qbMcnN3FEmaJDmZh3Qt
+qKNbpfOZQ79satGQHXcdAy6JKCsw6g==
+-----END PRIVATE KEY-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/client.key work/easy-rsa-3.0.5/easyrsa3/keys/client.key
--- work/easy-rsa-3.0.5/easyrsa3/keys/client.key	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/client.key	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
+MIIJQwIBADANBgkqhkiG9w0BAQEFAASCCS0wggkpAgEAAoICAQCyPaHLg+i7hpn7
+bdEXFKewMbT1OOEcjKya6d/3o4yZYBFOSYufO7Sdarc/5Q/tiz0R1HBn6Z3kOy35
+aDz7wbJpd+Q1c/jW20SuXG3FtlpmiJA3dCZTMUCPvol0O4LRT1Lu38IzCqkR5mW3
+KFskm+UwCZ7rR3gEnsx38m+c4BYmhtnlZ73NXZ2PCyEW6z8cJL607TH5sA6W4v6g
+pbK0j/hetuuPosN2PdqdNr28r6SF0fDgCwVhT11l9Y1/Ct+gj0mlyoM1+4XrnGJx
+s1fpqAhFHaIW0V+TGAhi5CAE9iY4wlaNGEFLYrExwVEEd5NjtdNz1UYwqY3LL/5M
+3TVvEdy25ikLuPnB4B5kitywa1v24hCpu87/8TlnAQj5FHkwjZPJfm9b+kyl3ox9
+9iomcEhSn9hpH/tKDhRjH8imd6qzSRgDdi7V91ty2UA1ufEuAodFiWdSj3h7dUG5
+IZXeydw/W7TYk6RgEgAtug7SUhChUECPV7dzuzTZjAH9qNkjxQl9tDCNwPE311ES
+rWm9mB4Ix3fJ13oQ6g6TYoQu4x6xgafB5ZqV85P3t/95WjVK0xvarBhd5YYuP84Y
++hduDx2XijYpGyI25sq46i72sDt42+gmSiSiN1nk/n0f0ukdRng9WqkjkdcUIv5E
+S5RIGarx5z+p79e9PMuTCz4i3H8m0wIDAQABAoICAQCcc1ayIZ3GBJbiR/lRDhNG
+p8rTuFJ30c6rx3+MiaUe92EZW5rPcQPPbN1emzxI4IABR3xG6ZtqurtXATUOMm3c
+/5xzgtE4Ho0BkyXTdoWRHGuiETSWr1OhGLjkKlxgRwVfoU5/qOrQSHCdvTGLR797
+U+aHVFs3iv+khUuM4nDu7k0SfOXv/i0PYtBvopvpoyM2AkFsjimwLZ0c+3SJ4xk6
+YcsQb+irwA1FE8ild3Ks96M7tNDLInOzzD6f9bZm437cpI5ZcuaQ8drDPI19Db38
+lM8ywiLnJSAvVVf5ounXxlVcUZ72gbFRIANyLVdVHUN2mAydKCbK49yOMVj2njIh
+bN4cqfjL01nQJTTgtVPd3zmDf26ojlLTNwh9sQ6IJsEqpZcugm2RDASqlzee/FeE
+/JqqR8Ypad8jjDEEO1l8/GkGuktVeZGAtuyxwQxjCukG47xTh6+Z1t0sQqFRyvsP
+kmZYxWXk7AZbyY8Z02nEklZxYczSISPOIslZ/DeQSU7ecfYM53L8BlegVa1P052A
+eKURqiMvne+AfYIfQNEX1utDa6XiXt/oU2Vuq/+yyYGpQguYocUTwEstd+Npf6Ju
+4kTpRwpBcQNYcgXF52hz1yEiJ+ARN5aFe24eyJTUf+KvOcFY8qp0S5os7sHY4K7E
+ouMo2NCB2F1lCctidguQ8QKCAQEA4yaMM2JDYRpOjqEyRVG1ZqDkn+B0BhEyJsbp
+0cCdqbk1VpjjMxXwV8VWZtQpuankqurm+Ll/6TmppheFBz5NKzgXVzXse1QYKeC2
+Jal/lPMFUbwQPhjisABe6WzawLCgZlxvhYTjurpXjh1or9KSR7sjt3SVFCwApH+a
+5p1FRAyP7l1x38NJuvmbzV/3uBQXf8oFJcggM25yd6EV5PCxO2Xkd6HIWyTmvbzz
+CnUBHf3y2c7/aJG25u/iAMK31P12jk5N3cgkjfYhyp6tTFAhPiyKIhUUnS03zHj6
+DI8n3ARGSdianegsJ9xGM1c3tXPl5verr83pGwPXpB3DkyQZawKCAQEAyODbPate
+qto4Y9abMcONz/DC6TLKgJuUuYX9QckTCI20Gbs1jNbu8f+zdUiCzJnN49ckW7AP
+rullJFDmtlDQl/QfEgIsjM5k9XCzMJXKowRrhhcS5WRp/fOGrZqY6s+/a7shF2vI
+nAX2kdvCsI0/EHJtlB677pBETudszgm0LZ741RUSKZpkyKXG0oUZpZMYEnOWx7PB
+z9A2Jkply551y9peYnHE+NIB8X1RA3fKP027utI1/4ojhnYG3LP7FwAEn7u2SPt+
+ODcwEmmFXqzi8Q8qmYXnAlMaoshXAA8y3/ldbr0g+RSSu2GKIpYY3edojwcq72O7
+39Awrdxyc6n6OQKCAQA1wQiKnGVflE8k16bC2rw35jlUUFSEnwskPAxXM+LbqFWk
+AZg6naV7z9pLiUe6hSb0/5PbJmLLuXaZNJWpiifO4346iCtHYDY7qs7QCNWFVSFc
+H0okt9TlW6EaUkVONj4vGw7VJQaTSMexfOoqOYWebf3cyz0AQg2A4CWzD2K14Ncq
+zHhHQLqxAaLKLjy8Fh0dg5xGPZ7JLQjvoHJC9GasrMctPF7ims/z9MP7JIc4JsGS
+o7azc1A6RGhST8IbKEQnFwa8RkHISofioDxYg9nqEbY6rIxtSacbv0FFOZdwLJ7J
+mPW2CpxwbmM6gRhQ23V7e/7oIq1YVEdysgx3RKKzAoIBAEtsWgVSte3Sl7+s0YKi
+2TlNPfHPV0doozPxgkDUHahlhEzxBM9H2FnW12ve+PWXfj3hBs7CUXEsQwSat7WW
+rGuRhqT9WfrYeAHL+gvxAi+pNjIXscmdpUpryJKgbHAl5sqtbWRc3OrwNj9X00b1
+Ecu8Gza1P3z4gB1cOXc3H4bNnMJs8e/a/6Jy1C8xOIr5qt0qt19uOAQuOlawMn58
+leAfsral3v5J497nXaa+sJqKNkWULiAvnt0arcLqdQmSwSm0kmHew+qZlhwjsKb2
+pRM7zZPtOeqeh269aCUmj8SW3aT5sEq1lc5zdik8UEPpYlHpsA5IVT9Y7R4qmnKh
+TMkCggEBAL6sSf3oMl17bSnuF831gKWjAwHkrHQQD1mGvO7b4tIsL/tSaCSF1kTq
+49nGA+VVa6EQT650f4Muredi1KFoebz4ZWvU1HPVOnVOJc5t5fuOSJ9GF8SGN01L
+WcnefN+YzYdzuNfACFsxxKvjKCzobK4fyS3ihAvsP6RfXTXkOUfiu8q4ch85fb0H
+QRcxIV1NZ4PKrJB5RADRZ0VfpzVRgH9P1FYayjfIi2LAs3iB/ZM/AWJ8xbIf3nmX
+ePNO91zfBVO8zGuM4FIFiYawikNecIlRDQmXCkR4zS2lSe8j0sVM8HQHYRlfHIPu
+2LP6KFHrnlU2VO7Lt8nbP8U8BC/RFUI=
+-----END PRIVATE KEY-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/server.csr work/easy-rsa-3.0.5/easyrsa3/keys/server.csr
--- work/easy-rsa-3.0.5/easyrsa3/keys/server.csr	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/server.csr	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE REQUEST-----
+MIIEwjCCAqoCAQAwfTELMAkGA1UEBhMCQ04xCzAJBgNVBAgMAkpTMQswCQYDVQQH
+DAJTWjEQMA4GA1UECgwHbmV0Z2VhcjEQMA4GA1UECwwHbmV0Z2VhcjEPMA0GA1UE
+AwwGc2VydmVyMR8wHQYJKoZIhvcNAQkBFhBtYWlsQG5ldGdlYXIuY29tMIICIjAN
+BgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAy9eeSl9lcJ1bUNfWXRaN5i3q5pMc
+G0BM5BxZ4gEBVa6KOJL1J/YeRuB8CswX4X7plV2qa5RvhOBwZj0TKpBFircepLKu
+Rqd+Un4x2pS+jUAxb4N+tMYwwzF28OzapmEM7JxE+1pQeRvIIpY5wT0b6VkuCaqt
+rbS/Xkv9QCsL8GWnNkwSehdsHj4OVpOfSPzDwgKyC1r9fAYYMT7faC4+m7YqHGaY
+8Zz4AVL+I0lPy8Wno0ocTXiHraKxbeBhSZ93qr/IuEs5qDPy6Z5nlhbpcHuTTxy8
+kZsECwGqp1dsHsFsky5AKFQbv2ZrH91Q4sLJaErOvEpyZeALM8eNotHcDuCGnJC2
+ptA4Din2+6FLG9jsIwMlUUf0UUNt/9HT2VGhJTS8QZ33GHrHKCfqqBCqFsecbfPQ
+sKs49RcgcvYKf+77ZiXZBviFfvpUt/pRMbr8lbX30rgB2MqZsoaHHUlYAZW3vqzc
+5sSgcN8BakN2i2AtaocaUB/8xx8qIzJXnmY005ikKZ9G2m8UX1J5/2FHFOvh27ML
+i7VrHAZe0lWGQ0QJtDBrZYkqs1qRwshFAMA9yXhoOc/56ALX3I8424zk3aX73KDt
+sHxp6rjADrEorVxIdiWNgoxLl9HOnULGmeNrdWFGPLIAtPQ2U3uMjVguCBJOlak/
+LwWsZTg2kW2aJXMCAwEAAaAAMA0GCSqGSIb3DQEBCwUAA4ICAQCbgwaZWiT+69Um
+1hnGserHCQpOrXH1JPpMpXOwnWDieaEDlbz4DOtpvsYt5QCZQXs7oavndasMO9Z5
+DPMNhK8wevVZhoiyXwqaVuNrS9BGn/4Qvgmff6gww6MjiD4FwIkprbg/kQu1n1Mw
+L0vTTXNAU7d465mc2Xp+V95n53dxtfAoJqirQbvU2u6TU15TTeggf/++Ug59umzA
+DEsifCa7qW3ULHGLHOYYKbchLyYFoHPIPi4mifqDbmFSLzKfZlYv/bkio0TRiWrv
+lhjP1rgtazQAIWPSLRgsEXm1FbEZ7090lu8dxr3QpkLy6iO/Dtpq4i1yGo6ACcQ/
+m1mkZpftRk1/mC0cBHAJtr36qynMhCrGmrg6y4Tp3ce7RF3/7McX37y2fUVpktoT
+m2DDqlvWgvaRkIduky8aOPeyO1fik0X9KgzfdjsfBEotMjDbLByqfgKh3MlFj1BL
+b3oNQNQvoviAH3vLVlCTUoRN74mY150L4dbBdZlLHPbwtBUqLc5NnX4rm74eYKQH
+Cj6k4IeGe/NWND1JxGTBgZQhJxislCcr6E/1hFy7T0gNTBy6KDI8gd27dvjMLFUP
+Vl7F44G/KUeUDjpCJgTWfnozjv6PE6VWvv0N3GJC3mom2sH3AIzgMjAzd8BfGhWh
+M0d8Ry2ztCO4zQ6m1rgXbnNUAlE1Fw==
+-----END CERTIFICATE REQUEST-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/client.csr work/easy-rsa-3.0.5/easyrsa3/keys/client.csr
--- work/easy-rsa-3.0.5/easyrsa3/keys/client.csr	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/client.csr	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE REQUEST-----
+MIIEwjCCAqoCAQAwfTELMAkGA1UEBhMCQ04xCzAJBgNVBAgMAkpTMQswCQYDVQQH
+DAJTWjEQMA4GA1UECgwHbmV0Z2VhcjEQMA4GA1UECwwHbmV0Z2VhcjEPMA0GA1UE
+AwwGY2xpZW50MR8wHQYJKoZIhvcNAQkBFhBtYWlsQG5ldGdlYXIuY29tMIICIjAN
+BgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAsj2hy4Pou4aZ+23RFxSnsDG09Tjh
+HIysmunf96OMmWARTkmLnzu0nWq3P+UP7Ys9EdRwZ+md5Dst+Wg8+8GyaXfkNXP4
+1ttErlxtxbZaZoiQN3QmUzFAj76JdDuC0U9S7t/CMwqpEeZltyhbJJvlMAme60d4
+BJ7Md/JvnOAWJobZ5We9zV2djwshFus/HCS+tO0x+bAOluL+oKWytI/4Xrbrj6LD
+dj3anTa9vK+khdHw4AsFYU9dZfWNfwrfoI9JpcqDNfuF65xicbNX6agIRR2iFtFf
+kxgIYuQgBPYmOMJWjRhBS2KxMcFRBHeTY7XTc9VGMKmNyy/+TN01bxHctuYpC7j5
+weAeZIrcsGtb9uIQqbvO//E5ZwEI+RR5MI2TyX5vW/pMpd6MffYqJnBIUp/YaR/7
+Sg4UYx/Ipneqs0kYA3Yu1fdbctlANbnxLgKHRYlnUo94e3VBuSGV3sncP1u02JOk
+YBIALboO0lIQoVBAj1e3c7s02YwB/ajZI8UJfbQwjcDxN9dREq1pvZgeCMd3ydd6
+EOoOk2KELuMesYGnweWalfOT97f/eVo1StMb2qwYXeWGLj/OGPoXbg8dl4o2KRsi
+NubKuOou9rA7eNvoJkokojdZ5P59H9LpHUZ4PVqpI5HXFCL+REuUSBmq8ec/qe/X
+vTzLkws+Itx/JtMCAwEAAaAAMA0GCSqGSIb3DQEBCwUAA4ICAQAck8E4IVa0eZyU
+QBTHFg4fPqBck590V5obXLQAHUqaoHGNkcZ1qVWsUzSJNoHakGTU2xA46lb8x4R4
+6Y/eZNJLeHNC7W0FXmfRVQZ/YjCcxoZ+/O6dJfysshc47IlwdELgycFPN9FH2haa
+Z9/wkzswPz1RggXyHRdg9XsZ6iUg93uMySYzsmWzNmVe9PrnyB/S6nvZ1/0AS0YK
+IN54rjOJb30PM3vWk5y6ziCqoxvdMr0FtQ2ePlZ3ZcJMhg1wVr1UhaNsRdt3qjot
+dEbb9AwrJMs1LY4E1xzbnBtmf/IdOmIuzpgTcvSvSe21t9I0S2DgqYoSdkfDAAxe
+a6CrWG2BqBmIimCLASQsmTcXMnDiZHLnI4CrDR/956aLglN71zFt33VeAfdp+JUD
+svp3F/YId2HsMI1R0Q58sLZeRfZ79HnQGsGvNRW3Lrmu26hINgMllDBB+/aQOEEA
+MGlOg4AvOz+5+aeia9xBdZmao45VjygQaQQvVVcs3Lt24EEq1qIUNQsv6FT3/t3/
+v/1NIZ5FobY10aYFHIb4MgYcIX2V7hXLazr3hnNo85V5HTYhsl2doTxLs/4bY8qj
+deUcFwIZox1FyY3mLYECpvkIqPAnhB/lZwkiE081pTNhhSNQq8tKEurR64gke22N
+WZcvT/E7diQT3WA/ECbyyscUXYNhTw==
+-----END CERTIFICATE REQUEST-----
diff -urN work/easy-rsa-3.0.5/easyrsa3/keys/dh2048.pem work/easy-rsa-3.0.5/easyrsa3/keys/dh2048.pem
--- work/easy-rsa-3.0.5/easyrsa3/keys/dh2048.pem	1970-01-01 01:00:00.000000000 +0100
+++ work/easy-rsa-3.0.5/easyrsa3/keys/dh2048.pem	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,13 @@
+-----BEGIN DH PARAMETERS-----
+MIICCAKCAgEA8nGfZvaHNDreodIHyd1SMJ93RymsIcCFFUCuIz/A7bieSdREKjTY
+BGqWh06RSmVxK6hRua034sYyhRI/c8EmPLQPKKeM4bSctYb7qx3Y42BFLVv/OfBX
+TYGbVtEXcGxQnUMwnbwO21oybnVsjziuAM7j0CmuBK/MPquGMaP4g3pJGu1Enxu+
+ngKt15j8piIvGZsxjiQaOqT+0aINGRNED126sH0TfQ+722nl2UK/4bUtDiHl4ikL
+14Nf69ooEyHqW7YqKfgvuZODNNMbVbRYV7I3W5Hv1imkJY9WONGQ/dzANB5TsnR6
+COWiO9AQsuD5vK3Lb2fle4SlQK8RYgy/UBw+O/RMa5BMbc3BTQ4d/8duNbfgjI3i
+JP+MFbyHnxvw1/2I2uqS+iHu4e7dd3xAIzuylIal6Tk72okfD+nd0Y5myTzd2TRN
+yoJnEn2puANMFfI0+KJZPzSOsOK+GdpUNuOAMKfm1WxS4S7h8XAXp0iwBWIWBhqm
+8adOk97iL/ghjO6JkYsHLzGg5SVeto+ETWc9BThNCvYM6RAM1GKes6Q5Pm7rCNUf
+3JuaSP2Ft9pOSfR8LZ7uAYs1AKkFq2wEHcJwC+JymgnVW+tTk9j62m9GllMrJZrK
+qLWWRNetWpWRk27Wl28dPPLHMdo7zfiUS5huN/8CDUjEGH6zPpE4R9sCAQI=
+-----END DH PARAMETERS-----
