diff -urN work/curl-7.74.0/Makefile work/curl-7.74.0/Makefile
--- work/curl-7.74.0/Makefile	2018-10-23 23:00:46.662292000 +0200
+++ work/curl-7.74.0/Makefile	2018-10-23 23:01:08.566300143 +0200
@@ -0,0 +1,59 @@
+include ../Rules.mak
+
+LIBSNAME		:= libcurl.so
+LIBS			:= $(LIBSNAME).$(RCCURLVER)
+BINS			:= curl
+
+all: config
+	$(MAKE) -C $(CURL)
+	$(STRIP) $(STRIPFLAGS) $(CURL)/lib/.libs/*.a
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(CURL) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) --prefix=$(USRDIR) --sysconfdir=$(TMPETC) --libdir=$(LDLIB) --sharedstatedir=$(LDLIB) --bindir=$(USRSBIN) --sbindir=$(USRSBIN) --libexecdir=$(USRSBIN) --localstatedir=$(VARLIB) --with-zlib=$(WORK_DIR)/$(ZLIB)/$(ZLIB) --with-ssl=$(WORK_DIR)/$(OPENSSL)/$(OPENSSL) --enable-shared=yes --enable-static=yes --enable-ipv6 --enable-tls-srp --disable-debug --disable-curldebug --disable-manual --disable-verbose --disable-socketpair --disable-get-easy-options --without-cyassl --without-brotli --without-mbedtls --without-wolfssl --without-zsh-functions-dir --without-bearssl CFLAGS="$(CFLAGS) -Os -s" CPPFLAGS="-I$(WORK_DIR)/$(ZLIB)/$(ZLIB) -I$(WORK_DIR)/$(OPENSSL)/$(OPENSSL)/include" LDFLAGS="-L$(WORK_DIR)/$(ZLIB)/$(ZLIB) -L$(WORK_DIR)/$(OPENSSL)/$(OPENSSL)" LIBS="-lz -lssl -lcrypto"
+
+install: control
+	$(STRIP) $(CURL)/lib/.libs/$(LIBS)
+	$(SSTRIP) $(CURL)/src/.libs/$(BINS)
+	install -m 0755 -d $(TMPINSTDIR)
+	install -m 0755 $(CURL)/lib/.libs/$(LIBS) $(CURL)/src/.libs/$(BINS) $(TMPINSTDIR)
+
+clean:
+	$(MAKE) -C $(CURL) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(CURL) distclean
+
+ifndef BUILD
+control:
+else
+PREINST			:= $(CTRL_DIR)/preinst
+POSTINST		:= $(CTRL_DIR)/postinst
+
+control: preinst postinst
+
+preinst:
+	@echo "#!/bin/sh" > $(PREINST)
+	@echo "echo \"Removing old $(BINS) lib ver & symlinks...\"" >> $(PREINST)
+	@echo "rm -f $(LDLIB)/$(LIBSNAME)*" >> $(PREINST)
+	@echo "sync" >> $(PREINST)
+	@echo "exit 0" >> $(PREINST)
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "[ ! -f $(TMPDIR)/$(BINS) ] && exit 0" >> $(POSTINST)
+	@echo "echo \"Updating $(BINS) lib & bins ...\"" >> $(POSTINST)
+	@echo "mv -f $(TMPDIR)/$(LIBS) $(LDLIB)/" >> $(POSTINST)
+	@echo "mv -f $(TMPDIR)/$(BINS) $(USRSBIN)/" >> $(POSTINST)
+	@echo "ln -sf $(LIBS) $(LDLIB)/$(LIBSNAME)" >> $(POSTINST)
+	@echo "ln -sf $(LIBS) $(LDLIB)/$(LIBSNAME).$(RCCURLMAINVER)" >> $(POSTINST)
+	@echo "sync" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+endif
+
