diff -urN work/e2fsprogs-1.45.4/rc_e2fsck work/e2fsprogs-1.45.4/rc_e2fsck
--- work/e2fsprogs-1.45.4/rc_e2fsck	2018-10-23 23:00:46.662292000 +0200
+++ work/e2fsprogs-1.45.4/rc_e2fsck	2018-10-23 23:01:08.566300143 +0200
@@ -0,0 +1,90 @@
+#!/bin/sh
+#
+# ancistrus
+#
+# Netgear's D7000 (V1) Nighthawk Router Experience Distributed Project
+#
+# https://github.com/negan07/ancistrus
+#
+# License: GPLv2
+#
+#
+# Rc script for e2fsck.
+#
+# Usage: $0 < start restart stop > <device_part>
+#
+# <device_part> is the partition device to check (mandatory)
+#
+
+stop() {
+    # Kill all pids
+    for PID in `pidof e2fsck`
+    do
+    [ -e /proc/${PID}/stat ] && while kill -9 $PID >/dev/null 2>&1; do :; done
+    done
+}
+
+start() {
+    # Checking if the service is already running
+    [ ! -z "`pidof e2fsck`" ] && echo "e2fsck already running..." && exit 127
+
+    # Checking for device partition param
+    local DEVPART=$1
+    [ -z "$DEVPART" ] && echo "Invalid device partition" && exit 128
+
+    # Retrieving partition mount point if any
+    local MOUNTPOINT=`cat /proc/mounts | grep $DEVPART | awk '{print $2}'`
+
+    # Unmounting device partition
+    umount $DEVPART
+
+    # Preliminary fs operations
+    cp -f /proc/mounts ${CONFDIR}/mtab
+
+    # Starting 'e2fsck'
+    $E2FSCK -p $DEVPART 2>&1
+
+    # Sanity check
+    local E2FSCK_RC=$?
+    echo
+    case $E2FSCK_RC in
+    0|1)
+    echo "$0: Success on device partition $DEVPART"
+    ;;
+    2)
+    echo "$0: Rebooting"
+    sleep 5
+    reboot
+    ;;
+    4)
+    echo "$0: Error ${E2FSCK_RC} starting ${E2FSCK} on device partition $DEVPART"
+    ;;
+    *)
+    echo "$0: Error ${E2FSCK_RC} starting ${E2FSCK} on device partition $DEVPART . Check not complete."
+    ;;
+    esac
+
+    # Remounting & exiting
+    mount $DEVPART $MOUNTPOINT >/dev/null 2>&1
+    exit $E2FSCK_RC
+}
+
+# Dirs
+CONFDIR=/etc
+BINDIR=/usr/sbin
+
+# Names
+E2FSCK=${BINDIR}/e2fsck
+
+case $1 in
+start|restart)
+stop
+start $2
+;;
+stop)
+stop
+;;
+*)
+echo "Usage: $0 < start restart stop > <device_part>"
+;;
+esac
