diff -urN work/e2fsprogs-1.45.4/Makefile work/e2fsprogs-1.45.4/Makefile
--- work/e2fsprogs-1.45.4/Makefile	2018-10-23 23:00:46.662292000 +0200
+++ work/e2fsprogs-1.45.4/Makefile	2018-10-23 23:01:08.566300143 +0200
@@ -0,0 +1,57 @@
+include ../Rules.mak
+
+BINS			:= $(E2FSPROGS)/e2fsck/e2fsck $(E2FSPROGS)/misc/blkid
+STARTUP			:= rc_e2fsck
+WEB			:= ../www/e2fsprogs.htm
+E2FSCK_CONF		:= e2fsck.conf
+E2FSCK_CONF_INST_DIR	:= $(ETCDIR)
+
+all: config
+	$(MAKE) -C $(E2FSPROGS)
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(E2FSPROGS) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) CFLAGS="$(CFLAGS) -Os -s" --prefix=$(USRDIR) --disable-elf-shlibs --disable-backtrace --disable-debugfs --disable-imager --disable-resizer --disable-defrag --disable-e2initrd-helper --enable-fsck --disable-uuidd --disable-mmp --disable-tdb --disable-tls --disable-nls --disable-rpath --disable-bmap-stats --disable-fuse2fs --with-libiconv-prefix=$(WORK_DIR)/$(LIBICONV)
+
+e2fsck.conf:
+	@echo "[options]" > $(E2FSCK_CONF)
+	@echo "broken_system_clock = true" >> $(E2FSCK_CONF)
+
+install: control e2fsck.conf
+	$(SSTRIP) $(BINS)
+	for D in $(E2FSCK_CONF_INST_DIR) $(BINDIR) $(APPDIR) $(WWWDIR); do install -m 0755 -d $$D; done
+	install -m 0644 $(E2FSCK_CONF) $(E2FSCK_CONF_INST_DIR)
+	install -m 0755 $(BINS) $(BINDIR)
+	install -m 0755 $(STARTUP) $(APPDIR)
+	install -m 0664 $(WEB) $(WWWDIR)
+
+clean:
+	$(MAKE) -C $(E2FSPROGS) clean
+
+distclean:
+	@rm -f $(E2FSCK_CONF) h_config
+	$(MAKE) -C $(E2FSPROGS) distclean
+
+ifndef BUILD
+control:
+else
+CONFFILES		:= $(CTRL_DIR)/conffiles
+PRERM			:= $(CTRL_DIR)/prerm
+
+control: conffiles prerm
+
+conffiles:
+	@echo "$(USRETC)/$(E2FSCK_CONF)" > $(CONFFILES)
+
+prerm:
+	@echo "#!/bin/sh" > $(PRERM)
+	@echo "$(USRSBINRCAPP)/$(STARTUP) stop" >> $(PRERM)
+	@echo "rm -f $(TMPETC)/$(E2FSCK_CONF) $(TMPETC)/mtab" >> $(PRERM)
+	@echo "exit 0" >> $(PRERM)
+endif
+
