--- work/libexif-0.6.23/libexif-0.6.23/libexif/exif-tag.h
+++ work/libexif-0.6.23/libexif-0.6.23/libexif/exif-tag.h
@@ -32,8 +32,11 @@ extern "C" {
 
 /*! EXIF tags */
 typedef enum {
+	EXIF_TAG_GPS_VERSION_ID			= 0x0000,
 	EXIF_TAG_INTEROPERABILITY_INDEX		= 0x0001,
 	EXIF_TAG_INTEROPERABILITY_VERSION	= 0x0002,
+	EXIF_TAG_GPS_ALTITUDE_REF		= 0x0005,
+	EXIF_TAG_GPS_TIME_STAMP			= 0x0007,
 	EXIF_TAG_NEW_SUBFILE_TYPE		= 0x00fe,
 	EXIF_TAG_IMAGE_WIDTH 			= 0x0100,
 	EXIF_TAG_IMAGE_LENGTH 			= 0x0101,
