diff -urN work/netperf-2.7.0/netperf-2.7.0/configure work/netperf-2.7.0/netperf-2.7.0/configure
--- work/netperf-2.7.0/netperf-2.7.0/configure	2015-07-20 19:30:24.000000000 +0200
+++ work/netperf-2.7.0/netperf-2.7.0/configure	2017-03-18 15:07:04.363079000 +0100
@@ -5833,7 +5833,7 @@
 _ACEOF
 
 rm -f conftest*
-
+if [ 1 -eq 0 ]; then
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether setpgrp takes no argument" >&5
 $as_echo_n "checking whether setpgrp takes no argument... " >&6; }
 if ${ac_cv_func_setpgrp_void+:} false; then :
@@ -5873,7 +5873,7 @@
 $as_echo "#define SETPGRP_VOID 1" >>confdefs.h
 
 fi
-
+fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking return type of signal handlers" >&5
 $as_echo_n "checking return type of signal handlers... " >&6; }
 if ${ac_cv_type_signal+:} false; then :
