diff -urN work/netperf-2.7.0/Makefile work/netperf-2.7.0/Makefile
--- work/netperf-2.7.0/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ work/netperf-2.7.0/Makefile	2017-03-23 20:35:45.893669092 +0100
@@ -0,0 +1,26 @@
+include ../Rules.mak
+
+all: config
+	$(MAKE) -C $(NETPERF)
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(NETPERF) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) --target=$(TARGET_NAME) CFLAGS="$(CFLAGS) -fcommon -Os" --disable-sctp --enable-omni
+
+install:
+	$(SSTRIP) src/netperf
+	install -m 0755 -d $(BINDIR)
+	install -m 0755 src/netperf $(BINDIR)
+
+clean:
+	$(MAKE) -C $(NETPERF) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(NETPERF) distclean
+
