diff -urN work/openssh-8.1p1/openssh-8.1p1/Makefile.in work/openssh-8.1p1/openssh-8.1p1/Makefile.in
--- work/openssh-8.1p1/openssh-8.1p1/Makefile.in	1970-01-01 01:00:00.000000000 +0100
+++ work/openssh-8.1p1/openssh-8.1p1/Makefile.in	2018-01-05 12:00:11.000000000 +0100
@@ -60,7 +60,7 @@
 MANFMT=@MANFMT@
 MKDIR_P=@MKDIR_P@
 
-TARGETS=ssh$(EXEEXT) sshd$(EXEEXT) ssh-add$(EXEEXT) ssh-keygen$(EXEEXT) ssh-keyscan${EXEEXT} ssh-keysign${EXEEXT} ssh-pkcs11-helper$(EXEEXT) ssh-agent$(EXEEXT) scp$(EXEEXT) sftp-server$(EXEEXT) sftp$(EXEEXT)
+TARGETS=sshd$(EXEEXT) ssh-keygen$(EXEEXT) scp$(EXEEXT)
 
 XMSS_OBJS=\
 	ssh-xmss.o \
