diff -urN work/openssh-8.1p1/openssh-8.1p1/ssh_config work/openssh-8.1p1/openssh-8.1p1/ssh_config
--- work/openssh-8.1p1/openssh-8.1p1/ssh_config	2018-04-02 07:38:28.000000000 +0200
+++ work/openssh-8.1p1/openssh-8.1p1/ssh_config	2018-05-08 18:46:37.213533875 +0200
@@ -43,3 +43,6 @@
 #   VisualHostKey no
 #   ProxyCommand ssh -q -W %h:%p gateway.example.com
 #   RekeyLimit 1G 1h
+
+# enable DSCP QoS values (per RFC-4594)
+#IPQoS AF21 AF11
diff -urN work/openssh-8.1p1/openssh-8.1p1/sshd_config work/openssh-8.1p1/openssh-8.1p1/sshd_config
--- work/openssh-8.1p1/openssh-8.1p1/sshd_config	2018-04-02 07:38:28.000000000 +0200
+++ work/openssh-8.1p1/openssh-8.1p1/sshd_config	2018-05-08 18:48:25.164852441 +0200
@@ -105,6 +105,9 @@
 # no default banner path
 #Banner none
 
+# enable DSCP QoS values (per RFC-4594)
+#IPQoS AF21 AF11
+
 # override default of no subsystems
 Subsystem	sftp	/usr/libexec/sftp-server
 
