diff -urN D7000_V1.0.1.54_WW_src/Source/env.mak D7000_V1.0.1.54_WW_src/Source/env.mak
--- D7000_V1.0.1.54_WW_src/Source/env.mak	2016-10-11 17:57:47.000000000 +0200
+++ D7000_V1.0.1.54_WW_src/Source/env.mak	2017-01-13 14:24:56.249727068 +0100
@@ -18,7 +18,7 @@
 export IMAGE_DIR=$(ROOT)/image
 export TARGET_DIR=$(ROOT)/target
 export SC_LIBS_DIR=$(APPS_DIR)/sc_libs
-export THIRD_PARTY_DIR=$(ROOT)/third_party
+export THIRD_PARTY_DIR=$(ROOT)/../../work
 export LOADER_DIR = $(TARGET_HOME)/cfe
 export CFERAM_DIR = $(TARGET_HOME)/cferam
 export DPF_DIR = $(TARGET_HOME)/dpf
@@ -29,6 +29,7 @@
 export HOSTTOOLS_DIR = $(TARGET_HOME)/../Kernel/bcm963xx/hostTools
 export MAKE_BIN_DIR = $(TOOLS_DIR)/make_bin
 export FUNC = 00
+export ATMDIR = linux-atm-2.4.0
 
 export ROOTFS = $(FS_DIR)
 
@@ -45,12 +46,18 @@
 export AS=$(CROSS)as
 export RANLIB=$(CROSS)ranlib
 export STRIP=$(CROSS)strip
+export SSTRIP=$(CROSS)sstrip
 CROSS_DIR = /opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/usr/bin/
 CROSS_DIR_LIB = /opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/lib/
 CROSS_DIR_USR_LIB = /opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/usr/arm-unknown-linux-uclibcgnueabi/lib/
+
 TOOLS_DIR = $(ROOT)/tools/
 export PATH := $(CROSS_DIR):$(TOOLS_DIR):$(PATH)
 
-#export CFLAGS += -Wno-strict-aliasing
-
+# flags settings
+export STRIPFLAGS := -x -R .note -R .comment -R .version --strip-unneeded
+
+export UCLIBC_EXTRA_CFLAGS := -mcpu=cortex-a9 -mtune=cortex-a9 -march=armv7-a -mabi=aapcs-linux -mfpu=neon-fp16 -mfloat-abi=soft
+
+export CFLAGS := -marm -march=armv7-a -mcpu=cortex-a9 -mtune=cortex-a9 -mfpu=neon-fp16 -pipe -funit-at-a-time -fomit-frame-pointer -ffixed-r8 -fno-common -mno-thumb-interwork -mabi=aapcs-linux -mfloat-abi=soft
 
