diff -urN D7000_V1.0.1.82_WW_src/Makefile diff -urN D7000_V1.0.1.82_WW_src/Makefile
--- D7000_V1.0.1.82_WW_src/Makefile	2016-10-11 17:57:55.000000000 +0200
+++ D7000_V1.0.1.82_WW_src/Makefile	2017-01-12 20:27:44.184650622 +0100
@@ -1,9 +1,12 @@
 #
 #	<MODULE> serial product makefile 
-#	Usage:  make kernel source bootcode
+#	Usage:  make kernel/source
 #
 #NOTE: do "make all_clean" if need change ANNEX after compiled
 
+ARCH_NAME="arm"
+PROFILE_NAME="963138GW"
+
 ANNEX=A
 #ANNEX=B
 
@@ -22,24 +25,30 @@
 prepare: creat_rules Kernel/bcm963xx 
 
 kernel:
-	@echo "PROFILE=963138GW" > Kernel/bcm963xx/PROFILE.make
-	@echo "963138GW" > Kernel/bcm963xx/.last_profile
-	@cd Kernel/bcm963xx/; make sc_kernel;
+	@echo -e "\\e[35mKERNEL: make \\e[36mvmlinux.lz \\e[0m"
+	@echo "PROFILE=$(PROFILE_NAME)" > Kernel/bcm963xx/PROFILE.make
+	@echo $(PROFILE_NAME) > Kernel/bcm963xx/.last_profile
+	@if [ ! -f Kernel/bcm963xx/.compiled ]; then \
+	cd Kernel/bcm963xx/; make sc_kernel; touch .compiled; \
+	fi
+
+kernel_menuconfig:
+	@cd Kernel/bcm963xx/kernel/linux-3.4rt; make ARCH=$(ARCH_NAME) menuconfig ;
+
+kernel_clean:
+	@echo -e "\\e[35mKERNEL: make \\e[36mclean \\e[0m"
+	@echo "PROFILE=$(PROFILE_NAME)" > Kernel/bcm963xx/PROFILE.make
+	@cd Kernel/bcm963xx/; make sc_clean ; rm -f .compiled;
 
 source: prepare
-#	if [ -h /opt/toolchains/uclibc-crosstools ]; then\
-#		rm -rf /opt/toolchains/uclibc-crosstools;\
-#	fi
-#	ln -sf uclibc-crosstools-gcc-4.4.2-1/usr/ /opt/toolchains/uclibc-crosstools;
 	make -C Source -s
 
+source_orig: prepare
+	make ORIG=1 -C Source orig -s
+
 source_clean:
 	make -C Source clean_all
 
-kernel_clean:
-	@echo "PROFILE=963138GW" > Kernel/bcm963xx/PROFILE.make
-	@cd Kernel/bcm963xx/; make sc_clean ;
-
 all_clean: kernel_clean source_clean
 
 creat_rules:
