--- D7000_V1.0.1.84_WW_src/Makefile
+++ D7000_V1.0.1.84_WW_src/Makefile
@@ -1,44 +1,48 @@
 #
 #	<MODULE> serial product makefile 
-#	Usage:  make kernel source bootcode
+#	Usage:  make kernel/source/source_orig
 #
-#NOTE: do "make all_clean" if need change ANNEX after compiled
 
+ARCH_NAME="arm"
+PROFILE_NAME="963138GW"
 ANNEX=A
-#ANNEX=B
-
-ifeq "$(ANNEX)" "B"
-PROJECT_NAME=D7000B
-else
 PROJECT_NAME=D7000
-endif
 
-CUR_PATH = $(shell pwd)
+CUR_PATH=$(shell pwd)
 RULES=$(CUR_PATH)/Source/Rules.mak
+KDIR=Kernel/bcm963xx
 
 all:
-	@echo 'make <kernel/source>';
-
-prepare: creat_rules Kernel/bcm963xx 
+	@echo 'make <kernel/source/source_orig>';
 
 kernel:
-	@echo "PROFILE=963138GW" > Kernel/bcm963xx/PROFILE.make
-	@echo "963138GW" > Kernel/bcm963xx/.last_profile
-	@cd Kernel/bcm963xx/; make sc_kernel;
-
-source: prepare
-#	if [ -h /opt/toolchains/uclibc-crosstools ]; then\
-#		rm -rf /opt/toolchains/uclibc-crosstools;\
-#	fi
-#	ln -sf uclibc-crosstools-gcc-4.4.2-1/usr/ /opt/toolchains/uclibc-crosstools;
-	make -C Source -s
+	@echo -e "\\e[35mKERNEL: make \\e[36mvmlinux.lz \\e[0m"
+	@echo "PROFILE=$(PROFILE_NAME)" > $(KDIR)/PROFILE.make
+	@echo $(PROFILE_NAME) > $(KDIR)/.last_profile
+	@if [ ! -f $(KDIR)/.compiled ]; then \
+	$(MAKE) -C $(KDIR) sc_kernel -s || exit 1; \
+	touch $(KDIR)/.compiled; \
+	fi
 
-source_clean:
-	make -C Source clean_all
+kernel_menuconfig:
+	@$(MAKE) ARCH=$(ARCH_NAME) -C $(KDIR)/kernel/linux-3.4rt menuconfig
 
 kernel_clean:
-	@echo "PROFILE=963138GW" > Kernel/bcm963xx/PROFILE.make
-	@cd Kernel/bcm963xx/; make sc_clean ;
+	@echo -e "\\e[35mKERNEL: make \\e[36mclean \\e[0m"
+	@echo "PROFILE=$(PROFILE_NAME)" > $(KDIR)/PROFILE.make
+	@$(MAKE) -C $(KDIR) sc_clean
+	@rm -f $(KDIR)/.compiled
+
+source: creat_rules
+	@echo -e "\\e[35mSOURCE: make \\e[36msources \\e[0m"
+	@$(MAKE) -C Source -s
+
+source_orig: creat_rules
+	@$(MAKE) ORIG=1 -C Source orig -s
+
+source_clean:
+	@echo -e "\\e[35mSOURCE: make \\e[36mclean \\e[0m"
+	@$(MAKE) -C Source clean_all
 
 all_clean: kernel_clean source_clean
 
@@ -47,3 +51,4 @@ creat_rules:
 	@echo "PROJECT=$(PROJECT_NAME)" > $(RULES)
 	@echo "export ROOT=$(CUR_PATH)/Source/" >> $(RULES)
 	@cat Source/rules.src >> $(RULES)
+
