--- D7000_V1.0.1.44_WW_src/Source/Makefile	2017-02-28 22:43:03.899045459 +0100
+++ test/D7000_V1.0.1.44_WW_src/Source/Makefile	2017-03-06 17:51:49.918734936 +0100
@@ -64,11 +64,11 @@
 
 all: build_test
 
-build_test: pre_fs apps drivers prefs fs img
+build_test: pre_fs apps third-party drivers prefs fs img
 
 build_image: clean_target build_target pid build_dpf fs img bin
 
-clean_all: clean_uclibc clean_apps clean_target clean_img
+clean_all: clean_uclibc clean_apps clean_third-party clean_target clean_img
 
 clean: clean_target
 
@@ -87,11 +87,9 @@
 	cp -af $(CROSS_DIR_LIB)/libgcc_s.so target/lib
 	cp -af $(CROSS_DIR_USR_LIB)/libstdc++.so.6.0.16-gdb.py target/lib
 	cd target/lib; ln -sf libscnvram.so libconfig.so
-ifeq ($(ITUNES_SERVER),1)
 	@echo -e "\\e[35mSOURCE: make \\e[36mUCLIBC \\e[0m"
 	make -C uClibc-0.9.32/
 	cp -af uClibc-0.9.32/lib/ld-uClibc-0.9.32.so target/lib/
-endif
 
 prepare:
 	# tools
@@ -116,9 +114,8 @@
 third-party:
 	@echo -e "\\e[35mSOURCE: make \\e[36mTHIRD PARTY \\e[0m"
 	$(MAKE) -C $(THIRD_PARTY_DIR)
-	$(MAKE) -C $(THIRD_PARTY_DIR)	install
+	$(MAKE) -C $(THIRD_PARTY_DIR) install
 
-	
 ui:
 	@echo -e "\\e[35mSOURCE: make \\e[36mUI \\e[0m"	
 	$(MAKE) -C $(UI_DIR) -s	
@@ -206,7 +203,7 @@
 	echo "InternetConnectionStatus=Up" >> $(FS_DIR)/www.eng/currentsetting.htm
 
 lang_pid:
-    # Generate language pid
+# Generate language pid. 
 	$(MAKE_PID) $(HW_ID) $(ANNEX)$(GUI_VER) $(PRODUCT_ID) $(LANG_PID)
 	
 langpkg: lang_pid
@@ -276,14 +273,17 @@
 
 
 clean_uclibc:
-ifeq ($(ITUNES_SERVER),1)
 	@echo -e "\\e[35mUCLIBC: make \\e[36mclean \\e[0m"
 	-$(MAKE) -C uClibc-0.9.32 clean -s
-endif
 
 clean_apps:
+	@echo -e "\\e[35mAPPS: make \\e[36mclean \\e[0m"
 	-$(MAKE) -C $(APPS_DIR) clean -s
 
+clean_third-party:
+	@echo -e "\\e[35mTHIRD-PARTY: make \\e[36mclean \\e[0m"
+	-$(MAKE) -C $(THIRD_PARTY_DIR) clean -s
+
 clean_target:
 	@echo -e "\\e[35mTARGET: \\e[36mclean \\e[0m"
 	-rm -rf $(TARGET_HOME)/target
--- D7000_V1.0.1.44_WW_src/Source/env.mak	2017-01-15 00:13:11.118776024 +0100
+++ test/D7000_V1.0.1.44_WW_src/Source/env.mak	2017-03-06 13:27:34.879442022 +0100
@@ -18,7 +18,7 @@
 export IMAGE_DIR=$(ROOT)/image
 export TARGET_DIR=$(ROOT)/target
 export SC_LIBS_DIR=$(APPS_DIR)/sc_libs
-export THIRD_PARTY_DIR=$(ROOT)/third_party
+export THIRD_PARTY_DIR=$(ROOT)/../../work
 export LOADER_DIR = $(TARGET_HOME)/cfe
 export CFERAM_DIR = $(TARGET_HOME)/cferam
 export DPF_DIR = $(TARGET_HOME)/dpf
