diff -urN D7000_V1.0.1.64_WW_src/Source/image/D7000.par.orig D7000_V1.0.1.64_WW_src/Source/image/D7000.par.orig
--- D7000_V1.0.1.64_WW_src/Source/image/D7000.par.orig	2017-12-11 00:00:12.738980780 +0100
+++ D7000_V1.0.1.64_WW_src/Source/image/D7000.par.orig	2017-12-11 00:00:12.738980780 +0100
@@ -0,0 +1,16 @@
+:D7000_boot.bin                      &00000000
+:rootfs.img                             &00200000
+:langpkg_ENU.img                        &02600000
+:langpkg_CHS.img                        &02700000
+:langpkg_ITA.img                        &02800000
+:langpkg_DEU.img                        &02900000
+:langpkg_NLD.img                        &02a00000
+:langpkg_KOR.img                        &02b00000
+:langpkg_FRA.img                        &02c00000
+:langpkg_SVE.img                        &02d00000
+:langpkg_RUS.img                        &02e00000
+:langpkg_ESP.img                        &02f00000
+:langpkg_CSY.img                        &03000000
+:langpkg_TUR.img                        &03100000
+%D7000.pid		                &00003fff
+$size				        &10000
diff -urN D7000_V1.0.1.64_WW_src/Source/image/D7000.par.work D7000_V1.0.1.64_WW_src/Source/image/D7000.par.work
--- D7000_V1.0.1.64_WW_src/Source/image/D7000.par.work	2017-12-11 00:00:15.738980780 +0100
+++ D7000_V1.0.1.64_WW_src/Source/image/D7000.par.work	2017-12-11 00:00:15.738980780 +0100
@@ -0,0 +1,17 @@
+:D7000_boot.bin                      &00000000
+:rootfs.img                             &00200000
+:langpkg_ENU.img                        &02600000
+:langpkg_CHS.img                        &02700000
+:langpkg_ITA.img                        &02800000
+:langpkg_DEU.img                        &02900000
+:langpkg_NLD.img                        &02a00000
+:langpkg_KOR.img                        &02b00000
+:langpkg_FRA.img                        &02c00000
+:langpkg_SVE.img                        &02d00000
+:langpkg_RUS.img                        &02e00000
+:langpkg_ESP.img                        &02f00000
+:init.d.img                             &03000000
+:opkg.img                               &03100000
+:xxx.img                                &03200000
+%D7000.pid		                &00003fff
+$size				        &10000
