diff -urN work/mini_httpd-1.17beta1/mini_httpd.c work/mini_httpd-1.17beta1/mini_httpd.c
--- work/mini_httpd-1.17beta1/mini_httpd.c	2020-03-30 05:16:27.000000000 +0200
+++ work/mini_httpd-1.17beta1/mini_httpd.c	2021-01-28 18:22:10.168802254 +0100
@@ -60,8 +60,6 @@
 #include "misc.h"
 #include "nvram.h"
 #include "sname_def.h"
-//#define _SC_DEBUG_ENABLE_
-#include "sc_debug.h"
 #include "filenames.h"
 extern char *strcasestr(char *haystack, char *needle);
 
@@ -79,8 +77,14 @@
 #endif /* USE_SSL */
 //extern char* crypt( const char* key, const char* setting );
 
-#include "sc_sessionid.h"
+#include "sc_sessionid_upd.h"
+#ifndef SC_CFPRINTF
+#define SC_CFPRINTF SC_CFPRINTF_SESSIONID
+#endif
 
+#ifndef SET_COOKIE
+#define SET_COOKIE snprintf(buf, sizeof(buf), "Set-Cookie: %s=%s;SameSite=Lax;\015\012", SESSIONID_MAGICWORD, sessionid_new_sessionid);
+#endif
 
 //#if defined(AF_INET6) && defined(IN6_IS_ADDR_V4MAPPED)
 #ifdef HTTPD_IPV6_SUPPORT
@@ -3043,7 +3047,7 @@
 			if (sessionid_new_sessionid[0])
 			{
 				SC_CFPRINTF_SESSIONID("!!!update client session id:%s\n", sessionid_new_sessionid);
-				(void)snprintf(buf, sizeof(buf), "Set-Cookie: %s=%s\015\012", SESSIONID_MAGICWORD, sessionid_new_sessionid);
+				(void)SET_COOKIE
 				(void)my_write(buf, strlen(buf));
 			}
 		}		
@@ -3131,7 +3135,7 @@
 			if (sessionid_new_sessionid[0])
 			{
 				SC_CFPRINTF_SESSIONID("!!!update client session id:%s\n", sessionid_new_sessionid);
-				(void)snprintf(buf, sizeof(buf), "Set-Cookie: %s=%s\015\012", SESSIONID_MAGICWORD, sessionid_new_sessionid);
+				(void)SET_COOKIE
 				(void)my_write(buf, strlen(buf));
 			}
 		}
@@ -3351,35 +3355,14 @@
 static void
 auth_check( char* dirname )
     {
-#if 0
-    char authpath[10000];
-#else
-//    char *authpath;
-#endif
-#if 0 
-	struct stat sb;
-#endif
-	char authinfo[500];
+    char authinfo[500];
     char* authpass;
     static char line[10000];
     int l;
-///    FILE* fp;
-//    char* cryp;
-    char *authname
-//	char *name;
+    char *authname;
 
     SC_CFPRINTF("dirname is <%s>\n", dirname);
 
-    /* Construct auth filename. */
-#if 0
-    if ( dirname[strlen(dirname) - 1] == '/' )
-    (void) snprintf( authpath, sizeof(authpath), "%s%s", dirname, AUTH_FILE );
-    else
-    (void) snprintf( authpath, sizeof(authpath), "%s/%s", dirname, AUTH_FILE );
-#else
-//    authpath = AUTH_FILE;
-#endif
-
 #ifdef SUPPORT_SETUPWIZARD
 	if (for_setupwizard == 1)
 	{
@@ -3459,60 +3442,12 @@
     send_authenticate( dirname );
 	}
     *authpass++ = '\0';
-
-    /* No need to check ':' here, otherwise password with ':' will be fail  -- Joe */
-#if 0
-    /* If there are more fields, cut them off. */
-    colon = strchr( authpass, ':' );
-    if ( colon != (char*) 0 )
-    *colon = '\0';
-#endif
-#if 0	
-    /* Open the password file. */
-    fp = fopen( authpath, "r" );
-    if ( fp == (FILE*) 0 )
-    {
-    /* The file exists but we can't open it?  Disallow access. */
-#ifdef SYSLOG
-    syslog(
-        LOG_ERR, "%.80s auth file %.80s could not be opened - %m",
-        ntoa( &client_addr ), authpath );
-#endif
-    send_error( 403, "Forbidden", "", "File is protected." );
-    }
-
-    /* Read it. */
-    while ( fgets( line, sizeof(line), fp ) != (char*) 0 )
-#endif
 	{
-#if 0
-	/* Nuke newline. */
-    l = strlen( line );
-    if ( line[l - 1] == '\n' )
-        line[l - 1] = '\0';
-    /* Split into user and encrypted password. */
-    cryp = strchr( line, ':' );
-    if ( cryp == (char*) 0 )
-        continue;
-    *cryp++ = '\0';
-    name = line;
-#endif
 	/* Is this the right user? */
     if (strcmp(authinfo,nvram_safe_get("http_username")) == 0)
         {
             struct sysinfo info;
             sysinfo(&info);
-        /* Yes. */
-#if 0
-			(void) fclose( fp );
-#endif
-			/* So is the password right? */
-/*Ron*/
-#if 0
-        if ( strcmp( crypt( authpass, cryp ), cryp ) == 0 )
-#endif
-
-
     /* OK !*/
     if (strcmp(authpass,nvram_safe_get("http_password")) == 0){
 
@@ -3615,9 +3550,6 @@
     }
 
     /* Didn't find that user.  Access denied. */
-#if 0
-	(void) fclose( fp );
-#endif
 	syslog(LOG_INFO,"Login Failed - IP:%s",remote_ip);
     not_auth=AUTH_FAIL;
 	if ( !match( "**MSIE**", useragent ) ){
@@ -3865,7 +3797,7 @@
 		if (sessionid_new_sessionid[0])
 		{
 			SC_CFPRINTF_SESSIONID("!!!update client session id:%s\n", sessionid_new_sessionid);
-			buflen = snprintf(buf, sizeof(buf), "Set-Cookie: %s=%s\015\012", SESSIONID_MAGICWORD, sessionid_new_sessionid);
+			buflen = SET_COOKIE
 			add_to_response(buf, buflen);
 		}
 	}
