diff -urN work/mini_httpd-1.17beta1/mini_httpd.c work/mini_httpd-1.17beta1/mini_httpd.c
--- work/mini_httpd-1.17beta1/mini_httpd.c	2002-05-25 22:16:55.000000000 +0200
+++ work/mini_httpd-1.17beta1/mini_httpd.c	2018-11-19 11:18:18.000000000 +0100
@@ -3870,11 +3870,11 @@
     buflen = snprintf( buf, sizeof(buf), "Last-Modified: %s\r\n", timebuf );
     add_to_response( buf, buflen );
     }
-	snprintf(buf, sizeof(buf), "X-Frame-Options: SAMEORIGIN\015\012");
+    buflen = snprintf(buf, sizeof(buf), "X-Frame-Options: SAMEORIGIN\015\012");
     add_to_response( buf, buflen );
-	snprintf(buf, sizeof(buf), "X-XSS-Protection: 1;mode=block\015\012");
+    buflen = snprintf(buf, sizeof(buf), "X-XSS-Protection: 1;mode=block\015\012");
     add_to_response( buf, buflen );
-	snprintf(buf, sizeof(buf), "X-Content-Type-Options: nosniff\015\012");
+    buflen = snprintf(buf, sizeof(buf), "X-Content-Type-Options: nosniff\015\012");
     add_to_response( buf, buflen );
     buflen = snprintf( buf, sizeof(buf), "Connection: close\r\n\r\n" );
     add_to_response( buf, buflen );
