diff -urN D7000_V1.0.1.74_WW_src/Source/apps/iptables/Makefile D7000_V1.0.1.74_WW_src/Source/apps/iptables/Makefile
--- D7000_V1.0.1.74_WW_src/Source/apps/iptables/Makefile	2016-10-11 17:56:27.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/iptables/Makefile	2017-04-29 21:29:16.541952901 +0200
@@ -1,7 +1,7 @@
 include ../../Rules.mak
 
 APP = iptables-1.4.16.3
-CFLAGS += -s -Os -Wno-strict-aliasing
+#CFLAGS += -s -Os -Wno-strict-aliasing
 
 all: prepare configure build install
 
@@ -38,6 +38,7 @@
 			--with-pfxmods="$(PFX_EXT)" \
 			--with-pf4mods="$(PF4_EXT)" \
 			--with-pf6mods="$(PF6_EXT)" \
+	        CFLAGS="$(CFLAGS) -s -Os -Wno-strict-aliasing" \
 	        LDFLAGS=-L$(TARGET_DIR)/lib/; \
 	    fi
 
@@ -46,7 +47,7 @@
 
 install:
 	install -m 755 $(APP)/iptables/xtables-multi $(TARGET_DIR)/bin
-	$(STRIP) $(TARGET_DIR)/bin/xtables-multi
+	$(SSTRIP) $(TARGET_DIR)/bin/xtables-multi
 	ln -sf xtables-multi $(TARGET_DIR)/bin/iptables
 	ln -sf xtables-multi $(TARGET_DIR)/bin/ip6tables
 
