diff -urN D7000_V1.0.1.68_WW_src/Source/apps/sysstat/Makefile D7000_V1.0.1.68_WW_src/Source/apps/sysstat/Makefile
--- D7000_V1.0.1.68_WW_src/Source/apps/sysstat/Makefile	2016-10-11 17:57:00.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/sysstat/Makefile	2017-05-01 21:49:32.193798601 +0200
@@ -64,7 +64,7 @@
 SYSCONFIG_DIR = /etc/sysconfig
 
 # Compiler flags
-CFLAGS = -g -O2 -Wall -Wstrict-prototypes -pipe -O2
+CFLAGS +=  -Os -Wall -Wstrict-prototypes -pipe
 DFLAGS = 
 LFLAGS =  -s
 # DFLAGS may already contain SMP_RACE definition
@@ -533,13 +533,8 @@
 uninstall: uninstall_base
 endif
 
-#ifeq ($(INSTALL_CRON),y)
-#install: install_all
-#else
-#install: install_base
-#endif
-
 install: mpstat
+	$(SSTRIP) mpstat
 	cp -a mpstat $(ROOT)/target/usr/sbin
 
 ifdef REQUIRE_NLS
