diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/brctl/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/brctl/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/brctl/Makefile	2016-10-11 17:57:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/brctl/Makefile	2017-05-03 23:51:39.710415474 +0200
@@ -8,11 +8,7 @@
 
 # brcm begin
 KERNEL_HEADERS=-I$(KERNEL_DIR)/include
-
-#CC=gcc
-CFLAGS = -Wall -Os
-#CFLAGS= -Wall -g -O2
-#CFLAGS= -Wall -s
+CFLAGS += -Wall -Os
 ifeq ($(strip $(BUILD_BRCTL)),static)
 CFLAGS += -DBUILD_STATIC
 endif
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/libbridge/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/libbridge/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/libbridge/Makefile	2016-10-11 17:57:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/libbridge/Makefile	2017-05-01 21:49:32.093801689 +0200
@@ -8,13 +8,7 @@
 
 # brcm begin
 KERNEL_HEADERS=-I$(KERNEL_DIR)/include
-
-#AR=ar
-#RANLIB=ranlib
-CFLAGS = -Wall -Os
-#CC=gcc
-#CFLAGS = -Wall -g $(KERNEL_HEADERS)
-#CFLAGS = -Wall -Os $(KERNEL_HEADERS)
+CFLAGS += -Wall -Os
 # brcm end
 
 prefix=/usr/local
@@ -44,7 +38,7 @@
 
 libbridge.a:	$(libbridge_OBJECTS)
 	$(AR) rcs $@ $(libbridge_OBJECTS)
-	#$(RANLIB) $@
+	$(STRIP) $(STRIPFLAGS) libbridge.a
 
 %.o: %.c libbridge.h libbridge_private.h
 	$(CC) $(CFLAGS) $(INCLUDE) -c $<
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/Makefile	2016-10-11 17:57:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bridge-utils/Makefile	2017-04-29 19:04:12.411778839 +0200
@@ -35,7 +35,7 @@
 	rm -f doc/Makefile
 
 install: brctl/brctl
-	$(CROSS)strip $<
+	$(SSTRIP) $<
 	cp $< ../../target/usr/sbin
 	#for x in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$x install; done
 # brcm begin
