diff -urN D7000_V1.0.1.82_WW_src/Source/apps/busybox/Makefile D7000_V1.0.1.82_WW_src/Source/apps/busybox/Makefile
--- D7000_V1.0.1.82_WW_src/Source/apps/busybox/Makefile	2016-10-11 17:57:13.000000000 +0200
+++ D7000_V1.0.1.82_WW_src/Source/apps/busybox/Makefile	2017-05-02 14:04:44.712152528 +0200
@@ -4,7 +4,7 @@
 	cd ./busybox-1.17.2 && cp config.sc .config && cd ..
 	cd ./busybox-1.17.2 && touch .config 
 	cd ./busybox-1.17.2 && touch .config.old
-	cd ./busybox-1.17.2 && make
+	cd ./busybox-1.17.2 && make EXTRA_CFLAGS="$(CFLAGS)"
 
 install:
 	cd ./busybox-1.17.2 && make install
