diff -urN work/opkg-utils-0.3.4/opkg.py work/opkg-utils-0.3.4/opkg.py
--- work/opkg-utils-0.3.4/opkg.py	2016-10-07 18:29:53.000000000 +0200
+++ work/opkg-utils-0.3.4/opkg.py	2017-03-30 13:45:35.456313977 +0200
@@ -479,8 +479,7 @@
         if self.description:
             printable_description = textwrap.dedent(self.description).strip()
             summary = printable_description.split('\n', 1)[0]
-            printable_description = printable_description.split('\n', 1)[-1].strip()
-            out = out + "Description: %s\n%s\n" % (summary, textwrap.fill(printable_description, width=74, initial_indent=' ', subsequent_indent=' '))
+            out = out + "Description: %s\n" % (summary)
         if self.oe: out = out + "OE: %s\n" % (self.oe)
         if self.homepage: out = out + "HomePage: %s\n" % (self.homepage)
         if self.license: out = out + "License: %s\n" % (self.license)
