diff -urN work/opkg-utils-0.3.4/opkg-build work/opkg-utils-0.3.4/opkg-build
--- work/opkg-utils-0.3.4/opkg-build	2016-10-07 18:29:53.000000000 +0200
+++ work/opkg-utils-0.3.4/opkg-build	2017-04-04 14:47:25.570386375 +0200
@@ -208,15 +208,22 @@
     esac
 }
 
+toolchain_test() {
+	if [ ! -f $1 ] || [ ! -x $1 ]; then
+	echo "*** Error: unable to find ar path: $1" >&2
+	exit 1
+	fi
+}
+
 : <<=cut
 =head1 SYNOPSIS
 
-B<opkg-build> [B<-c>] [B<-C>] [B<-Z> I<compressor>] [B<-O>] [B<-o> I<owner>] [B<-g> I<group>] I<pkg_directory> [I<destination_directory>]
+B<opkg-build> [B<-c>] [B<-C>] [B<-Z> I<compressor>] [B<-T> I<toolchain_prefix_path>] [B<-O>] [B<-o> I<owner>] [B<-g> I<group>] I<pkg_directory> [I<destination_directory>]
 
 =cut
 
-usage="Usage: $0 [-c] [-C] [-Z compressor] [-O] [-o owner] [-g group] <pkg_directory> [<destination_directory>]"
-while getopts "cCg:ho:vOZ:" opt; do
+usage="Usage: $0 [-c] [-C] [-Z compressor] [-T toolchain_prefix_path] [-O] [-o owner] [-g group] <pkg_directory> [<destination_directory>]"
+while getopts "cCg:ho:vOZ:T:" opt; do
     case $opt in
 	o ) owner=$OPTARG
 	    ogargs="--owner=$owner"
@@ -232,6 +239,8 @@
 	    ;;
 	Z ) compressor=$OPTARG
 	    ;;
+	T ) [ "$outer" = "ar" ] && outer=${OPTARG}ar && toolchain_test $outer
+	    ;;
 	v ) echo $version
 	    exit 0
 	    ;;
@@ -305,11 +314,15 @@
 fi
 
 rm -f $pkg_file
-if [ "$outer" = "ar" ] ; then
-  ( cd $tmp_dir && ar -crf $pkg_file ./debian-binary ./control.tar.gz ./data.tar.$cext )
-else
-  ( cd $tmp_dir && tar -cz $tarformat -f $pkg_file ./debian-binary ./control.tar.gz ./data.tar.$cext )
-fi
+case $outer in
+tar)
+opts="-cz ${tarformat} -f"
+;;
+*)
+opts="-crf"
+;;
+esac
+( cd $tmp_dir && $outer $opts $pkg_file ./debian-binary ./control.tar.gz ./data.tar.$cext )
 
 rm $tmp_dir/debian-binary $tmp_dir/data.tar.$cext $tmp_dir/control.tar.gz
 rmdir $tmp_dir
@@ -346,6 +359,10 @@
 
 Specify which compression type to use when building a package. Allowed values are B<gzip>, B<bzip2> and B<xz> (default is B<gzip>).
 
+=item B<-T> I<toolchain_prefix_path>
+
+Specify a path toolchain prefix for ar, relative or absolute. Typical values are B</opt/toolchain/usr/bin/arm-linux->, B<mips-linux-uclibc-> and so on (default is B<no toolchain prefix>). Note that this option is always overrided by B<-c> one (generate a binary in tar format).
+
 =item B<-O>
 
 Use B<.opk> extension. By default, B<.ipk> is used.
