diff -urN work/libvorbis-1.3.7/Makefile work/libvorbis-1.3.7/Makefile
--- work/libvorbis-1.3.7/Makefile	2018-10-23 23:00:46.662292000 +0200
+++ work/libvorbis-1.3.7/Makefile	2018-10-23 23:01:08.566300143 +0200
@@ -0,0 +1,49 @@
+include ../Rules.mak
+
+LIBSNAME		:= libvorbis.so
+LIBSMAINVER		:= 0
+LIBSVER			:= $(LIBSMAINVER).4.9
+LIBSDIRNAME		:= $(VORBIS)/lib/.libs
+LIBS			:= $(LIBSNAME).$(LIBSVER)
+
+all: config
+	$(MAKE) -C $(VORBIS)
+	$(STRIP) $(STRIPFLAGS) $(LIBSDIRNAME)/*.a
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(VORBIS) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) CFLAGS="$(CFLAGS) -Os -s" CPPFLAGS="-I$(WORK_DIR)/$(OGG)/$(OGG)/include" LDFLAGS="-L$(WORK_DIR)/$(OGG)/$(OGG)/src/.libs" LIBS="-logg" --enable-static --disable-shared --disable-dependency-tracking --disable-docs --disable-examples --disable-oggtest
+
+install: control
+	#$(STRIP) $(LIBSDIRNAME)/$(LIBS)
+	#for D in $(LIBDIR); do install -m 0755 -d $$D; done
+	#install -m 0755 $(LIBSDIRNAME)/$(LIBS) $(LIBDIR)
+	#for L in $(LIBSNAME) $(LIBSNAME).$(LIBSMAINVER); do ln -sf $(LIBS) $(LIBDIR)/$$L; done
+
+clean:
+	$(MAKE) -C $(VORBIS) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(VORBIS) distclean
+
+ifndef BUILD
+control:
+else
+PREINST		:= $(CTRL_DIR)/preinst
+
+control: preinst
+
+preinst:
+	@echo "#!/bin/sh" > $(PREINST)
+	@echo "echo \"Removing old lib ver & symlinks...\"" >> $(PREINST)
+	@echo "rm -f $(LDLIB)/$(LIBSNAME)*" >> $(PREINST)
+	@echo "sync" >> $(PREINST)
+	@echo "exit 0" >> $(PREINST)
+endif
+
