diff -urN D7000_V1.0.1.50_WW_src/Source/apps/bwping-1.1/Makefile D7000_V1.0.1.50_WW_src/Source/apps/bwping-1.1/Makefile
--- D7000_V1.0.1.50_WW_src/Source/apps/bwping-1.1/Makefile	2016-10-11 17:56:08.000000000 +0200
+++ D7000_V1.0.1.50_WW_src/Source/apps/bwping-1.1/Makefile	2017-05-01 21:51:00.487113477 +0200
@@ -1,7 +1,7 @@
 include ../../Rules.mak
 CC=$(CROSS)gcc
 #remove -Os: fix dereferencing pointer 'oui' does break strict-aliasing rules
-CFLAGS=-Wall -Werror
+CFLAGS +=  -Wall -Werror
 
 all: bwping_ttl
 
@@ -13,4 +13,5 @@
 	-rm -f ../../target/usr/sbin/bwping_ttl 
 
 install:
+	$(SSTRIP) bwping_ttl
 	cp -a bwping_ttl ../../target/usr/sbin
