diff -urN work/opkg-0.4.5/opkg-0.4.5/libopkg/opkg_download_curl.c work/opkg-0.4.5/opkg-0.4.5/libopkg/opkg_download_curl.c
--- work/opkg-0.4.5/opkg-0.4.5/libopkg/opkg_download_curl.c	1970-01-01 01:00:00.000000000 +0100
+++ work/opkg-0.4.5/opkg-0.4.5/libopkg/opkg_download_curl.c	2017-03-23 20:35:42.745736890 +0100
@@ -234,6 +234,7 @@
     double src_size = -1;
     int ret = 1;
 
+    curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);
     curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, &dummy_write);
     curl_easy_setopt(curl, CURLOPT_HEADERFUNCTION, &header_write);
     curl_easy_setopt(curl, CURLOPT_WRITEHEADER, &etag);
