diff -urN work/smartmontools-6.5/Makefile work/smartmontools-6.5/Makefile
--- work/smartmontools-6.5/Makefile 1970-01-01 01:00:00.000000000 +0100
+++ work/smartmontools-6.5/Makefile 2017-03-23 20:35:45.893669092 +0100
@@ -0,0 +1,68 @@
+include ../Rules.mak
+
+SMARTD_CONF_PATH	:= $(SMONTOOLS)/examplescripts
+SMARTD_CONF_NAME	:= smartd.conf
+SMARTD_CONF 		:= $(SMARTD_CONF_PATH)/$(SMARTD_CONF_NAME)
+
+all: config
+	$(MAKE) -C $(SMONTOOLS)
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(SMONTOOLS) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) CFLAGS="$(CFLAGS) -Os -s -W -Wall" CXXFLAGS="$(CFLAGS) -Os -s -W -Wall" --sbindir=$(USRSBIN) --sysconfdir=$(TMPETC) --with-smartdplugindir=no --with-systemdsystemunitdir=no --with-systemdenvfile=no --with-initscriptdir=no --without-drivedbdir --without-nvme-devicescan --with-drivedbdir=$(TMPETC) --with-smartdscriptdir=$(USRSBINSCR)
+
+install: sm_conf control
+	$(SSTRIP) $(SMONTOOLS)/smartctl $(SMONTOOLS)/smartd
+#	install -m 0644 -d $(ETCDIR)
+	install -m 0755 -d $(BINDIR)
+#	install -m 0644 $(SMARTD_CONF) $(ETCDIR)
+	install -m 0755 $(SMONTOOLS)/smartctl $(BINDIR)
+#	install -m 0755 $(SMONTOOLS)/smartd $(BINDIR)
+#	install -m 0755 $(SMONTOOLS)/smartd_warning.sh $(SCRDIR)
+#	install -m 0755 $(SMONTOOLS)/update-smart-drivedb $(SCRDIR)
+
+clean:
+	$(MAKE) -C $(SMONTOOLS) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(SMONTOOLS) distclean
+
+sm_conf:
+	@echo "/dev/sda -H" > $(SMARTD_CONF)
+	@echo "/dev/sdb -H" >> $(SMARTD_CONF)
+	@echo "/dev/sda -W 0,0,50" >> $(SMARTD_CONF)
+	@echo "/dev/sdb -W 0,0,50" >> $(SMARTD_CONF)
+
+ifndef BUILD
+control:
+else
+CONFFILES		:= $(CTRL_DIR)/conffiles
+PREINST			:= $(CTRL_DIR)/preinst
+PRERM			:= $(CTRL_DIR)/prerm
+
+control: conffiles preinst prerm
+
+conffiles:
+#	@echo "$(USRETC)/$(SMARTD_CONF_NAME)" > $(CONFFILES)
+
+preinst:
+	@echo "#!/bin/sh" > $(PREINST)
+	@echo "killall -9 smartctl > /dev/null 2>&1" >> $(PREINST)
+	@echo "killall -9 smartd > /dev/null 2>&1" >> $(PREINST)
+	@echo "exit 0" >> $(PREINST)
+
+prerm:
+	@echo "#!/bin/sh" > $(PRERM)
+	@echo "rm -f $(TMPETC)/smart_drivedb.h" >> $(PRERM)
+	@echo "killall -9 smartctl > /dev/null 2>&1" >> $(PRERM)
+	@echo "killall -9 smartd > /dev/null 2>&1" >> $(PRERM)
+	@echo "exit 0" >> $(PRERM)
+
+endif
+
