diff -urN D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/bsd/Makefile D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/bsd/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/bsd/Makefile	2016-10-11 17:54:23.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/bsd/Makefile	2017-05-01 21:49:32.161799589 +0200
@@ -1,6 +1,6 @@
 # $Id: Makefile,v 1.1 2007/10/06 10:42:53 nanard Exp $
 # made for GNU Make
-CFLAGS = -Wall -g
+CFLAGS +=  -Os -Wall
 EXECUTABLES = testgetifstats
 
 all:	$(EXECUTABLES)
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/ipf/Makefile D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/ipf/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/ipf/Makefile	2016-10-11 17:54:23.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/ipf/Makefile	2017-05-01 21:49:32.161799589 +0200
@@ -1,6 +1,6 @@
 # $Id: Makefile,v 1.1 2007/09/25 19:44:43 nanard Exp $
 CC=gcc
-CFLAGS=-Wall -g -I.
+CFLAGS +=  -Os -Wall -I.
 
 all:	testipfrdr
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/Makefile D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/Makefile	2016-10-11 17:54:23.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/Makefile	2017-05-05 13:34:07.167635237 +0200
@@ -10,12 +10,11 @@
 # or :
 # $ make -f Makefile.linux install
 #
-#CFLAGS = -Wall -O -D_GNU_SOURCE -g -DDEBUG
 include ../../Rules.mak
 
 IPTABLE_DIR = ../iptables/iptables-1.4.16.3/
 
-CFLAGS = -Wall -Os -MD -D_GNU_SOURCE 
+CFLAGS += -Wall -Os -MD -D_GNU_SOURCE 
 CFLAGS += -I$(IPTABLE_DIR)include -L$(IPTABLE_DIR)libiptc 
 CFLAGS += -Werror -Wno-unused-but-set-variable
 #CFLAGS += -DUPNP_DEBUG
@@ -102,7 +101,7 @@
 	-rm -f ../../target/usr/sbin/miniupnpd 
 
 install: miniupnpd
-	$(STRIP) -s miniupnpd
+	$(SSTRIP) miniupnpd
 	cp -f miniupnpd ../../target/usr/sbin
 
 #install:	miniupnpd genuuid
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/netfilter/Makefile D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/netfilter/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/netfilter/Makefile	2016-10-11 17:54:23.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/netfilter/Makefile	2017-05-01 21:49:32.161799589 +0200
@@ -1,5 +1,5 @@
 # $Id: Makefile,v 1.1 2007/10/06 11:36:22 nanard Exp $
-CFLAGS=-Wall -g
+CFLAGS +=  -Os -Wall
 
 all:	iptcrdr.o testiptcrdr
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/pf/Makefile D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/pf/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/pf/Makefile	2016-10-11 17:54:23.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/miniupnpd-20080427/pf/Makefile	2017-05-01 21:49:32.161799589 +0200
@@ -1,6 +1,6 @@
 # $Id: Makefile,v 1.1 2007/10/06 10:42:53 nanard Exp $
 # made for GNU Make
-CFLAGS = -Wall -g
+CFLAGS +=  -Os -Wall
 EXECUTABLES = testobsdrdr
 
 all:	$(EXECUTABLES)
