diff -urN D7000_V1.0.1.54_WW_src/Source/apps/ez-ipupdate-3.0.11b8/Makefile D7000_V1.0.1.54_WW_src/Source/apps/ez-ipupdate-3.0.11b8/Makefile
--- D7000_V1.0.1.54_WW_src/Source/apps/ez-ipupdate-3.0.11b8/Makefile	2016-10-11 18:30:08.000000000 +0200
+++ D7000_V1.0.1.54_WW_src/Source/apps/ez-ipupdate-3.0.11b8/Makefile	2017-05-01 21:49:32.133800454 +0200
@@ -104,7 +104,7 @@
 endif
 ez_ipupdate_DEPENDENCIES = 
 ez_ipupdate_LDFLAGS = 
-CFLAGS = -s -Os -DDYNDNS -Wall
+CFLAGS +=  -s -Os -DDYNDNS -Wall
 CFLAGS += -I$(SHARED_DIR)
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(CPPFLAGS) $(CFLAGS)
 LINK = $(CC) $(CFLAGS) $(LDFLAGS) -o $@
