--- work/bftpd/bftpd/Makefile.in	2021-01-07 17:05:36.000000000 +0100
+++ work/bftpd/bftpd/Makefile.in	2021-05-24 23:01:19.928890880 +0200
@@ -1,3 +1,4 @@
+include ../../Rules.mak
 VERSION=5.7
 CC?=@CC@
 INSTALL=@INSTALL@
@@ -17,6 +18,16 @@ SRCS=bftpdutmp.c commands.c commands_adm
 OBJS2LINK=$(OBJS) $(PAX)
 LDFLAGS=
 
+ifeq ($(USB),1)
+HEADERS += ftp_usb_api.h
+OBJS += ftp_usb_api.o
+SRCS += ftp_usb_api.c
+CFLAGS += -DUSB -I$(SHARED_DIR) -I$(RC_LIBS_DIR)/include
+ifeq ($(READY_SHARE_CLOUD),1)
+CFLAGS += -DREMOTE_SMB_CONF
+endif
+endif
+
 all: bftpd
 
 bftpd: $(OBJS)
