diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/apps/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/apps/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/apps/Makefile	2020-03-30 05:16:39.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/apps/Makefile	2020-11-15 21:07:44.572691211 +0100
@@ -6,7 +6,7 @@
 TOP=		..
 CC=		cc
 INCLUDES=	-I$(TOP) -I../include $(KRB5_INCLUDES)
-CFLAG=		-g -static
+CFLAG=		$(CFLAGS) -Os -static
 MAKEFILE=	Makefile
 PERL=		perl
 RM=		rm -f
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/aes/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/aes/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/aes/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/aes/Makefile	2020-11-15 21:07:44.572691211 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 AES_ENC=aes_core.o aes_cbc.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/asn1/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/asn1/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/asn1/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/asn1/Makefile	2020-11-15 21:07:44.572691211 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
@@ -55,12 +55,12 @@
 	(cd ../..; $(MAKE) DIRS=crypto SDIRS=$(DIR) sub_all)
 
 test:	test.c
-	cc -g -I../../include -c test.c
-	cc -g -I../../include -o test test.o -L../.. -lcrypto
+	cc -I../../include -c test.c
+	cc -I../../include -o test test.o -L../.. -lcrypto
 
 pk:	pk.c
-	cc -g -I../../include -c pk.c
-	cc -g -I../../include -o pk pk.o -L../.. -lcrypto
+	cc -I../../include -c pk.c
+	cc -I../../include -o pk pk.o -L../.. -lcrypto
 
 all:	lib
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bf/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bf/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bf/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bf/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 BF_ENC=		bf_enc.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bio/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bio/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bio/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bio/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bn/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bn/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bn/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/bn/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=    $(CC) -E
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 BN_ASM=		bn_asm.o
 
@@ -53,7 +53,7 @@
 	cc -I../../include divtest.c -o divtest ../../libcrypto.a
 
 bnbug: bnbug.c ../../libcrypto.a top
-	cc -g -I../../include bnbug.c -o bnbug ../../libcrypto.a
+	cc -I../../include bnbug.c -o bnbug ../../libcrypto.a
 
 lib:	$(LIBOBJ)
 	$(AR) $(LIB) $(LIBOBJ)
@@ -166,7 +166,7 @@
 
 div:
 	rm -f a.out
-	gcc -I.. -g div.c ../../libcrypto.a
+	gcc -I.. div.c ../../libcrypto.a
 
 tags:
 	ctags $(SRC)
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/buffer/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/buffer/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/buffer/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/buffer/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/camellia/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/camellia/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/camellia/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/camellia/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CMLL_ENC= camellia.o cmll_misc.o cmll_cbc.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cast/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cast/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cast/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cast/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CAST_ENC=c_enc.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cmac/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cmac/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cmac/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cmac/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cms/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cms/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cms/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/cms/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/comp/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/comp/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/comp/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/comp/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/conf/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/conf/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/conf/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/conf/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/des/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/des/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/des/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/des/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=-I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 RANLIB=		ranlib
 DES_ENC=	des_enc.o fcrypt_b.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dh/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dh/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dh/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dh/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dsa/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dsa/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dsa/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dsa/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dso/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dso/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dso/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/dso/Makefile	2020-11-15 21:07:44.576691188 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ec/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ec/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ec/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ec/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdh/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdh/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdh/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdh/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g -Wall
+CFLAG=$(CFLAGS) -Os -Wall
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdsa/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdsa/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdsa/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ecdsa/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g -Wall
+CFLAG=$(CFLAGS) -Os -Wall
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/engine/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/engine/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/engine/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/engine/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/err/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/err/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/err/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/err/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/evp/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/evp/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/evp/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/evp/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/hmac/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/hmac/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/hmac/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/hmac/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/idea/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/idea/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/idea/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/idea/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/krb5/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/krb5/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/krb5/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/krb5/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/lhash/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/lhash/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/lhash/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/lhash/Makefile	2020-11-15 21:07:44.580691163 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -13,7 +13,7 @@
 MAKEDEPEND=	$(TOP)/util/domd $(TOP) -MD $(MAKEDEPPROG)
 MAKEFILE=       Makefile
 RM=             rm -f
-AR=		ar r
+AR=		$(CROSS)ar r
 
 RECURSIVE_MAKE=	[ -n "$(SDIRS)" ] && for i in $(SDIRS) ; do \
 		    (cd $$i && echo "making $$target in $(DIR)/$$i..." && \
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md2/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md2/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md2/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md2/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md4/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md4/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md4/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md4/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -7,7 +7,7 @@
 CC=     cc
 CPP=    $(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=       Makefile
 AR=             ar r
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md5/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md5/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md5/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/md5/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -7,7 +7,7 @@
 CC=     cc
 CPP=    $(CC) -E
 INCLUDES=-I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=       Makefile
 AR=             ar r
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/mdc2/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/mdc2/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/mdc2/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/mdc2/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/modes/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/modes/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/modes/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/modes/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 MODES_ASM_OBJ=
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/objects/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/objects/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/objects/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/objects/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 PERL=		perl
 
 CFLAGS= $(INCLUDES) $(CFLAG)
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ocsp/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ocsp/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ocsp/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ocsp/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pem/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pem/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pem/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pem/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs12/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs12/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs12/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs12/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs7/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs7/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs7/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pkcs7/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 PEX_LIBS=
 EX_LIBS=
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pqueue/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pqueue/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pqueue/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/pqueue/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rand/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rand/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rand/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rand/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc2/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc2/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc2/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc2/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc4/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc4/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc4/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc4/Makefile	2020-11-15 21:07:44.584691139 +0100
@@ -7,8 +7,8 @@
 CC=	cc
 CPP=    $(CC) -E
 INCLUDES=
-CFLAG=-g
-AR=		ar r
+CFLAG=$(CFLAGS) -Os
+AR=		$(CROSS)ar r
 
 RC4_ENC=rc4_enc.o rc4_skey.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc5/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc5/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc5/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rc5/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 RC5_ENC=		rc5_enc.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ripemd/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ripemd/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ripemd/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ripemd/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -7,7 +7,7 @@
 CC=     cc
 CPP=    $(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=       Makefile
 AR=             ar r
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rsa/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rsa/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rsa/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/rsa/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/seed/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/seed/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/seed/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/seed/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/sha/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/sha/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/sha/Makefile	2020-03-30 05:16:36.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/sha/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -7,7 +7,7 @@
 CC=     cc
 CPP=    $(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=       Makefile
 AR=             ar r
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/srp/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/srp/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/srp/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/srp/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -2,7 +2,7 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 INSTALL_PREFIX=
 OPENSSLDIR=     /usr/local/ssl
 INSTALLTOP=/usr/local/ssl
@@ -10,7 +10,7 @@
 MAKEDEPPROG=	makedepend
 MAKEDEPEND=	$(TOP)/util/domd $(TOP) -MD $(MAKEDEPPROG)
 MAKEFILE=	Makefile.ssl
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/stack/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/stack/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/stack/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/stack/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/store/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/store/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/store/Makefile	2020-03-30 05:16:39.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/store/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ts/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ts/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ts/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ts/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -13,7 +13,7 @@
 MAKEDEPPROG=	makedepend
 MAKEDEPEND=	$(TOP)/util/domd $(TOP) -MD $(MAKEDEPPROG)
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 PEX_LIBS=
 EX_LIBS=
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/txt_db/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/txt_db/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/txt_db/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/txt_db/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ui/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ui/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ui/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/ui/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/whrlpool/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/whrlpool/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/whrlpool/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/whrlpool/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -7,9 +7,9 @@
 CC=	cc
 CPP=	$(CC) -E
 INCLUDES=
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 WP_ASM_OBJ=wp_block.o
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509/Makefile	2020-03-30 05:16:37.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509/Makefile	2020-11-15 21:07:44.588691114 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509v3/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509v3/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509v3/Makefile	2020-03-30 05:16:38.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/crypto/x509v3/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -6,9 +6,9 @@
 TOP=	../..
 CC=	cc
 INCLUDES= -I.. -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 
 CFLAGS= $(INCLUDES) $(CFLAG)
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/bio/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/bio/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/bio/Makefile	2020-03-30 05:16:34.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/bio/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -1,5 +1,5 @@
 CC=cc
-CFLAGS= -g -I../../include
+CFLAGS= $(CFLAGS) -Os -I../../include
 LIBS= -L../.. ../../libssl.a ../../libcrypto.a -ldl
 EXAMPLES=saccept sconnect client-arg client-conf
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/easy_tls/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/easy_tls/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/easy_tls/Makefile	2020-03-30 05:16:34.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/easy_tls/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -1,10 +1,10 @@
 # Makefile for easy-tls example application (rudimentary client and server)
 # $Id: Makefile,v 1.2 2001/09/18 09:15:40 bodo Exp $
 
-SOLARIS_CFLAGS=-Wall -pedantic -g -O2
+SOLARIS_CFLAGS=$(CFLAGS) -Wall -pedantic -Os
 SOLARIS_LIBS=-lxnet
 
-LINUX_CFLAGS=-Wall -pedantic -g -O2
+LINUX_CFLAGS=$(CFLAGS) -Wall -pedantic -Os
 LINUX_LIBS=
 
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/cluster_labs/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/cluster_labs/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/cluster_labs/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/cluster_labs/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -5,8 +5,8 @@
 
 CC=		gcc
 PIC=		-fPIC
-CFLAGS=		-g -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT -DFLAT_INC
-AR=		ar r
+CFLAGS=		$(CFLAGS) -Os -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT -DFLAT_INC
+AR=		$(CROSS)ar r
 RANLIB=		ranlib
 
 LIB=		$(LIBNAME).a
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/ibmca/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/ibmca/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/ibmca/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/ibmca/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -5,8 +5,8 @@
 
 CC=		gcc
 PIC=		-fPIC
-CFLAGS=		-g -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT -DFLAT_INC
-AR=		ar r
+CFLAGS=		$(CFLAGS) -Os -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT -DFLAT_INC
+AR=		$(CROSS)ar r
 RANLIB=		ranlib
 
 LIB=		$(LIBNAME).a
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/rsaref/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/rsaref/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/rsaref/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/rsaref/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -5,8 +5,8 @@
 
 CC=		gcc
 PIC=		-fPIC
-CFLAGS=		-g -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT
-AR=		ar r
+CFLAGS=		$(CFLAGS) -Os -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT
+AR=		$(CROSS)ar r
 RANLIB=		ranlib
 
 LIB=		$(LIBNAME).a
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/zencod/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/zencod/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/zencod/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/engines/zencod/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -5,8 +5,8 @@
 
 CC=		gcc
 PIC=		-fPIC
-CFLAGS=		-g -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT -DFLAT_INC
-AR=		ar r
+CFLAGS=		$(CFLAGS) -Os -I../../../include $(PIC) -DENGINE_DYNAMIC_SUPPORT -DFLAT_INC
+AR=		$(CROSS)ar r
 RANLIB=		ranlib
 
 LIB=		$(LIBNAME).a
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/prime/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/prime/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/prime/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/prime/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -1,5 +1,5 @@
 CC=cc
-CFLAGS= -g -I../../include -Wall
+CFLAGS= $(CFLAGS) -Os -I../../include -Wall
 LIBS=  -L../.. -lcrypto
 EXAMPLES=prime
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/sign/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/sign/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/sign/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/sign/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -1,5 +1,5 @@
 CC=cc
-CFLAGS= -g -I../../include -Wall
+CFLAGS= $(CFLAGS) -Os -I../../include -Wall
 LIBS=  -L../.. -lcrypto
 EXAMPLES=sign
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/state_machine/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/state_machine/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/state_machine/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/state_machine/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -1,4 +1,4 @@
-CFLAGS=-I../../include -Wall -Werror -g
+CFLAGS=-I../../include -Wall -Werror $(CFLAGS) -Os
 
 all: state_machine
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/tunala/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/tunala/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/tunala/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/demos/tunala/Makefile	2020-11-15 21:07:44.592691091 +0100
@@ -9,9 +9,9 @@
 
 RM=rm -f
 CC=gcc
-DEBUG_FLAGS=-g -ggdb3 -Wall -Wshadow
+DEBUG_FLAGS=-Wall -Wshadow
 INCLUDE_FLAGS=-I$(SSL_INCLUDEDIR)
-CFLAGS=$(DEBUG_FLAGS) $(INCLUDE_FLAGS) -DNO_CONFIG_H
+CFLAGS +=  $(DEBUG_FLAGS) $(INCLUDE_FLAGS) -DNO_CONFIG_H -Os
 COMPILE=$(CC) $(CFLAGS) -c
 
 # Edit, particularly the "-ldl" if not building with "dlfcn" support
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/ccgost/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/ccgost/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/ccgost/Makefile	2020-03-30 05:16:39.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/ccgost/Makefile	2020-11-15 21:07:44.596691066 +0100
@@ -2,7 +2,7 @@
 TOP=../..
 CC=cc
 INCLUDES= -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE= Makefile
 AR= ar r
 CFLAGS= $(INCLUDES) $(CFLAG)
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/Makefile	2020-03-30 05:16:39.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/engines/Makefile	2020-11-15 21:07:44.596691066 +0100
@@ -6,9 +6,9 @@
 TOP=	..
 CC=	cc
 INCLUDES= -I../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 ENGDIRS= ccgost
 
 RECURSIVE_MAKE=	[ -z "$(ENGDIRS)" ] || for i in $(ENGDIRS) ; do \
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/Makefile	2020-11-15 21:07:44.600691042 +0100
@@ -64,7 +64,7 @@
 CC= $(CROSS)gcc
 STRIP = $(CROSS)strip
 RANLIBRANLIB = $(CROSS)ranlib
-CFLAG= -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -DTLS_FALLBACK_SCSV -DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall -DOPENSSL_NO_SSL3
+CFLAG= $(CFLAGS) -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -DTLS_FALLBACK_SCSV -DL_ENDIAN -DTERMIO -Os -fomit-frame-pointer -Wall -DOPENSSL_NO_SSL3
 DEPFLAG= -DOPENSSL_NO_EC_NISTP_64_GCC_128 -DOPENSSL_NO_GMP -DOPENSSL_NO_JPAKE -DOPENSSL_NO_LIBUNBOUND -DOPENSSL_NO_MD2 -DOPENSSL_NO_RC5 -DOPENSSL_NO_RFC3779 -DOPENSSL_NO_SCTP -DOPENSSL_NO_SSL_TRACE -DOPENSSL_NO_SSL2 -DOPENSSL_NO_STORE -DOPENSSL_NO_UNIT_TEST -DOPENSSL_NO_WEAK_SSL_CIPHERS -DOPENSSL_NO_SSL3
 PEX_LIBS= 
 EX_LIBS= -ldl
@@ -541,7 +541,7 @@
 	$(STRIP) -g  $(SHARED_CRYPTO) $(SHARED_SSL) 
 	cp -a $(SHARED_CRYPTO) $(SHARED_SSL) ../../target/lib
 	cp -af libcrypto.so* libssl.so* ../../target/lib
-	$(STRIP) apps/openssl
+	$(SSTRIP) apps/openssl
 	cp -af apps/openssl $(ROOT)/target/usr/sbin
 install_sw:
 	@$(PERL) $(TOP)/util/mkdir-p.pl $(INSTALL_PREFIX)$(INSTALLTOP)/bin \
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/ssl/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/ssl/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/ssl/Makefile	2020-03-30 05:16:39.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/ssl/Makefile	2020-11-15 21:07:44.596691066 +0100
@@ -6,9 +6,9 @@
 TOP=	..
 CC=	cc
 INCLUDES= -I../crypto -I$(TOP) -I../include $(KRB5_INCLUDES)
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
-AR=		ar r
+AR=		$(CROSS)ar r
 # KRB5 stuff
 KRB5_INCLUDES=
 
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/test/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/test/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/test/Makefile	2020-03-30 05:16:39.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/test/Makefile	2020-11-15 21:07:44.596691066 +0100
@@ -6,7 +6,7 @@
 TOP=		..
 CC=		cc
 INCLUDES=	-I$(TOP) -I../include $(KRB5_INCLUDES)
-CFLAG=		-g
+CFLAG=		$(CFLAGS) -Os
 MAKEDEPEND=	$(TOP)/util/domd $(TOP) -MD $(MAKEDEPPROG)
 PERL=		perl
 # KRB5 stuff
diff -urN D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/tools/Makefile D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/tools/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/tools/Makefile	2020-03-30 05:16:35.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/openssl-1.0.2h/tools/Makefile	2020-11-15 21:07:44.596691066 +0100
@@ -6,7 +6,7 @@
 TOP=	..
 CC=	cc
 INCLUDES= -I$(TOP) -I../../include
-CFLAG=-g
+CFLAG=$(CFLAGS) -Os
 MAKEFILE=	Makefile
 
 CFLAGS= $(INCLUDES) $(CFLAG)
