diff -urN D7000_V1.0.1.84_WW_src/Source/apps/hd-idle/Makefile D7000_V1.0.1.84_WW_src/Source/apps/hd-idle/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/hd-idle/Makefile	2016-10-11 17:56:23.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/hd-idle/Makefile	2017-05-02 13:25:35.755702707 +0200
@@ -11,7 +11,7 @@
 LIB_DIRS   =  
 
 CC = $(CROSS)gcc
-CFLAGS    += -Wall
+CFLAGS    += -Os -Wall
 
 LD         = $(CC)
 LDFLAGS   += $(LIB_DIRS)
@@ -39,9 +39,7 @@
 	rm -f $(OBJS) $(TARGET)
 
 install: 
-	#install -D -g root -o root $(TARGET) $(TARGET_DIR)/sbin/$(TARGET)
-	#install -D -g root -o root $(TARGET).1 $(TARGET_DIR)/share/man/man1/$(TARGET).1
-	$(STRIP) $(TARGET)
+	$(SSTRIP) $(TARGET)
 	cp -a  $(TARGET) $(TARGET_DIR)
 
 hd-idle.o:     hd-idle.c
