diff -urN work/e2fsprogs-1.44.5/e2fsprogs-1.44.5/Makefile.in work/e2fsprogs-1.44.5/e2fsprogs-1.44.5/Makefile.in
--- work/e2fsprogs-1.44.5/e2fsprogs-1.44.5/Makefile.in	2018-10-23 23:00:46.662292000 +0200
+++ work/e2fsprogs-1.44.5/e2fsprogs-1.44.5/Makefile.in	2018-10-23 23:01:08.566300143 +0200
@@ -18,20 +18,19 @@
 @ALL_CMT@EXT2FS_LIB_SUBDIR= lib/ext2fs
 
 LIB_SUBDIRS=lib/et lib/ss $(E2P_LIB_SUBDIR) $(UUID_LIB_SUBDIR) \
-	$(BLKID_LIB_SUBDIR) $(SUPPORT_LIB_SUBDIR) $(EXT2FS_LIB_SUBDIR) intl
+	$(BLKID_LIB_SUBDIR) $(SUPPORT_LIB_SUBDIR) $(EXT2FS_LIB_SUBDIR)
 
-PROG_SUBDIRS=e2fsck $(DEBUGFS_DIR) misc $(RESIZE_DIR) tests/progs po
-SUBDIRS=util $(LIB_SUBDIRS) $(PROG_SUBDIRS) tests
+PROG_SUBDIRS=e2fsck
+SUBDIRS=$(LIB_SUBDIRS) $(PROG_SUBDIRS)
 
 SUBS= util/subst.conf lib/config.h $(top_builddir)/lib/dirpaths.h \
 	lib/ext2fs/ext2_types.h lib/blkid/blkid_types.h lib/uuid/uuid_types.h
 
 TAR=tar
 
-all:: subs
+all:: 
 	$(MAKE) libs
 @ALL_CMT@	$(MAKE) progs
-@ALL_CMT@	$(MAKE) docs
 
 subs: $(DEP_SUBSTITUTE)
 	@for i in $(SUBS) ; do if test -d `dirname $$i` ; \
