diff -urN work/iproute2-4.10.0/Makefile work/iproute2-4.10.0/Makefile
--- work/iproute2-4.10.0/Makefile	2017-02-20 17:47:52.000000000 +0100
+++ work/iproute2-4.10.0/Makefile	2017-04-01 12:47:22.848442441 +0200
@@ -1,3 +1,5 @@
+include ../Rules.mak
+
 # Include "Config" if already generated
 ifneq ($(wildcard Config),)
 include Config
@@ -16,7 +18,7 @@
 DOCDIR?=$(DATADIR)/doc/iproute2
 MANDIR?=$(DATADIR)/man
 ARPDDIR?=/var/lib/arpd
-KERNEL_INCLUDE?=/usr/include
+KERNEL_INCLUDE=$(KERNEL_INC)
 BASH_COMPDIR?=$(DATADIR)/bash-completion/completions
 
 # Path to db_185.h include
@@ -40,19 +42,20 @@
 #options for mpls
 ADDLIB+=mpls_ntop.o mpls_pton.o
 
-CC := gcc
+CC := $(CROSS)gcc
 HOSTCC ?= $(CC)
 DEFINES += -D_GNU_SOURCE
 # Turn on transparent support for LFS
 DEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CCOPTS = -O2
+CCOPTS = -Os
+#CCOPTS += -O2
 WFLAGS := -Wall -Wstrict-prototypes  -Wmissing-prototypes
 WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
 
-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include $(DEFINES) $(CFLAGS)
+CFLAGS := $(WFLAGS) $(CCOPTS) -I../include $(DEFINES) $(CFLAGS) -fcommon
 YACCFLAGS = -d -t -v
 
-SUBDIRS=lib ip tc bridge misc netem genl tipc devlink man
+SUBDIRS=lib ip tc #bridge misc netem genl tipc devlink man
 
 LIBNETLINK=../lib/libnetlink.a ../lib/libutil.a
 LDLIBS += $(LIBNETLINK)
@@ -70,22 +73,16 @@
 Config:
 	sh configure $(KERNEL_INCLUDE)
 
-install: all
-	install -m 0755 -d $(DESTDIR)$(SBINDIR)
-	install -m 0755 -d $(DESTDIR)$(CONFDIR)
-	install -m 0755 -d $(DESTDIR)$(ARPDDIR)
-	install -m 0755 -d $(DESTDIR)$(HDRDIR)
-	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples
-	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples/diffserv
-	install -m 0644 README.iproute2+tc $(shell find examples -maxdepth 1 -type f) \
-		$(DESTDIR)$(DOCDIR)/examples
-	install -m 0644 $(shell find examples/diffserv -maxdepth 1 -type f) \
-		$(DESTDIR)$(DOCDIR)/examples/diffserv
-	@for i in $(SUBDIRS) doc; do $(MAKE) -C $$i install; done
-	install -m 0644 $(shell find etc/iproute2 -maxdepth 1 -type f) $(DESTDIR)$(CONFDIR)
-	install -m 0755 -d $(DESTDIR)$(BASH_COMPDIR)
-	install -m 0644 bash-completion/tc $(DESTDIR)$(BASH_COMPDIR)
-	install -m 0644 include/bpf_elf.h $(DESTDIR)$(HDRDIR)
+install: control
+	$(SSTRIP) tc/tc ip/ip
+	if [ -e $(ETCDIR)/iproute2 ]; then \
+	rm -rf $(ETCDIR)/iproute2;\
+	fi
+	install -m 0755 -d $(ETCDIR)
+	install -m 0755 -d $(BINDIR)
+	install -m 0755 tc/tc $(BINDIR)
+	install -m 0755 ip/ip $(BINDIR)
+	cp -rf etc/iproute2 $(ETCDIR)/iproute2
 
 snapshot:
 	echo "static const char SNAPSHOT[] = \""`date +%y%m%d`"\";" \
@@ -105,4 +102,46 @@
 cscope:
 	cscope -b -q -R -Iinclude -sip -slib -smisc -snetem -stc
 
+ifndef BUILD
+control:
+else
+PREINST			:= $(CTRL_DIR)/preinst
+POSTINST		:= $(CTRL_DIR)/postinst
+PRERM			:= $(CTRL_DIR)/prerm
+
+control: preinst postinst prerm
+
+preinst:
+	@echo "#!/bin/sh" > $(PREINST)
+	@echo "[ -d $(USRETC)/iproute2 ] && mv -f $(USRETC)/iproute2 $(USRETC)/iproute2.orig" >> $(PREINST)
+	@echo "[ -d $(TMPETC)/iproute2 ] && mv -f $(TMPETC)/iproute2 $(TMPETC)/iproute2.orig" >> $(PREINST)
+	@echo "[ -f $(USRSBIN)/ip ] && cp -f $(USRSBIN)/ip $(USRSBIN)/ip.orig" >> $(PREINST)
+	@echo "[ -f $(USRSBIN)/tc ] && cp -f $(USRSBIN)/tc $(USRSBIN)/tc.orig" >> $(PREINST)
+	@echo "exit 0" >> $(PREINST)
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "cp -rf $(USRETC)/iproute2 $(TMPETC)/iproute2" >> $(POSTINST)
+	@echo "echo \"Testing iproute2 basic functionality...\"" >> $(POSTINST)
+	@echo "echo" >> $(POSTINST)
+	@echo "$(USRSBIN)/tc qdisc show > /dev/null 2>&1" >> $(POSTINST)
+	@echo "if [ \$$? -eq 0 ]; then" >> $(POSTINST)
+	@echo "echo \"Looking good\"" >> $(POSTINST)
+	@echo "rm -rf $(USRETC)/iproute2.orig $(TMPETC)/iproute2.orig $(USRSBIN)/ip.orig $(USRSBIN)/tc.orig" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+	@echo "else" >> $(POSTINST)
+	@echo "echo \"Something wrong: reverting to original iproute2 ...\"" >> $(POSTINST)
+	@echo "[ -d $(USRETC)/iproute2.orig ] && mv -f $(USRETC)/iproute2.orig $(USRETC)/iproute2" >> $(POSTINST)
+	@echo "[ -d $(TMPETC)/iproute2.orig ] && mv -f $(TMPETC)/iproute2.orig $(TMPETC)/iproute2" >> $(POSTINST)
+	@echo "[ -f $(USRSBIN)/ip.orig ] && mv -f $(USRSBIN)/ip.orig $(USRSBIN)/ip" >> $(POSTINST)
+	@echo "[ -f $(USRSBIN)/tc.orig ] && mv -f $(USRSBIN)/tc.orig $(USRSBIN)/tc" >> $(POSTINST)
+	@echo "exit 1" >> $(POSTINST)
+	@echo "fi" >> $(POSTINST)
+prerm:
+	@echo "#!/bin/sh" > $(PRERM)
+	@echo "echo \"iproute2 builtin required: cannot uninstall...\"" >> $(PRERM)
+	@echo "exit 1" >> $(PRERM)
+	@echo "fi" >> $(PRERM)
+endif
+
 .EXPORT_ALL_VARIABLES:
