--- work/opkg-0.6.0/opkg-0.6.0/libopkg/opkg_download_curl.c
+++ work/opkg-0.6.0/opkg-0.6.0/libopkg/opkg_download_curl.c
@@ -219,6 +219,7 @@
     double src_size = -1;
     int ret = 1;
 
+    curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);
     curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, &dummy_write);
     curl_easy_setopt(curl, CURLOPT_HEADERFUNCTION, &header_write);
     curl_easy_setopt(curl, CURLOPT_WRITEHEADER, &etag);
