diff -urN D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/.config D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/.config
--- D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/.config	2016-10-11 17:53:04.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/.config	2017-02-20 20:35:15.261056548 +0100
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Version: 0.9.32
-# Thu Jun 11 14:45:26 2015
+# Mon Feb 20 20:35:15 2017
 #
 # TARGET_alpha is not set
 TARGET_arm=y
@@ -67,8 +67,7 @@
 ARCH_HAS_MMU=y
 ARCH_USE_MMU=y
 UCLIBC_HAS_FLOATS=y
-# UCLIBC_HAS_FPU is not set
-UCLIBC_HAS_SOFT_FLOAT=y
+UCLIBC_HAS_FPU=y
 DO_C99_MATH=y
 # DO_XSI_MATH is not set
 # UCLIBC_HAS_FENV is not set
diff -urN D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/auto.conf D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/auto.conf
--- D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/auto.conf	2016-10-11 18:30:54.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/auto.conf	2017-02-20 20:35:15.269056448 +0100
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Version: 0.9.32
-# Wed Oct 12 00:30:53 2016
+# Mon Feb 20 20:35:15 2017
 #
 LDSO_LDD_SUPPORT=y
 ASSUME_DEVPTS=y
@@ -25,6 +25,7 @@
 UCLIBC_HAS_FOPEN_LARGEFILE_MODE=y
 UCLIBC_HAS_STDIO_BUFSIZ_4096=y
 UCLIBC_HAS_ADVANCED_REALTIME=y
+UCLIBC_HAS_FPU=y
 UCLIBC_HAS_COMPAT_RES_STATE=y
 HAVE_SHARED=y
 UCLIBC_HAS_STDIO_PUTC_MACRO=y
@@ -51,7 +52,6 @@
 ARCH_LITTLE_ENDIAN=y
 UCLIBC_HAS_RESOLVER_SUPPORT=y
 UCLIBC_HAS_TM_EXTENSIONS=y
-UCLIBC_HAS_SOFT_FLOAT=y
 UCLIBC_HAS_FOPEN_EXCLUSIVE_MODE=y
 UCLIBC_HAS_LIBUTIL=y
 UCLIBC_HAS___PROGNAME=y
diff -urN D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/autoconf.h D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/autoconf.h
--- D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/autoconf.h	2016-10-11 18:30:54.000000000 +0200
+++ D7000_V1.0.1.44_WW_src/Source/uClibc-0.9.32/include/config/autoconf.h	2017-02-20 20:35:15.269056448 +0100
@@ -1,7 +1,7 @@
 /*
  * Automatically generated C config: don't edit
  * Version: 0.9.32
- * Wed Oct 12 00:30:53 2016
+ * Mon Feb 20 20:35:15 2017
  */
 #define AUTOCONF_INCLUDED
 #define LDSO_LDD_SUPPORT 1
@@ -26,6 +26,7 @@
 #define UCLIBC_HAS_FOPEN_LARGEFILE_MODE 1
 #define UCLIBC_HAS_STDIO_BUFSIZ_4096 1
 #define UCLIBC_HAS_ADVANCED_REALTIME 1
+#define UCLIBC_HAS_FPU 1
 #define UCLIBC_HAS_COMPAT_RES_STATE 1
 #define HAVE_SHARED 1
 #define UCLIBC_HAS_STDIO_PUTC_MACRO 1
@@ -52,7 +53,6 @@
 #define ARCH_LITTLE_ENDIAN 1
 #define UCLIBC_HAS_RESOLVER_SUPPORT 1
 #define UCLIBC_HAS_TM_EXTENSIONS 1
-#define UCLIBC_HAS_SOFT_FLOAT 1
 #define UCLIBC_HAS_FOPEN_EXCLUSIVE_MODE 1
 #define UCLIBC_HAS_LIBUTIL 1
 #define UCLIBC_HAS___PROGNAME 1

