diff -urN D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/build.sh D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/build.sh
--- D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/build.sh	2016-10-11 17:53:53.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/build.sh	2017-05-04 21:56:31.481920266 +0200
@@ -2,4 +2,4 @@
 build_dir=build_sc
 cd $build_dir
 
-CPPFLAGS="-I`pwd`/../ -I`pwd`/../include/ " CFLAGS="-I`pwd`/../ -I`pwd`/../include/  -I`pwd`/../../expat-2.0.1/lib/" LDFLAGS="-L`pwd`/../../target/usr/local/lib/" ../configure --target=arm-linux-uclibc --host=arm-linux-uclibc --build=i486-linux-gnu --enable-shared --enable-static --prefix=
+../configure --target=arm-linux-uclibc --host=arm-linux-uclibc --build=i486-linux-gnu --enable-shared --enable-static --prefix= CPPFLAGS="${1} -I`pwd`/../ -I`pwd`/../include/ -I`pwd`/../ -I`pwd`/../include/  -I`pwd`/../../expat-2.0.1/lib/" CXXFLAGS="-Os -s" LDFLAGS="-L`pwd`/../../target/usr/local/lib/"
diff -urN D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/configure D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/configure
--- D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/configure	2016-10-11 17:53:53.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/apple/libantlr3c-3.2/configure	2017-05-04 22:24:54.827738851 +0200
@@ -19493,7 +19493,7 @@
  if test x"$enable_64bit" = xyes; then
   CCFLAGS64="+DD64"
  fi
- OPTIMFLAGS="-O"
+ OPTIMFLAGS=""
  if test $DEBUGINF = 1; then
   DEBUGFLAGS="-g"
  fi
@@ -19524,7 +19524,7 @@
  if test x"$enable_64bit" = xyes; then
   CCFLAGS64="-fast -xtarget=ultra4 -m64 -xarch=sparcvis"
  fi
- OPTIMFLAGS="-O"
+ OPTIMFLAGS=""
  if test x"$enable_debuginfo" = xyes; then
   DEBUGFLAGS='-g'
  fi
@@ -19533,7 +19533,7 @@
 *)
  CPPCMNTFLAGS=
  CCFLAGS64=
- OPTIMFLAGS="-O"
+ OPTIMFLAGS=""
  if test x"$enable_debuginfo" = xyes; then
   DEBUGFLAGS='-g'
  fi
