diff -urN D7000_V1.0.1.84_WW_src/Source/apps/apple/avahi-0.6.25/build.sh D7000_V1.0.1.84_WW_src/Source/apps/apple/avahi-0.6.25/build.sh
--- D7000_V1.0.1.84_WW_src/Source/apps/apple/avahi-0.6.25/build.sh	2016-10-11 17:54:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/apple/avahi-0.6.25/build.sh	2017-05-04 20:53:03.767753219 +0200
@@ -6,7 +6,7 @@
 export LIBDAEMON_LIBS="-L`pwd`/../../target/usr/local/lib -ldaemon" 
 
 
-CC=/opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/usr/bin/arm-linux-gcc  CFLAGS="-I`pwd`/../../target/lib/dbus-1.0/include/ -DHAVE_DBUS  -I`pwd`/../../target/include/dbus-1.0/ -I`pwd`/../../target/usr/include -I`pwd`/../../target/usr/local/include" DBUS_LIBS="`pwd`/../../dbus-1.8.0/build_sc/dbus/.libs/libdbus-1.so" LDFLAGS="-L`pwd`/../../dbus-1.8.0/build_sc/dbus/.libs/ -ldbus-1 -L`pwd`/../../target/usr/lib -L`pwd`/../../target/usr/local/lib" ../configure --target=arm-linux --host=arm-linux --build=i686-pc-linux-gnu --prefix= --sbindir=/usr/sbin --sysconfdir=/etc --exec-prefix=/usr--sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --mandir=/usr/man --exec-prefix=/usr\
+CC=/opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/usr/bin/arm-linux-gcc  CFLAGS="${1} -I`pwd`/../../target/lib/dbus-1.0/include/ -DHAVE_DBUS  -I`pwd`/../../target/include/dbus-1.0/ -I`pwd`/../../target/usr/include -I`pwd`/../../target/usr/local/include" DBUS_LIBS="`pwd`/../../dbus-1.8.0/build_sc/dbus/.libs/libdbus-1.so" LDFLAGS="-L`pwd`/../../dbus-1.8.0/build_sc/dbus/.libs/ -ldbus-1 -L`pwd`/../../target/usr/lib -L`pwd`/../../target/usr/local/lib" ../configure --target=arm-linux --host=arm-linux --build=i686-pc-linux-gnu --prefix= --sbindir=/usr/sbin --sysconfdir=/etc --exec-prefix=/usr--sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --mandir=/usr/man --exec-prefix=/usr\
 	--enable-shared \
 	--enable-static \
 	--disable-glib \
