diff -urN D7000_V1.0.1.68_WW_src/Source/apps/apple/db-4.7.25.NC/build.sh D7000_V1.0.1.68_WW_src/Source/apps/apple/db-4.7.25.NC/build.sh
--- D7000_V1.0.1.68_WW_src/Source/apps/apple/db-4.7.25.NC/build.sh	2016-10-11 17:53:48.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/apple/db-4.7.25.NC/build.sh	2017-05-04 20:53:03.767753219 +0200
@@ -1,4 +1,4 @@
 #!/bin/bash
 build_dir=build_sc
 cd $build_dir
-CC=/opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/usr/bin/arm-linux-gcc ../dist/configure --target=arm-linux-gcc --host=arm-linux -build=i686-pc-linux-gnu --prefix=/usr --enable-shared --enable-static --disable-java --disable-cxx --with-mutex=UNIX/fcntl --disable-tcl --disable-rpc --enable-compat185 --enable-debug --enable-statistics --enable-replication --enable-cryptography
+CFLAGS="${1}" CC=/opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/usr/bin/arm-linux-gcc ../dist/configure --target=arm-linux-gcc --host=arm-linux -build=i686-pc-linux-gnu --prefix=/usr --enable-shared --enable-static --disable-java --disable-cxx --with-mutex=UNIX/fcntl --disable-tcl --disable-rpc --enable-compat185 --enable-debug --enable-statistics --enable-replication --enable-cryptography
