diff -urN D7000_V1.0.1.68_WW_src/Source/apps/apple/confuse-2.6/build.sh D7000_V1.0.1.68_WW_src/Source/apps/apple/confuse-2.6/build.sh
--- D7000_V1.0.1.68_WW_src/Source/apps/apple/confuse-2.6/build.sh	2016-10-11 17:54:04.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/apple/confuse-2.6/build.sh	2017-05-04 20:53:03.767753219 +0200
@@ -2,4 +2,4 @@
 build_dir=build_sc
 cd $build_dir
 
-CPPFLAGS="" CFLAGS=" -I`pwd`/../../expat-2.0.1/lib/" LDFLAGS="-L`pwd`/../../target/usr/local/lib/" ../configure --target=arm-linux-uclibc --host=arm-linux-uclibc --build=i486-linux-gnu --enable-shared --enable-static --prefix=
+CPPFLAGS="" CFLAGS="${1} -I`pwd`/../../expat-2.0.1/lib/" LDFLAGS="-L`pwd`/../../target/usr/local/lib/" ../configure --target=arm-linux-uclibc --host=arm-linux-uclibc --build=i486-linux-gnu --enable-shared --enable-static --prefix=
