diff -urN D7000_V1.0.1.68_WW_src/Source/apps/apple/dbus-1.8.0/build.sh D7000_V1.0.1.68_WW_src/Source/apps/apple/dbus-1.8.0/build.sh
--- D7000_V1.0.1.68_WW_src/Source/apps/apple/dbus-1.8.0/build.sh	2016-10-11 17:53:58.000000000 +0200
+++ D7000_V1.0.1.68_WW_src/Source/apps/apple/dbus-1.8.0/build.sh	2017-05-04 22:07:02.425250532 +0200
@@ -2,6 +2,4 @@
 build_dir=build_sc
 cd $build_dir
 
-CFLAGS_VARS=" -I`pwd/`../ -I`pwd`/../../expat-2.0.1/lib/ ";
-
-CPPFLAGS="$CFLAGS_VARS" CFLAGS="$CFLAGS_VARS" LDFLAGS="-L`pwd`/../../target/usr/local/lib/" ../configure --target=arm-linux-uclibc --host=arm-linux-uclibc --build=i486-linux-gnu --enable-shared --enable-static --disable-launchd --disable-systemd --prefix=
+CPPFLAGS="$CFLAGS_VARS" CFLAGS="${1}  -I`pwd`/../ -I`pwd`/../../expat-2.0.1/lib/" LDFLAGS="-L`pwd`/../../target/usr/local/lib/" ../configure --target=arm-linux-uclibc --host=arm-linux-uclibc --build=i486-linux-gnu --enable-shared --enable-static --disable-launchd --disable-systemd --prefix=
