diff -urN work/e2fsprogs-1.46.5/e2fsprogs-1.46.5/configure work/e2fsprogs-1.46.5/e2fsprogs-1.46.5/configure
--- work/e2fsprogs-1.46.5/e2fsprogs-1.46.5/configure	2018-10-23 23:00:46.662292000 +0200
+++ work/e2fsprogs-1.46.5/e2fsprogs-1.46.5/configure	2018-10-23 23:01:08.566300143 +0200
@@ -12538,7 +12538,7 @@
 
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ${crond_dir}" >&5
 $as_echo "${crond_dir}" >&6; }
-		have_crond="yes"
+		have_crond="no"; with_crond_dir=""
 
 else
 
