diff -urN D7000_V1.0.1.54_WW_src/Source/apps/zip30/unix/Makefile D7000_V1.0.1.54_WW_src/Source/apps/zip30/unix/Makefile
--- D7000_V1.0.1.54_WW_src/Source/apps/zip30/unix/Makefile	2016-10-11 17:56:23.000000000 +0200
+++ D7000_V1.0.1.54_WW_src/Source/apps/zip30/unix/Makefile	2017-05-01 23:03:09.358383818 +0200
@@ -59,7 +59,7 @@
 #   LFLAGS1   flags after output file spec, before obj file list
 #   LFLAGS2   flags after obj file list (libraries, etc)
 CFLAGS_NOOPT = -I. -DUNIX $(LOCAL_ZIP)
-CFLAGS = -O2 $(CFLAGS_NOOPT)
+CFLAGS += -Os $(CFLAGS_NOOPT)
 LFLAGS1 =
 LFLAGS2 = -s
 
@@ -158,7 +158,7 @@
 
 # install
 install:        $(ZIPS)
-	$(STRIP) $(ZIPS)
+	$(SSTRIP) $(ZIPS)
 	$(INSTALL_PROGRAM) $(ZIPS) $(BINDIR)
 	-cd $(BINDIR); $(CHMOD) $(BINFLAGS) $(ZIPS)
 #	-$(INSTALL_D) $(MANDIR)
@@ -178,7 +178,7 @@
 
 
 flags:  unix/configure
-	sh unix/configure "${CC}" "${CFLAGS_NOOPT}" "${IZ_BZIP2}"
+	sh unix/configure "${CC}" "${CFLAGS}" "${IZ_BZIP2}"
 
 # These symbols, when #defined using -D have these effects on compilation:
 # ZMEM                  - includes C language versions of memset(), memcpy(),
