diff -urN D7000_V1.0.1.64_WW_src/Source/apps/flex-2.5.4/Makefile D7000_V1.0.1.64_WW_src/Source/apps/flex-2.5.4/Makefile
--- D7000_V1.0.1.64_WW_src/Source/apps/flex-2.5.4/Makefile	2016-10-11 17:54:42.000000000 +0200
+++ D7000_V1.0.1.64_WW_src/Source/apps/flex-2.5.4/Makefile	2017-04-29 21:20:10.102554955 +0200
@@ -18,7 +18,7 @@
 include ../../Rules.mak
 CROSS_PREFIX=$(CROSS)
 
-CFLAGS = -g -O
+CFLAGS += -Os
 CPPFLAGS = 
 DEFS = -DHAVE_CONFIG_H
 LDFLAGS = 
@@ -88,6 +88,7 @@
 
 
 all: $(FLEX)
+	$(SSTRIP) $(FLEX)
 
 $(FLEX): .bootstrap $(OBJECTS) $(FLEXLIB)
 	$(CC) $(CFLAGS) -o $(FLEX) $(LDFLAGS) $(OBJECTS) $(FLEXLIB) $(LIBS)
@@ -158,6 +159,7 @@
 $(FLEXLIB): $(LIBOBJS)
 	$(AR) cru $(FLEXLIB) $(LIBOBJS)
 	-$(RANLIB) $(FLEXLIB)
+	$(STRIP) $(STRIPFLAGS) $(FLEXLIB)
 
 $(FLEX).man: flex.1
 	cd $(srcdir) && nroff -man flex.1 >$(FLEX).man
