diff -urN work/utelnetd-0.1.11/Makefile work/utelnetd-0.1.11/Makefile
--- work/utelnetd-0.1.11/Makefile	2008-08-11 11:44:01.000000000 +0200
+++ work/utelnetd-0.1.11/Makefile	2017-03-04 14:43:44.182317642 +0100
@@ -1,63 +1,68 @@
-# Makefile for utelnetd
-# 
-# Configure this with the following environment variables:
-#
+include ../Rules.mak
 
-# where to install
-INSTDIR		:= /usr/local/bin/
+BINS	:= utelnetd
+STARTUP	:= rc_debug_mode
+OBJS	:= $(BINS).o
+LDFLAGS	:=
+LDLIBS	:=
 
 # where to find login program
-ifneq ("", "$(BSD)")
-LOGIN		:= /usr/bin/login
-else
-LOGIN		:= /bin/login
-endif
-
-ifneq ("", "$(BSD)")
-CORE		:= utelnetd.core
-else
-CORE		:= core
-endif
-
-# nothing to configure below this line... ---8<---8<---8<---
-
-PROGS     = utelnetd
-
-INSTMODE  = 0755
-INSTOWNER = root
-INSTGROUP = root
+LOGIN	:= $(USRSBIN)/login
 
-OBJS      = utelnetd.o
+CFLAGS	+= -Os -I. -pipe -DSHELLPATH=\"$(LOGIN)\" -Wall
 
-CC        = $(CROSS_COMPILE)gcc
-INSTALL   = install
-
-CFLAGS	 += -I. -pipe -DSHELLPATH=\"$(LOGIN)\" -Wall
+ifndef LOCAL
+CFLAGS	+= -Werror -Wno-unused-but-set-variable
+endif
 
-ifneq ("","$(DEBUG)")
-CFLAGS   += -DDEBUG -g -Os
-STRIP	  = \#
+ifdef DEBUG
+CFLAGS	+= -DDEBUG -g
 else
-CFLAGS	 += -fomit-frame-pointer
-STRIP	  = $(CROSS_COMPILE)strip 
+CFLAGS	+= -fomit-frame-pointer
 endif
 
 ifeq ("1", "$(BSD)")
-CFLAGS   += -DBSD
+CFLAGS	+= -DBSD
 endif
 
 
-all: $(PROGS)
+all: $(BINS)
 
-$(PROGS): $(OBJS)
-	$(CC) $(LDFLAGS) $(CFLAGS) $^ $(LDLIBS) -o $@
-	$(STRIP) --remove-section=.comment --remove-section=.note $@
+$(BINS): $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(LDLIBS) $^ -o $@
 
 .PHONY: install
-install: $(PROGS)
-	$(INSTALL) -d $(DESTDIR)$(INSTDIR)
-	$(INSTALL) -m $(INSTMODE) -o $(INSTOWNER) -g $(INSTGROUP) $(PROGS) $(DESTDIR)$(INSTDIR)
+install: control
+	$(SSTRIP) $(BINS)
+	install -m 0755 -d $(BINDIR)
+	install -m 0755 -d $(APPDIR)
+	install -m 0755 $(BINS) $(BINDIR)
+	install -m 0755 $(STARTUP) $(APPDIR)
 
 .PHONY: clean
 clean:
-	rm -f $(PROGS) *.o $(CORE)
+	rm -f *.o $(BINS)
+
+ifndef BUILD
+control:
+else
+POSTINST		:= $(CTRL_DIR)/postinst
+PRERM			:= $(CTRL_DIR)/prerm
+
+control: postinst prerm
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "[ -z \"\`pidof $(BINS)\`\" ] && $(USRSBINRCAPP)/$(STARTUP) start" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+
+prerm:
+	@echo "#!/bin/sh" > $(PRERM)
+	@echo "if [ \"\$$TERM\" != \"vt102\" ]; then" >> $(PRERM)
+	@echo "killall -9 $(BINS) >/dev/null 2>&1" >> $(PRERM)
+	@echo "else" >> $(PRERM)
+	@echo "echo \"When finished, kill manually $(BINS) typing: 'killall -9 $(BINS)'\"" >> $(PRERM)
+	@echo "fi" >> $(PRERM)
+	@echo "exit 0" >> $(PRERM)
+endif
+
