diff -urN work/openssh-7.6p1/openssh-7.6p1/Makefile.in work/openssh-7.6p1/openssh-7.6p1/Makefile.in
--- work/openssh-7.6p1/openssh-7.6p1/Makefile.in	1970-01-01 01:00:00.000000000 +0100
+++ work/openssh-7.6p1/openssh-7.6p1/Makefile.in	2018-01-05 12:00:11.000000000 +0100
@@ -62,7 +62,7 @@
 EXEEXT=@EXEEXT@
 MANFMT=@MANFMT@
 
-TARGETS=ssh$(EXEEXT) sshd$(EXEEXT) ssh-add$(EXEEXT) ssh-keygen$(EXEEXT) ssh-keyscan${EXEEXT} ssh-keysign${EXEEXT} ssh-pkcs11-helper$(EXEEXT) ssh-agent$(EXEEXT) scp$(EXEEXT) sftp-server$(EXEEXT) sftp$(EXEEXT)
+TARGETS=sshd$(EXEEXT) ssh-keygen$(EXEEXT) scp$(EXEEXT)
 
 LIBOPENSSH_OBJS=\
 	ssh_api.o \
