diff -urN work/openssh-7.6p1/openssh-7.6p1/log.c work/openssh-7.6p1/openssh-7.6p1/log.c
--- work/openssh-7.6p1/openssh-7.6p1/log.c	1970-01-01 01:00:00.000000000 +0100
+++ work/openssh-7.6p1/openssh-7.6p1/log.c	2018-01-05 12:00:11.000000000 +0100
@@ -52,6 +52,8 @@
 
 #include "log.h"
 
+#define NETGEAR_SYSLOG(fmt, args...)    syslog(LOG_INFO, fmt, ##args)
+
 static LogLevel log_level = SYSLOG_LEVEL_INFO;
 static int log_on_stderr = 1;
 static int log_stderr_fd = STDERR_FILENO;
@@ -400,7 +402,7 @@
 	char msgbuf[MSGBUFSIZ];
 	char fmtbuf[MSGBUFSIZ];
 	char *txt = NULL;
-	int pri = LOG_INFO;
+	/* int pri = LOG_INFO; */
 	int saved_errno = errno;
 	log_handler_fn *tmp_handler;
 
@@ -411,34 +413,34 @@
 	case SYSLOG_LEVEL_FATAL:
 		if (!log_on_stderr)
 			txt = "fatal";
-		pri = LOG_CRIT;
+		/* pri = LOG_CRIT; */
 		break;
 	case SYSLOG_LEVEL_ERROR:
 		if (!log_on_stderr)
 			txt = "error";
-		pri = LOG_ERR;
+		/* pri = LOG_ERR; */
 		break;
 	case SYSLOG_LEVEL_INFO:
-		pri = LOG_INFO;
+		/* pri = LOG_INFO; */
 		break;
 	case SYSLOG_LEVEL_VERBOSE:
-		pri = LOG_INFO;
+		/* pri = LOG_INFO; */
 		break;
 	case SYSLOG_LEVEL_DEBUG1:
 		txt = "debug1";
-		pri = LOG_DEBUG;
+		/* pri = LOG_DEBUG; */
 		break;
 	case SYSLOG_LEVEL_DEBUG2:
 		txt = "debug2";
-		pri = LOG_DEBUG;
+		/* pri = LOG_DEBUG; */
 		break;
 	case SYSLOG_LEVEL_DEBUG3:
 		txt = "debug3";
-		pri = LOG_DEBUG;
+		/* pri = LOG_DEBUG; */
 		break;
 	default:
 		txt = "internal error";
-		pri = LOG_ERR;
+		/*pri = LOG_ERR;*/
 		break;
 	}
 	if (txt != NULL && log_handler == NULL) {
@@ -462,11 +464,13 @@
 	} else {
 #if defined(HAVE_OPENLOG_R) && defined(SYSLOG_DATA_INIT)
 		openlog_r(argv0 ? argv0 : __progname, LOG_PID, log_facility, &sdata);
-		syslog_r(pri, &sdata, "%.500s", fmtbuf);
+		/* syslog_r(pri, &sdata, "%.500s", fmtbuf);*/
+		syslog_r(LOG_INFO, &sdata, "%.500s", fmtbuf);
 		closelog_r(&sdata);
 #else
 		openlog(argv0 ? argv0 : __progname, LOG_PID, log_facility);
-		syslog(pri, "%.500s", fmtbuf);
+		/* syslog(pri, "%.500s", fmtbuf); */
+		NETGEAR_SYSLOG("10 00[SSHD] %.500s", fmtbuf);
 		closelog();
 #endif
 	}
