--- work/linux-atm-2.5.2/Makefile
+++ work/linux-atm-2.5.2/Makefile
@@ -0,0 +1,69 @@
+include ../Rules.mak
+
+BINNAME			:= atmarp
+LIBSNAME		:= libatm.so
+LIBSMAINVER		:= 1
+LIBSVER			:= $(LIBSMAINVER).0.0
+LIBSDIRNAME		:= $(ATM)/src/lib/.libs
+BINSDIRNAME		:= $(ATM)/src/arpd/.libs
+LIBS			:= $(LIBSNAME).$(LIBSVER)
+BINS			:= $(BINSDIRNAME)/$(BINNAME) $(BINSDIRNAME)/$(BINNAME)d
+STARTUP			:= rc_wan
+
+all: config
+	$(MAKE) -C $(ATM)
+	$(STRIP) $(STRIPFLAGS) $(LIBSDIRNAME)/*.a
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(ATM) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) CFLAGS="$(CFLAGS) -Os -s" --with-kernel-headers=$(KERNEL_INC)
+
+install: control
+	$(STRIP) $(LIBSDIRNAME)/$(LIBS)
+	$(SSTRIP) $(BINS)
+	for D in $(TMPINSTDIR); do install -m 0755 -d $$D; done
+	install -m 0755 $(LIBSDIRNAME)/$(LIBS) $(BINS) $(TMPINSTDIR)
+	for L in $(LIBSNAME) $(LIBSNAME).$(LIBSMAINVER); do ln -sf $(LIBS) $(TMPINSTDIR)/$$L; done
+
+clean:
+	$(MAKE) -C $(ATM) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(ATM) distclean
+
+ifndef BUILD
+control:
+else
+PREINST			:= $(CTRL_DIR)/preinst
+POSTINST		:= $(CTRL_DIR)/postinst
+
+control: preinst postinst
+
+preinst:
+	@echo "#!/bin/sh" > $(PREINST)
+	@echo "echo \"Kill active services...\"" >> $(PREINST)
+	@echo "for B in $(BINNAME) $(BINNAME)d; do killall -9 \$$B >/dev/null 2>&1; sleep 3; done" >> $(PREINST)
+	@echo "echo \"Removing old atm lib & bins...\"" >> $(PREINST)
+	@echo "rm -f $(LDLIB)/$(LIBSNAME)*" >> $(PREINST)
+	@echo "sync" >> $(PREINST)
+	@echo "for B in $(BINNAME) $(BINNAME)d; do [ -f $(USRSBIN)/\$$B ] && rm -f $(USRSBIN)/\$$B && sync; done" >> $(PREINST)
+	@echo "exit 0" >> $(PREINST)
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "echo \"Updating atm libs & bins ...\"" >> $(POSTINST)
+	@echo "mv -f $(TMPDIR)/$(LIBSNAME)* $(LDLIB)/" >> $(POSTINST)
+	@echo "sync" >> $(POSTINST)
+	@echo "for B in $(BINNAME) $(BINNAME)d; do mv -f $(TMPDIR)/\$$B $(USRSBIN)/ && sync; done" >> $(POSTINST)
+	@echo "echo" >> $(POSTINST)
+	@echo "echo \"Rejoin wan connection or reboot...\"" >> $(POSTINST)
+	@echo "echo" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+endif
+
