--- work/rp-pppoe-3.15/Makefile
+++ work/rp-pppoe-3.15/Makefile
@@ -0,0 +1,57 @@
+include ../Rules.mak
+
+BINNAME			:= pppoe
+BINS			:= src/$(BINNAME)
+STARTUP			:= rc_wan
+
+all: config
+	$(MAKE) -C src
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd src && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) CFLAGS="$(CFLAGS) -Os -s" --disable-debugging
+
+install: control
+	$(SSTRIP) $(BINS)
+	install -m 0755 -d $(TMPINSTDIR)
+	install -m 0755 $(BINS) $(TMPINSTDIR)
+	ln -sf $(BINNAME) $(TMPINSTDIR)/pppoe-relay
+
+clean:
+	$(MAKE) -C src clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C src distclean
+
+ifndef BUILD
+control:
+else
+PREINST			:= $(CTRL_DIR)/preinst
+POSTINST		:= $(CTRL_DIR)/postinst
+
+control: preinst postinst
+
+preinst:
+	@echo "#!/bin/sh" > $(PREINST)
+	@echo "echo \"Kill active services...\"" >> $(PREINST)
+	@echo "for B in $(BINNAME) $(BINNAME)-relay; do killall -9 \$$B >/dev/null 2>&1; sleep 3; done" >> $(PREINST)
+	@echo "echo \"Removing old $(BINNAME)...\"" >> $(PREINST)
+	@echo "for B in $(BINNAME) $(BINNAME)-relay; do [ -f $(USRSBIN)/\$$B ] && rm -f $(USRSBIN)/\$$B && sync; done" >> $(PREINST)
+	@echo "exit 0" >> $(PREINST)
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "echo \"Updating $(BINNAME) ...\"" >> $(POSTINST)
+	@echo "for B in $(BINNAME) $(BINNAME)-relay; do mv -f $(TMPDIR)/\$$B $(USRSBIN)/; sync; done" >> $(POSTINST)
+	@echo "echo" >> $(POSTINST)
+	@echo "echo \"Rejoin wan connection or reboot...\"" >> $(POSTINST)
+	@echo "echo" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+endif
+
