--- work/rp-pppoe-3.15/src/Makefile.in
+++ work/rp-pppoe-3.15/src/Makefile.in
@@ -61,13 +61,11 @@ PATHS='-DPPPOE_PATH="$(PPPOE_PATH)"' '-D
 	'-DPPPOE_SERVER_OPTIONS="$(PPPOESERVER_PPPD_OPTIONS)"'
 
 CFLAGS= @CFLAGS@ $(LIC_INCDIR) $(DEFINES) $(LIC_DEFINE) $(PATHS) -Ilibevent
-TARGETS=@TARGETS@
+TARGETS=pppoe
 PPPOE_SERVER_LIBS=$(LIC_LIBDIR) $(LIC_LIB)
 LDFLAGS= @LDFLAGS@
 
 all: $(TARGETS)
-	@echo ""
-	@echo "Type 'make install' as root to install the software."
 
 pppoe-sniff: pppoe-sniff.o if.o common.o debug.o
 	@CC@ -o $@ $^ $(LDFLAGS)
@@ -80,7 +78,7 @@ pppoe-server: pppoe-server.o if.o debug.
 #pppoe-bridge: pppoe-bridge.o ppp.o modem.o rs232.o if.o debug.o common.o md5.o libevent/libevent.a @PPPOE_SERVER_DEPS@
 #	@CC@ -o $@ @RDYNAMIC@ $^ $(LDFLAGS) $(PPPOE_SERVER_LIBS) -Llibevent -levent
 
-pppoe: pppoe.o if.o debug.o common.o ppp.o discovery.o
+pppoe: pppoe.o if.o common.o ppp.o discovery.o relay.o
 	@CC@ -o $@ $^ $(LDFLAGS)
 
 pppoe-relay: relay.o if.o debug.o common.o
