diff -urN work/nvram-0.1.0/Makefile work/nvram-0.1.0/Makefile
--- work/nvram-0.1.0/Makefile	2017-05-02 12:33:28.031781668 +0200
+++ work/nvram-0.1.0/Makefile	2017-04-25 23:37:09.822096143 +0200
@@ -1,35 +1,45 @@
-include ../../Rules.mak
-CC=$(CROSS)gcc
+include ../Rules.mak
 
-CFLAGS += -Os -s -Wall
-CFLAGS += -Werror
-CFLAGS += -DTEST
+LIBS = libscnvram.so
+
+DEFS +=
+CFLAGS += -W -Wall -Werror
 
 ifeq ($(BOOT_NAND_FLASH),1)
-CFLAGS += -DBOOT_NAND_FLASH
+DEFS += -DBOOT_NAND_FLASH
 endif
 
-DESTLIB = ../../target/lib
-DESTBIN = ../../target/usr/sbin
+ifeq ($(TR069),1)
+DEFS += -DTR069
+endif
 
-BINS = libscnvram.so 
+ifeq ($(LOCAL),1)
+DEFS += -DLOCAL=1 -DLROOT=\"$(PREFIX)\"
+else
+CFLAGS += -Os -s
+endif
 
-all: $(BINS)
+ifeq ($(DEBUG),1)
+DEFS += -DTEST
+endif
 
-libscnvram.so : nvram.o nvram_lock.o
-	$(CC) $(CFLAGS) -shared -Wl,-soname,libscnvram.so -o libscnvram.so $^ 
-	$(STRIP) libscnvram.so
+all: $(LIBS)
 
-nvram.o : nvram.c
-	$(CC) $(CFLAGS) -fPIC -c $^ -o $@
+libscnvram.so: nvram.o nvram_lock.o
+	$(CC) $(CFLAGS) $(DEFS) -shared -Wl,-soname,$@ $^ -o $@ 
+	$(STRIP) $@
+
+nvram.o: nvram.c
+	$(CC) $(CFLAGS) $(DEFS) -c -fPIC $^ -o $@
+	$(STRIP) $(STRIPFLAGS) $@
 
 nvram_lock.o: nvram_lock.c
-	$(CC) $(CFLAGS) -fPIC -c $^ -o $@
+	$(CC) $(CFLAGS) $(DEFS) -c -fPIC $^ -o $@
+	$(STRIP) $(STRIPFLAGS) $@
 
-install :
-	cp -f libscnvram.so $(DESTLIB)
+install:
+	@echo "Nothing to install: install will be made by $(CORE_WORK) ."
 
 clean:
-	rm -rf *~ *.o *.so nvram $(BINS)
-	-rm -rf $(DESTLIB)/libscnvram.so
+	rm -rf *~ *.o *.so $(LIBS)
 
