diff -urN D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/lib/Makefile D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/lib/Makefile
--- D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/lib/Makefile	2016-10-11 17:57:02.000000000 +0200
+++ D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/lib/Makefile	2017-05-04 12:48:13.614863946 +0200
@@ -11,13 +11,13 @@
 
 CFGL=	 -L./lib -llsof
 
-DEBUG=	-O
+DEBUG=	-Os
 
-AR=	ar cr ${LIB} ${OBJ}
+AR=	$(CROSS)ar cr ${LIB} ${OBJ}
 
-RANLIB=	ranlib ${LIB}
+RANLIB=	$(CROSS)ranlib ${LIB}
 
-CFLAGS=	${CDEFS} ${INCL} ${DEBUG}
+CFLAGS += ${CDEFS} ${INCL} ${DEBUG}
 
 # Lsof library Makefile skeleton
 #
diff -urN D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/Makefile D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/Makefile
--- D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/Makefile	2016-10-11 17:57:02.000000000 +0200
+++ D7000_V1.0.1.80_WW_src/Source/apps/lsof_4.82_src/Makefile	2017-05-01 22:04:23.156997441 +0200
@@ -11,7 +11,7 @@
 
 CFGL=	 -L./lib -llsof
 
-DEBUG=	-O
+DEBUG=	-Os
 
 # Linux /proc-based Makefile
 #
@@ -32,7 +32,7 @@
 CDEFS=  ${CDEF} ${CFGF}
 DEP=	${CFGD} ${CFGDN}
 INCL=	${DINC}
-CFLAGS=	${CDEFS} ${INCL} ${DEP} ${DEBUG}
+CFLAGS += ${CDEFS} ${INCL} ${DEP} ${DEBUG}
 
 GRP=
 
@@ -66,7 +66,7 @@
 	(cd lib; ${MAKE} -f Makefile.skel clean)
 
 install: all FRC
-	$(STRIP) lsof
+	$(SSTRIP) lsof
 	cp -f lsof ../../target/usr/sbin
 
 ${LIB}: FRC
