diff -urN D7000_V1.0.1.74_WW_src/Source/apps/syslogd/Makefile D7000_V1.0.1.74_WW_src/Source/apps/syslogd/Makefile
--- D7000_V1.0.1.74_WW_src/Source/apps/syslogd/Makefile	2016-10-11 17:56:25.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/syslogd/Makefile	2017-05-01 21:49:32.193798601 +0200
@@ -3,7 +3,7 @@
 CC = $(CROSS)gcc
 STRIP=$(CROSS)strip
 LDFLAGS=
-CFLAGS += -Os -Wall -s -Werror -Wno-unused-but-set-variable
+CFLAGS +=  -Os -Wall -s -Werror -Wno-unused-but-set-variable
 BIN =syslogd klogd
 
 NETGEAR_LOG=1
@@ -33,16 +33,15 @@
 
 syslogd:$(LIBS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(LIBS)
-	$(STRIP) $@
+	$(SSTRIP) $@
 
 klogd:$(LIBS_klog)
 	$(CC) $(CFLAGS) -o $@ $(LIBS_klog)
-	$(STRIP) $@
+	$(SSTRIP) $@
 
 install:
-	#rm -f klogd
-	#ln -s syslogd klogd
 	cp -a syslogd klogd ../../target/sbin
 clean:
 	rm -f $(BIN) 
 	rm -f ../../target/sbin/syslogd  ../../target/sbin/klogd
+
