diff -urN work/openvpn-2.5.2/openvpn-2.5.2/src/openvpn/ssl_mbedtls.c work/openvpn-2.5.2/openvpn-2.5.2/src/openvpn/ssl_mbedtls.c
--- work/openvpn-2.5.2/openvpn-2.5.2/src/openvpn/ssl_mbedtls.c
+++ work/openvpn-2.5.2/openvpn-2.5.2/src/openvpn/ssl_mbedtls.c
@@ -1098,10 +1098,13 @@ key_state_ssl_init(struct key_state_ssl
     {
         mbedtls_ssl_conf_curves(ks_ssl->ssl_config, ssl_ctx->groups);
     }
-    /* Disable TLS renegotiations. OpenVPN's renegotiation creates new SSL
-     * session and does not depend on this feature. And TLS renegotiations have
-     * been problematic in the past */
+
+    /* Disable TLS renegotiations if the mbedtls library supports that feature.
+     * OpenVPN's renegotiation creates new SSL sessions and does not depend on
+     * this feature and TLS renegotiations have been problematic in the past. */
+#if defined(MBEDTLS_SSL_RENEGOTIATION)
     mbedtls_ssl_conf_renegotiation(ks_ssl->ssl_config, MBEDTLS_SSL_RENEGOTIATION_DISABLED);
+#endif /* MBEDTLS_SSL_RENEGOTIATION */
 
     /* Disable record splitting (for now).  OpenVPN assumes records are sent
      * unfragmented, and changing that will require thorough review and
