diff -urN work/busybox-1.29.3/include/usage.src.h work/busybox-1.29.3/include/usage.src.h
--- work/busybox-1.29.3/include/usage.src.h	2018-07-02 13:23:06.000000000 +0200
+++ work/busybox-1.29.3/include/usage.src.h	2018-10-31 21:56:14.837218353 +0100
@@ -28,3 +28,12 @@
        "Hello world!\n"
 
 #endif
+     
+#define sendarp_trivial_usage \
+       "-s <src device name> -d <dst device name>"
+#define sendarp_full_usage "\n\n" \
+       "Send ARP packet with src device's IP and Hardware address to dst device.\n" \
+     "\nOptions:" \
+     "\n	-s DEVNAME	src device name." \
+     "\n	-d DEVNAME	dst device name." \
+
diff -urN work/busybox-1.29.3/util-linux/chrt.c work/busybox-1.29.3/util-linux/chrt.c
--- work/busybox-1.29.3/util-linux/chrt.c	2018-07-02 13:23:06.000000000 +0200
+++ work/busybox-1.29.3/util-linux/chrt.c	2018-10-31 22:15:36.748785153 +0100
@@ -131,12 +131,8 @@
 		pol = sched_getscheduler(pid);
 		if (pol < 0)
 			bb_perror_msg_and_die("can't %cet pid %d's policy", 'g', (int)pid);
-		printf("pid %d's %s scheduling policy: %s\n",
-				pid, current_new, policies[pol].name);
 		if (sched_getparam(pid, &sp))
 			bb_perror_msg_and_die("can't get pid %d's attributes", (int)pid);
-		printf("pid %d's %s scheduling priority: %d\n",
-				(int)pid, current_new, sp.sched_priority);
 		if (!*argv) {
 			/* Either it was just "-p <pid>",
 			 * or it was "-p <priority> <pid>" and we came here
