diff -urN work/zlib-1.2.12/zlib-1.2.12/Makefile.in work/zlib-1.2.12/zlib-1.2.12/Makefile.in
--- work/zlib-1.2.12/zlib-1.2.12/Makefile.in	2017-04-06 18:00:46.662292000 +0200
+++ work/zlib-1.2.12/zlib-1.2.12/Makefile.in	2017-04-06 18:01:08.566300143 +0200
@@ -75,11 +75,11 @@
 
 all: static shared
 
-static: example$(EXE) minigzip$(EXE)
+static: $(STATICLIB) #example$(EXE) minigzip$(EXE)
 
-shared: examplesh$(EXE) minigzipsh$(EXE)
+shared: $(SHAREDLIBV) #examplesh$(EXE) minigzipsh$(EXE)
 
-all64: example64$(EXE) minigzip64$(EXE)
+all64: #example64$(EXE) minigzip64$(EXE)
 
 check: test
 
