diff -urN work/opkg-0.3.6/opkg-0.3.6/Makefile.am work/opkg-0.3.6/opkg-0.3.6/Makefile.am
--- work/opkg-0.3.6/opkg-0.3.6/Makefile.am	2016-12-22 18:39:20.000000000 +0100
+++ work/opkg-0.3.6/opkg-0.3.6/Makefile.am	2017-03-23 21:24:19.232029164 +0100
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = libopkg src utils man
+SUBDIRS = libopkg src #utils man
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libopkg.pc
