diff -urN work/iproute2-4.16.0/ip/ipxfrm.c work/iproute2-4.16.0/ip/ipxfrm.c
--- work/iproute2-4.16.0/ip/ipxfrm.c	2018-04-02 19:06:08.000000000 +0200
+++ work/iproute2-4.16.0/ip/ipxfrm.c	2018-04-20 21:38:38.597470630 +0200
@@ -454,7 +454,6 @@
 	switch (sel->proto) {
 	case IPPROTO_TCP:
 	case IPPROTO_UDP:
-	case IPPROTO_SCTP:
 	case IPPROTO_DCCP:
 	default: /* XXX */
 		if (sel->sport_mask)
@@ -1329,7 +1328,6 @@
 		switch (sel->proto) {
 		case IPPROTO_TCP:
 		case IPPROTO_UDP:
-		case IPPROTO_SCTP:
 		case IPPROTO_DCCP:
 		case IPPROTO_IP: /* to allow shared SA for different protocols */
 			break;
diff -urN work/iproute2-4.16.0/ip/xfrm_policy.c work/iproute2-4.16.0/ip/xfrm_policy.c
--- work/iproute2-4.16.0/ip/xfrm_policy.c	2018-04-02 19:06:08.000000000 +0200
+++ work/iproute2-4.16.0/ip/xfrm_policy.c	2018-04-20 21:40:50.429080215 +0200
@@ -68,7 +68,6 @@
 	fprintf(stderr, "UPSPEC := proto { { ");
 	fprintf(stderr, "%s | ", strxf_proto(IPPROTO_TCP));
 	fprintf(stderr, "%s | ", strxf_proto(IPPROTO_UDP));
-	fprintf(stderr, "%s | ", strxf_proto(IPPROTO_SCTP));
 	fprintf(stderr, "%s", strxf_proto(IPPROTO_DCCP));
 	fprintf(stderr, " } [ sport PORT ] [ dport PORT ] |\n");
 	fprintf(stderr, "                  { ");
diff -urN work/iproute2-4.16.0/ip/xfrm_state.c work/iproute2-4.16.0/ip/xfrm_state.c
--- work/iproute2-4.16.0/ip/xfrm_state.c	2018-04-02 19:06:08.000000000 +0200
+++ work/iproute2-4.16.0/ip/xfrm_state.c	2018-04-20 21:40:48.273086640 +0200
@@ -95,7 +95,6 @@
 	fprintf(stderr, "UPSPEC := proto { { ");
 	fprintf(stderr, "%s | ", strxf_proto(IPPROTO_TCP));
 	fprintf(stderr, "%s | ", strxf_proto(IPPROTO_UDP));
-	fprintf(stderr, "%s | ", strxf_proto(IPPROTO_SCTP));
 	fprintf(stderr, "%s", strxf_proto(IPPROTO_DCCP));
 	fprintf(stderr, " } [ sport PORT ] [ dport PORT ] |\n");
 	fprintf(stderr, "                  { ");
diff -urN work/iproute2-4.16.0/tc/f_flower.c work/iproute2-4.16.0/tc/f_flower.c
--- work/iproute2-4.16.0/tc/f_flower.c	2018-04-02 19:06:08.000000000 +0200
+++ work/iproute2-4.16.0/tc/f_flower.c	2018-04-20 21:41:47.220910562 +0200
@@ -52,7 +52,7 @@
 		"                       vlan_ethtype [ ipv4 | ipv6 | ETH-TYPE ] |\n"
 		"                       dst_mac MASKED-LLADDR |\n"
 		"                       src_mac MASKED-LLADDR |\n"
-		"                       ip_proto [tcp | udp | sctp | icmp | icmpv6 | IP-PROTO ] |\n"
+		"                       ip_proto [tcp | udp | icmp | icmpv6 | IP-PROTO ] |\n"
 		"                       ip_tos MASKED-IP_TOS |\n"
 		"                       ip_ttl MASKED-IP_TTL |\n"
 		"                       mpls_label LABEL |\n"
@@ -213,8 +213,6 @@
 		ip_proto = IPPROTO_TCP;
 	} else if (matches(str, "udp") == 0) {
 		ip_proto = IPPROTO_UDP;
-	} else if (matches(str, "sctp") == 0) {
-		ip_proto = IPPROTO_SCTP;
 	} else if (matches(str, "icmp") == 0) {
 		if (eth_type != htons(ETH_P_IP))
 			goto err;
@@ -452,10 +450,6 @@
 		return endpoint == FLOWER_ENDPOINT_SRC ?
 			TCA_FLOWER_KEY_UDP_SRC :
 			TCA_FLOWER_KEY_UDP_DST;
-	else if (ip_proto == IPPROTO_SCTP)
-		return endpoint == FLOWER_ENDPOINT_SRC ?
-			TCA_FLOWER_KEY_SCTP_SRC :
-			TCA_FLOWER_KEY_SCTP_DST;
 	else
 		return -1;
 }
@@ -1104,8 +1098,6 @@
 		sprintf(out, "tcp");
 	else if (ip_proto == IPPROTO_UDP)
 		sprintf(out, "udp");
-	else if (ip_proto == IPPROTO_SCTP)
-		sprintf(out, "sctp");
 	else if (ip_proto == IPPROTO_ICMP)
 		sprintf(out, "icmp");
 	else if (ip_proto == IPPROTO_ICMPV6)
