diff -urN work/flac-1.3.3/flac-1.3.3/include/share/compat.h work/flac-1.3.3/flac-1.3.3/include/share/compat.h
--- work/flac-1.3.3/flac-1.3.3/include/share/compat.h
+++ work/flac-1.3.3/flac-1.3.3/include/share/compat.h
@@ -112,9 +112,13 @@
 #include <sys/utime.h> /* for utime() */
 #endif
 #else
+#if _POSIX_C_SOURCE >= 200809L
+#include <fcntl.h>
+#else
 #include <sys/types.h> /* some flavors of BSD (like OS X) require this to get time_t */
 #include <utime.h> /* for utime() */
 #endif
+#endif
 
 #if defined _MSC_VER
 #  if _MSC_VER >= 1800
@@ -160,11 +164,15 @@
 
 #define flac_fopen fopen
 #define flac_chmod chmod
-#define flac_utime utime
 #define flac_unlink unlink
 #define flac_rename rename
 #define flac_stat stat
 
+#if _POSIX_C_SOURCE >= 200809L
+#define flac_utime(a, b) utimensat (AT_FDCWD, a, *b, 0)
+#else
+#define flac_utime utime
+#endif
 #endif
 
 #ifdef _WIN32
diff -urN work/flac-1.3.3/flac-1.3.3/src/libFLAC/metadata_iterators.c work/flac-1.3.3/flac-1.3.3/src/libFLAC/metadata_iterators.c
--- work/flac-1.3.3/flac-1.3.3/src/libFLAC/metadata_iterators.c
+++ work/flac-1.3.3/flac-1.3.3/src/libFLAC/metadata_iterators.c
@@ -3422,13 +3422,18 @@ FLAC__bool get_file_stats_(const char *filename, struct flac_stat_s *stats)
 
 void set_file_stats_(const char *filename, struct flac_stat_s *stats)
 {
-	struct utimbuf srctime;
-
 	FLAC__ASSERT(0 != filename);
 	FLAC__ASSERT(0 != stats);
 
+#if _POSIX_C_SOURCE >= 200809L
+	struct timespec srctime[2] = {};
+	srctime[0].tv_sec = stats->st_atime;
+	srctime[1].tv_sec = stats->st_mtime;
+#else
+	struct utimbuf srctime;
 	srctime.actime = stats->st_atime;
 	srctime.modtime = stats->st_mtime;
+#endif
 	(void)flac_chmod(filename, stats->st_mode);
 	(void)flac_utime(filename, &srctime);
 #if !defined _MSC_VER && !defined __BORLANDC__ && !defined __MINGW32__
diff -urN work/flac-1.3.3/flac-1.3.3/src/share/grabbag/file.c work/flac-1.3.3/flac-1.3.3/src/share/grabbag/file.c
--- work/flac-1.3.3/flac-1.3.3/src/share/grabbag/file.c
+++ work/flac-1.3.3/flac-1.3.3/src/share/grabbag/file.c
@@ -27,7 +27,6 @@
 #include <fcntl.h> /* for _O_BINARY */
 #else
 #include <sys/types.h> /* some flavors of BSD (like OS X) require this to get time_t */
-#include <utime.h> /* for utime() */
 #endif
 #if defined __EMX__
 #include <io.h> /* for setmode(), O_BINARY */
@@ -53,11 +52,17 @@
 void grabbag__file_copy_metadata(const char *srcpath, const char *destpath)
 {
 	struct flac_stat_s srcstat;
-	struct utimbuf srctime;
 
 	if(0 == flac_stat(srcpath, &srcstat)) {
+#if _POSIX_C_SOURCE >= 200809L
+		struct timespec srctime[2] = {};
+		srctime[0].tv_sec = srcstat.st_atime;
+		srctime[1].tv_sec = srcstat.st_mtime;
+#else
+		struct utimbuf srctime;
 		srctime.actime = srcstat.st_atime;
 		srctime.modtime = srcstat.st_mtime;
+#endif
 		(void)flac_chmod(destpath, srcstat.st_mode);
 		(void)flac_utime(destpath, &srctime);
 	}
