--- work/sqlite-src-3360000/sqlite-src-3360000/Makefile.in	2018-10-23 23:00:46.662292000 +0200
+++ work/sqlite-src-3360000/sqlite-src-3360000/Makefile.in	2018-10-23 23:01:08.566300143 +0200
@@ -633,7 +633,7 @@
 # This is the default Makefile target.  The objects listed here
 # are what get build when you type just "make" with no arguments.
 #
-all:	sqlite3.h libsqlite3.la sqlite3$(TEXE) $(HAVE_TCL:1=libtclsqlite3.la)
+all:	sqlite3.h libsqlite3.la $(HAVE_TCL:1=libtclsqlite3.la)
 
 Makefile: $(TOP)/Makefile.in
 	./config.status
