diff -urN ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux
--- ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux	2016-10-11 17:57:04.000000000 +0200
+++ ppp-2.4.1.pppoe4.orig/pppd/Makefile.linux	2017-01-13 21:46:49.819090391 +0100
@@ -31,9 +31,9 @@
 COPTS = -Os -pipe -Wall
 LIBS =
 
-ifneq ($(wildcard /usr/lib/libcrypt.*),)
+#ifneq ($(wildcard /usr/lib/libcrypt.*),)
 LIBS += -lcrypt
-endif
+#endif
 
 # Uncomment the next 2 lines to include support for Microsoft's
 # MS-CHAP authentication protocol.
@@ -150,7 +150,7 @@
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
-	$(CC)  $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS) -Lplugins/pppoe -lpppoe -L../../linux-atm-2.4.0/src/lib/.libs -latm -Lplugins -lpppoatm
+	$(CC)  $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS) -Lplugins/pppoe -lpppoe -L../../$(ATMDIR)/src/lib/.libs -latm -Lplugins -lpppoatm
 
 clean:
 	rm -f $(PPPDOBJS) pppd *~ #* core
