diff -urN work/iproute2-4.20.0/Makefile work/iproute2-4.20.0/Makefile
--- work/iproute2-4.20.0/Makefile	2018-08-13 21:11:32.000000000 +0200
+++ work/iproute2-4.20.0/Makefile	2018-10-06 23:26:15.681053749 +0200
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 # Top level Makefile for iproute2
+include ../Rules.mak
 
 ifeq ("$(origin V)", "command line")
 VERBOSE = $(V)
@@ -23,7 +24,7 @@
 DOCDIR?=$(DATADIR)/doc/iproute2
 MANDIR?=$(DATADIR)/man
 ARPDDIR?=/var/lib/arpd
-KERNEL_INCLUDE?=/usr/include
+KERNEL_INCLUDE=$(KERNEL_INC)
 BASH_COMPDIR?=$(DATADIR)/bash-completion/completions
 
 # Path to db_185.h include
@@ -31,12 +32,12 @@
 
 SHARED_LIBS = y
 
-DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
+DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LDLIB)\"
 ifneq ($(SHARED_LIBS),y)
 DEFINES+= -DNO_SHARED_LIBS
 endif
 
-DEFINES+=-DCONFDIR=\"$(CONFDIR)\" \
+DEFINES+=-DCONFDIR=\"$(TMPETC)/iproute2\" \
          -DNETNS_RUN_DIR=\"$(NETNS_RUN_DIR)\" \
          -DNETNS_ETC_DIR=\"$(NETNS_ETC_DIR)\"
 
@@ -49,19 +50,19 @@
 #options for mpls
 ADDLIB+=mpls_ntop.o mpls_pton.o
 
-CC := gcc
+CC := $(CROSS)gcc
 HOSTCC ?= $(CC)
-DEFINES += -D_GNU_SOURCE
+DEFINES += -D_GNU_SOURCE -D_SC_BUILD_
 # Turn on transparent support for LFS
 DEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CCOPTS = -O2
+CCOPTS = -Os
 WFLAGS := -Wall -Wstrict-prototypes  -Wmissing-prototypes
 WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
 
-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include -I../include/uapi $(DEFINES) $(CFLAGS)
+CFLAGS := $(WFLAGS) $(CCOPTS) -I../include -I../include/uapi $(DEFINES) $(CFLAGS) -fcommon
 YACCFLAGS = -d -t -v
 
-SUBDIRS=lib ip tc bridge misc netem genl tipc devlink rdma man
+SUBDIRS=lib ip tc
 
 LIBNETLINK=../lib/libutil.a ../lib/libnetlink.a
 LDLIBS += $(LIBNETLINK)
@@ -69,7 +70,7 @@
 all: config.mk
 	@set -e; \
 	for i in $(SUBDIRS); \
-	do echo; echo $$i; $(MAKE) $(MFLAGS) -C $$i; done
+	do $(MAKE) $(MFLAGS) -C $$i; done
 
 help:
 	@echo "Make Targets:"
@@ -87,22 +88,14 @@
 config.mk:
 	sh configure $(KERNEL_INCLUDE)
 
-install: all
-	install -m 0755 -d $(DESTDIR)$(SBINDIR)
-	install -m 0755 -d $(DESTDIR)$(CONFDIR)
-	install -m 0755 -d $(DESTDIR)$(ARPDDIR)
-	install -m 0755 -d $(DESTDIR)$(HDRDIR)
-	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples
-	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples/diffserv
-	install -m 0644 README.iproute2+tc $(shell find examples -maxdepth 1 -type f) \
-		$(DESTDIR)$(DOCDIR)/examples
-	install -m 0644 $(shell find examples/diffserv -maxdepth 1 -type f) \
-		$(DESTDIR)$(DOCDIR)/examples/diffserv
-	@for i in $(SUBDIRS);  do $(MAKE) -C $$i install; done
-	install -m 0644 $(shell find etc/iproute2 -maxdepth 1 -type f) $(DESTDIR)$(CONFDIR)
-	install -m 0755 -d $(DESTDIR)$(BASH_COMPDIR)
-	install -m 0644 bash-completion/tc $(DESTDIR)$(BASH_COMPDIR)
-	install -m 0644 include/bpf_elf.h $(DESTDIR)$(HDRDIR)
+install: control
+	$(SSTRIP) tc/tc ip/ip
+	if [ -e $(ETCDIR)/iproute2 ]; then \
+	rm -rf $(ETCDIR)/iproute2;\
+	fi
+	for D in $(ETCDIR) $(BINDIR); do install -m 0755 -d $$D; done
+	install -m 0755 tc/tc ip/ip $(BINDIR)
+	cp -rf etc/iproute2 $(ETCDIR)/iproute2
 
 snapshot:
 	echo "static const char SNAPSHOT[] = \""`date +%y%m%d`"\";" \
@@ -125,4 +118,17 @@
 cscope:
 	cscope -b -q -R -Iinclude -sip -slib -smisc -snetem -stc
 
+ifndef BUILD
+control:
+else
+POSTINST		:= $(CTRL_DIR)/postinst
+
+control: postinst
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "cp -rf $(USRETC)/iproute2 $(TMPETC)/iproute2" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+endif
+
 .EXPORT_ALL_VARIABLES:
