diff -urN work/iproute2-4.20.0/configure work/iproute2-4.20.0/configure
--- work/iproute2-4.20.0/configure	2018-04-02 19:06:08.000000000 +0200
+++ work/iproute2-4.20.0/configure	2018-04-20 22:46:49.538797017 +0200
@@ -231,8 +231,9 @@
 	echo "HAVE_ELF:=y" >>$CONFIG
 	echo "yes"
 
-	echo 'CFLAGS += -DHAVE_ELF' `${PKG_CONFIG} libelf --cflags` >> $CONFIG
-	echo 'LDLIBS += ' `${PKG_CONFIG} libelf --libs` >>$CONFIG
+	echo 'CFLAGS += -DHAVE_ELF' >> $CONFIG
+	echo 'ELF_CFLAGS += ' `${PKG_CONFIG} libelf --cflags` >> $CONFIG
+	echo 'ELF_LDLIBS += ' `${PKG_CONFIG} libelf --libs` >>$CONFIG
     else
 	echo "no"
     fi
diff -urN work/iproute2-4.20.0/lib/Makefile work/iproute2-4.20.0/lib/Makefile
--- work/iproute2-4.20.0/lib/Makefile	2018-08-13 21:11:32.000000000 +0200
+++ work/iproute2-4.20.0/lib/Makefile	2018-10-06 23:59:22.967035093 +0200
@@ -7,9 +7,17 @@
 	inet_proto.o namespace.o json_writer.o json_print.o \
 	names.o color.o bpf.o exec.o fs.o
 
+ELFOBJ=$(patsubst %.o,%.elf.o,$(UTILOBJ))
+
 NLOBJ=libgenl.o libnetlink.o
 
-all: libnetlink.a libutil.a
+all: libnetlink.a libutil.a libutil-elf.a
+
+%.o: %.c
+	$(QUIET_CC)$(CC) $(CFLAGS) -UHAVE_ELF $(EXTRA_CFLAGS) -c -o $@ $<
+
+%.elf.o: %.c
+	$(QUIET_CC)$(CC) $(CFLAGS) $(ELF_CFLAGS) $(EXTRA_CFLAGS) -c -o $@ $<
 
 libnetlink.a: $(NLOBJ)
 	$(QUIET_AR)$(AR) rcs $@ $^
@@ -17,7 +25,10 @@
 libutil.a: $(UTILOBJ) $(ADDLIB)
 	$(QUIET_AR)$(AR) rcs $@ $^
 
+libutil-elf.a: $(ELFOBJ) $(ADDLIB)
+	$(QUIET_AR)$(AR) rcs $@ $^
+
 install:
 
 clean:
-	rm -f $(NLOBJ) $(UTILOBJ) $(ADDLIB) libnetlink.a libutil.a
+	rm -f $(NLOBJ) $(UTILOBJ) $(ELFOBJ) $(ADDLIB) libnetlink.a libutil.a libutil-elf.a
diff -urN work/iproute2-4.20.0/tc/Makefile work/iproute2-4.20.0/tc/Makefile
--- work/iproute2-4.20.0/tc/Makefile	2018-08-13 21:11:32.000000000 +0200
+++ work/iproute2-4.20.0/tc/Makefile	2018-10-06 23:59:22.967035093 +0200
@@ -133,8 +133,8 @@
 
 all: tc $(TCSO)
 
-tc: $(TCOBJ) $(LIBNETLINK) libtc.a
-	$(QUIET_LINK)$(CC) $^ $(LDFLAGS) $(LDLIBS) -o $@
+tc: $(TCOBJ) $(subst libutil.a,libutil-elf.a,$(LIBNETLINK)) libtc.a
+	$(QUIET_LINK)$(CC) $^ $(LDFLAGS) $(LDLIBS) $(ELF_LDLIBS) -o $@
 
 libtc.a: $(TCLIB)
 	$(QUIET_AR)$(AR) rcs $@ $^
