diff -urN work/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini_snmpd.h work/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini_snmpd.h
--- work/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini_snmpd.h	1970-01-01 01:00:00.000000000 +0100
+++ work/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini-snmpd-3f0f4d93560a86358548c4f48459e9e9a8a4a6ca/mini_snmpd.h	2018-01-05 12:00:11.000000000 +0100
@@ -107,7 +107,7 @@
 	do {							\
 		if (g_verbose || (level != LOG_DEBUG)) {	\
 			if (g_daemon || g_syslog)		\
-				syslog(level, format);		\
+				syslog(LOG_INFO, "10 00[MINI-SNMPD]" format);	 \
 			else					\
 				fprintf(stderr, format);	\
 		}						\
