--- work/smartmontools-7.3/smartmontools-7.3/smartctl.cpp	2021-03-02 22:39:12.863609549 +0100
+++ work/smartmontools-7.3/smartmontools-7.3/smartctl.cpp	2021-03-02 22:38:08.720017000 +0100
@@ -1569,7 +1569,8 @@
   {
     time_t now = time(nullptr);
     dateandtimezoneepoch(startup_datetime_buf, now);
-    jglb["local_time"] += { {"time_t", now}, {"asctime", startup_datetime_buf} };
+    jglb["local_time"]["time_t"] = now;
+    jglb["local_time"]["asctime"] = startup_datetime_buf;
   }
 
   const char * name = argv[argc-1];
--- work/smartmontools-7.3/smartmontools-7.3/nvmeprint.cpp	2021-06-04 17:46:36.000000000 +0200
+++ work/smartmontools-7.3/smartmontools-7.3/nvmeprint.cpp	2021-10-11 23:15:23.932919017 +0200
@@ -202,7 +202,8 @@
   }
 
   // SMART/Health Information is mandatory
-  jglb["smart_support"] += { {"available", true}, {"enabled", true} };
+  jglb["smart_support"]["available"] = true;
+  jglb["smart_support"]["enabled"] = true;
 
   jout_startup_datetime("Local Time is:                      ");
 }
