diff -urN work/minidlna-1.3.0/minidlna-1.3.0/Makefile.am diff -urN work/minidlna-1.3.0/minidlna-1.3.0/Makefile.am
--- work/minidlna-1.3.0/minidlna-1.3.0/Makefile.am	2020-12-08 14:31:33.940160000 +0100
+++ work/minidlna-1.3.0/minidlna-1.3.0/Makefile.am	2020-12-08 17:51:42.555956686 +0100
@@ -17,9 +17,9 @@
 
 AM_CFLAGS = -Wall -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 
-SUBDIRS=po
+SUBDIRS=
 
-sbin_PROGRAMS = minidlnad
+sbin_PROGRAMS = minidlna
 check_PROGRAMS = testupnpdescgen
 minidlnad_SOURCES = minidlna.c upnphttp.c upnpdescgen.c upnpsoap.c \
 			upnpreplyparse.c minixml.c clients.c \
