diff -urN work/minidlna-1.3.0/minidlna-1.3.0/Makefile.in diff -urN work/minidlna-1.3.0/minidlna-1.3.0/Makefile.in
--- work/minidlna-1.3.0/minidlna-1.3.0/Makefile.in	2020-12-08 14:31:33.940160000 +0100
+++ work/minidlna-1.3.0/minidlna-1.3.0/Makefile.in	2020-12-08 17:51:42.555956686 +0100
@@ -107,7 +107,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-sbin_PROGRAMS = minidlnad$(EXEEXT)
+sbin_PROGRAMS = minidlna$(EXEEXT)
 check_PROGRAMS = testupnpdescgen$(EXEEXT)
 @HAVE_KQUEUE_TRUE@am__append_1 = kqueue.c monitor_kqueue.c
 @HAVE_KQUEUE_FALSE@am__append_2 = select.c
@@ -425,7 +425,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_CFLAGS = -Wall -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-SUBDIRS = po
+SUBDIRS = 
 minidlnad_SOURCES = minidlna.c upnphttp.c upnpdescgen.c upnpsoap.c \
 	upnpreplyparse.c minixml.c clients.c getifaddr.c process.c \
 	upnpglobalvars.c options.c minissdp.c uuid.c upnpevents.c \
@@ -574,8 +574,8 @@
 tagutils/tagutils.$(OBJEXT): tagutils/$(am__dirstamp) \
 	tagutils/$(DEPDIR)/$(am__dirstamp)
 
-minidlnad$(EXEEXT): $(minidlnad_OBJECTS) $(minidlnad_DEPENDENCIES) $(EXTRA_minidlnad_DEPENDENCIES) 
-	@rm -f minidlnad$(EXEEXT)
+minidlna$(EXEEXT): $(minidlnad_OBJECTS) $(minidlnad_DEPENDENCIES) $(EXTRA_minidlnad_DEPENDENCIES) 
+	@rm -f minidlna$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(minidlnad_OBJECTS) $(minidlnad_LDADD) $(LIBS)
 
 testupnpdescgen$(EXEEXT): $(testupnpdescgen_OBJECTS) $(testupnpdescgen_DEPENDENCIES) $(EXTRA_testupnpdescgen_DEPENDENCIES) 
