--- work/minidlna-1.3.0/minidlna-1.3.0/minidlna.c
+++ work/minidlna-1.3.0/minidlna-1.3.0/minidlna.c
@@ -418,6 +418,7 @@ rescan:
 		open_db(&db);
 		if (*scanner_pid == 0) /* child (scanner) process */
 		{
+			open_db(&db);
 			start_scanner();
 			sqlite3_close(db);
 			log_close();
@@ -433,6 +434,7 @@ rescan:
 			SETFLAG(SCANNING_MASK);
 #else
 		start_scanner();
+		sqlite3_close(db);
 #endif
 	}
 }
