--- work/minidlna-1.3.0/minidlna-1.3.0/clients.c
+++ work/minidlna-1.3.0/minidlna-1.3.0/clients.c
@@ -155,7 +155,7 @@ struct client_type_s client_types[] =

        /* USER-AGENT: Linux/2.6.35 UPnP/1.0 DLNADOC/1.50 INTEL_NMPR/2.0 LGE_DLNA_SDK/1.6.0 */
        { ELGNetCastDevice,
-         FLAG_DLNA | FLAG_CAPTION_RES,
+         FLAG_DLNA | FLAG_CAPTION_RES | FLAG_MIME_FLAC_FLAC,
          "LG",
          "LGE_DLNA_SDK/1.6.0",
          EUserAgent
