diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/check/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/check/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/check/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/check/Makefile	2017-05-04 23:02:58.455070499 +0200
@@ -194,7 +194,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dig/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dig/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dig/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dig/Makefile	2017-05-04 23:02:58.455070499 +0200
@@ -196,7 +196,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dnssec/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dnssec/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dnssec/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/dnssec/Makefile	2017-05-04 23:02:58.459070440 +0200
@@ -204,7 +204,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/Makefile	2017-05-04 23:02:58.475070202 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/Makefile	2017-05-04 23:02:58.459070440 +0200
@@ -229,7 +229,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/unix/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/unix/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/unix/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/named/unix/Makefile	2017-05-04 23:02:58.459070440 +0200
@@ -175,7 +175,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/nsupdate/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/nsupdate/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/nsupdate/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/nsupdate/Makefile	2017-05-04 23:02:58.459070440 +0200
@@ -198,7 +198,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/Makefile	2017-05-04 23:02:58.463070381 +0200
@@ -199,7 +199,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/unix/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/unix/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/unix/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/rndc/unix/Makefile	2017-05-04 23:02:58.463070381 +0200
@@ -175,7 +175,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/db/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/db/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/db/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/db/Makefile	2017-05-04 23:02:58.463070381 +0200
@@ -184,7 +184,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/dst/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/dst/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/dst/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/dst/Makefile	2017-05-04 23:02:58.463070381 +0200
@@ -184,7 +184,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/Makefile	2017-05-04 23:02:58.475070202 +0200
@@ -270,7 +270,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/master/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/master/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/master/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/master/Makefile	2017-05-04 23:02:58.467070321 +0200
@@ -185,7 +185,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/mem/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/mem/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/mem/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/mem/Makefile	2017-05-04 23:02:58.467070321 +0200
@@ -182,7 +182,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/names/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/names/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/names/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/names/Makefile	2017-05-04 23:02:58.467070321 +0200
@@ -185,7 +185,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) 
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/net/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/net/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/net/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/net/Makefile	2017-05-04 23:02:58.467070321 +0200
@@ -182,7 +182,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/rbt/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/rbt/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/rbt/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/rbt/Makefile	2017-05-04 23:02:58.467070321 +0200
@@ -185,7 +185,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/sockaddr/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/sockaddr/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/sockaddr/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/sockaddr/Makefile	2017-05-04 23:02:58.471070262 +0200
@@ -182,7 +182,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/lwresd/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/lwresd/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/lwresd/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/lwresd/Makefile	2017-05-04 23:02:58.471070262 +0200
@@ -186,7 +186,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/Makefile	2017-05-04 23:02:58.471070262 +0200
@@ -166,7 +166,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/tkey/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/tkey/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/tkey/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/system/tkey/Makefile	2017-05-04 23:02:58.471070262 +0200
@@ -187,7 +187,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/tasks/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/tasks/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/tasks/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/tasks/Makefile	2017-05-04 23:02:58.471070262 +0200
@@ -182,7 +182,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/timers/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/timers/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/timers/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/bin/tests/timers/Makefile	2017-05-04 23:02:58.475070202 +0200
@@ -182,7 +182,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/arm/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/arm/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/arm/Makefile	2016-10-11 17:56:08.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/arm/Makefile	2017-05-04 23:02:58.475070202 +0200
@@ -116,7 +116,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/Makefile	2016-10-11 17:56:09.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/Makefile	2017-05-04 23:02:58.479070144 +0200
@@ -123,7 +123,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/misc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/misc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/misc/Makefile	2016-10-11 17:56:09.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/doc/misc/Makefile	2017-05-04 23:02:58.479070144 +0200
@@ -116,7 +116,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/bsd/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/bsd/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/bsd/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/bsd/Makefile	2017-05-04 23:02:58.479070144 +0200
@@ -133,7 +133,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/dst/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/dst/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/dst/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/dst/Makefile	2017-05-04 23:02:58.479070144 +0200
@@ -126,7 +126,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/include/Makefile	2016-10-11 17:56:11.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/include/Makefile	2017-05-04 23:02:58.479070144 +0200
@@ -125,7 +125,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/inet/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/inet/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/inet/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/inet/Makefile	2017-05-04 23:02:58.479070144 +0200
@@ -129,7 +129,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/irs/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/irs/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/irs/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/irs/Makefile	2017-05-04 23:02:58.483070083 +0200
@@ -164,7 +164,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/isc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/isc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/isc/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/isc/Makefile	2017-05-04 23:02:58.483070083 +0200
@@ -129,7 +129,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/Makefile	2017-05-04 23:02:58.483070083 +0200
@@ -192,7 +192,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/nameser/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/nameser/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/nameser/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/nameser/Makefile	2017-05-04 23:02:58.483070083 +0200
@@ -125,7 +125,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/port/linux/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/port/linux/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/port/linux/include/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/port/linux/include/Makefile	2017-05-04 23:02:58.483070083 +0200
@@ -121,7 +121,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/resolv/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/resolv/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/resolv/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/bind/resolv/Makefile	2017-05-04 23:02:58.483070083 +0200
@@ -128,7 +128,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	 -D_GNU_SOURCE
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/dns/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/dns/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/dns/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/dns/Makefile	2017-05-04 23:02:58.483070083 +0200
@@ -137,7 +137,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/include/Makefile	2017-05-04 23:02:58.487070024 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/Makefile	2017-05-04 23:06:56.712891387 +0200
@@ -228,8 +228,8 @@
 ### User may define externally
 ###     EXT_CFLAGS
 
-CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CC = $(CROSS)gcc
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
@@ -434,7 +434,7 @@
 	./gen -s ${srcdir} > code.h
 
 gen: gen.c
-	gcc ${ALL_CFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
+	gcc ${EXT_CFLAGS} ${ALL_CPPFLAGS} ${ALWAYS_WARNINGS} ${STD_CWARNINGS} ${CWARNINGS} -o $@ ${srcdir}/gen.c ${LIBS}
 
 rbtdb64.lo: rbtdb.c
 
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/dst/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/dst/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/dst/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/dst/Makefile	2017-05-04 23:02:58.487070024 +0200
@@ -123,7 +123,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/include/Makefile	2017-05-04 23:02:58.487070024 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/dst/Makefile	2017-05-04 23:02:58.487070024 +0200
@@ -187,7 +187,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/dns/sec/Makefile	2017-05-04 23:02:58.487070024 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/isc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/isc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/isc/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/isc/Makefile	2017-05-04 23:02:58.491069965 +0200
@@ -139,7 +139,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/include/Makefile	2017-05-04 23:02:58.491069965 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/Makefile	2017-05-04 23:02:58.499069846 +0200
@@ -178,7 +178,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nls/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nls/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nls/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nls/Makefile	2017-05-04 23:02:58.491069965 +0200
@@ -131,7 +131,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/isc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/isc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/isc/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/isc/Makefile	2017-05-04 23:02:58.491069965 +0200
@@ -123,7 +123,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/include/Makefile	2017-05-04 23:02:58.495069905 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/nothreads/Makefile	2017-05-04 23:02:58.495069905 +0200
@@ -132,7 +132,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/isc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/isc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/isc/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/isc/Makefile	2017-05-04 23:02:58.495069905 +0200
@@ -124,7 +124,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/include/Makefile	2017-05-04 23:02:58.495069905 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isc/unix/Makefile	2017-05-04 23:02:58.495069905 +0200
@@ -142,7 +142,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/isccc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/isccc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/isccc/Makefile	2016-10-11 17:56:10.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/isccc/Makefile	2017-05-04 23:02:58.499069846 +0200
@@ -128,7 +128,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/Makefile	2016-10-11 17:56:11.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/include/Makefile	2017-05-04 23:02:58.499069846 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/Makefile	2016-10-11 17:56:11.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccc/Makefile	2017-05-04 23:02:58.499069846 +0200
@@ -198,7 +198,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/isccfg/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/isccfg/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/isccfg/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/isccfg/Makefile	2017-05-04 23:02:58.499069846 +0200
@@ -128,7 +128,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/include/Makefile	2017-05-04 23:02:58.499069846 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/isccfg/Makefile	2017-05-04 23:02:58.503069787 +0200
@@ -195,7 +195,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/lwres/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/lwres/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/lwres/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/lwres/Makefile	2017-05-04 23:02:58.503069787 +0200
@@ -127,7 +127,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/include/Makefile	2017-05-04 23:02:58.503069787 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/Makefile	2017-05-04 23:02:58.507069727 +0200
@@ -194,7 +194,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/man/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/man/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/man/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/man/Makefile	2017-05-04 23:02:58.503069787 +0200
@@ -118,7 +118,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/lwres/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/lwres/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/lwres/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/lwres/Makefile	2017-05-04 23:02:58.507069727 +0200
@@ -120,7 +120,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/include/Makefile	2017-05-04 23:02:58.507069727 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/Makefile	2016-10-11 17:56:12.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/lwres/unix/Makefile	2017-05-04 23:02:58.507069727 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/Makefile	2016-10-11 17:56:13.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/Makefile	2017-05-04 23:02:58.511069668 +0200
@@ -123,7 +123,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/Makefile	2016-10-11 17:56:09.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/Makefile	2017-05-04 23:02:58.507069727 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/tests/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/tests/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/tests/Makefile	2016-10-11 17:56:09.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/include/tests/Makefile	2017-05-04 23:02:58.507069727 +0200
@@ -121,7 +121,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/Makefile	2016-10-11 17:56:09.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/lib/tests/Makefile	2017-05-04 23:02:58.511069668 +0200
@@ -174,7 +174,7 @@
 ###     EXT_CFLAGS
 
 CC = 		$(CROSS)gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/make/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/make/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/make/Makefile	2016-10-11 17:56:14.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/make/Makefile	2017-05-04 23:02:58.511069668 +0200
@@ -119,7 +119,7 @@
 ###     EXT_CFLAGS
 
 CC = 		/opt/toolchains/uclibc-crosstools-gcc-4.4.2-1-with-ftw/usr/bin/mips-linux-gcc
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS)  
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
diff -urN D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/Makefile D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/Makefile
--- D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/Makefile	2016-10-11 17:56:15.000000000 +0200
+++ D7000_V1.0.1.84_WW_src/Source/apps/bind-9.2.3/Makefile	2017-05-04 23:05:15.405514719 +0200
@@ -122,7 +122,7 @@
 ###     EXT_CFLAGS
 
 CC = $(CROSS)gcc		
-CFLAGS =	-g -O2
+CFLAGS := $(CFLAGS) -Os
 STD_CINCLUDES =	
 STD_CDEFINES =	
 STD_CWARNINGS =	 -W -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wformat
@@ -275,11 +275,13 @@
 
 install:: isc-config.sh installdirs
 	${INSTALL_SCRIPT} isc-config.sh ${DESTDIR}${bindir}
-	$(STRIP) ${bindir}/dig
-	cp -a ${bindir}/dig ../../target/usr/sbin/ 
-	cp -a ${libdir}/libdns* ../../target/lib/ 
-	cp -a ${libdir}/libisc* ../../target/lib/ 
-	rm -rf ../../target/lib/*.a
+	$(STRIP) ${libdir}/libdns*.so
+	$(STRIP) ${libdir}/libisc*.so
+	$(SSTRIP) ${bindir}/dig
+	sudo chgrp -h root ${libdir}/libdns*.so* ${libdir}/libisc*.so*
+	sudo chmod 0755 ${libdir}/libdns*.so* ${libdir}/libisc*.so*
+	cp -a ${libdir}/libdns*.so* ${libdir}/libisc*.so* ../../target/lib/
+	cp -a ${bindir}/dig ../../target/usr/sbin/
 	
 tags:
 	rm -f TAGS
