diff -urN work/dnsmasq-2.80/src/cache.c work/dnsmasq-2.80/src/cache.c
--- work/dnsmasq-2.80/src/cache.c
+++ work/dnsmasq-2.80/src/cache.c
@@ -742,8 +742,7 @@ int cache_recv_insert(time_t now, int fd
 	   else  if (flags & F_DS)
 	     {
 	        if (!read_write(fd, (unsigned char *)&class, sizeof(class), 1) ||
-		   (flags & F_NEG) ||
-		    !(addr.key.keydata = blockdata_read(fd, addr.key.keylen)))
+		    (!(flags & F_NEG) && !(addr.key.keydata = blockdata_read(fd, addr.key.keylen))))
 		  return 0;
 	     }
 #endif
