diff -urN work/dnsmasq-2.80/src/cache.c work/dnsmasq-2.80/src/cache.c
--- work/dnsmasq-2.80/src/cache.c
+++ work/dnsmasq-2.80/src/cache.c
@@ -312,7 +312,7 @@ static int is_outdated_cname_pointer(str
   /* NB. record may be reused as DS or DNSKEY, where uid is 
      overloaded for something completely different */
   if (crecp->addr.cname.target.cache && 
-      (crecp->addr.cname.target.cache->flags & (F_IPV4 | F_IPV6 | F_CNAME)) &&
+      !(crecp->addr.cname.target.cache->flags & (F_DNSKEY | F_DS)) &&
       crecp->addr.cname.uid == crecp->addr.cname.target.cache->uid)
     return 0;
   
