diff -urN D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/libflash/Makefile D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/libflash/Makefile
--- D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/libflash/Makefile	2016-10-11 17:56:23.000000000 +0200
+++ D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/libflash/Makefile	2017-04-29 23:46:23.214479173 +0200
@@ -1,18 +1,18 @@
 include ../../../Rules.mak
 
-CFLAGS  = -Os -s -Wall -Werror -Wno-unused-but-set-variable -MD
-CFLAGS += -I$(SHARED_DIR)
-CFLAGS += -I../include
+MFLAGS = -Os -s -Wall -Werror -Wno-unused-but-set-variable -MD
+MFLAGS += -I$(SHARED_DIR)
+MFLAGS += -I../include
 
 ifeq ($(BOOT_NAND_FLASH),1)
-CFLAGS += -DBOOT_NAND_FLASH
+MFLAGS += -DBOOT_NAND_FLASH
 endif
 
 ifeq ($(CHIP_VENDOR),BCM)
-CFLAGS += -DBCM_BSP
+MFLAGS += -DBCM_BSP
 else
 ifeq ($(CHIP_VENDOR),RTL)
-CFLAGS += -DRTL_BSP
+MFLAGS += -DRTL_BSP
 endif
 endif
 
@@ -26,10 +26,11 @@
 all: $(TARGET)
 
 libflash.so : $(OBJS)
-	$(CC) -Os -s -shared -Wl,-soname,libflash.so -o $@ $(LIBS) $^
+	$(CC) $(CFLAGS) -Os -s -shared -Wl,-soname,libflash.so -o $@ $(LIBS) $^
+	$(STRIP) $@
 
 %.o : %.c
-	$(CC) $(CFLAGS) -fPIC -c $< -o $@
+	$(CC) $(CFLAGS) $(MFLAGS) -fPIC -c $< -o $@
 
 install :
 	cp -f $(TARGET) $(DESTLIB)
diff -urN D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/liblink/Makefile D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/liblink/Makefile
--- D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/liblink/Makefile	2016-10-11 17:56:23.000000000 +0200
+++ D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/liblink/Makefile	2017-04-29 23:45:36.198684719 +0200
@@ -1,35 +1,34 @@
 include ../../../Rules.mak
 
-CFLAGS  = -Os -s -Wall -Werror -MD -Wno-strict-aliasing
-CFLAGS += -I$(SHARED_DIR) -I../include
-CFLAGS += -I$(ROOT)../Kernel/bcm963xx/bcmdrivers/broadcom/include/bcm963xx
-CFLAGS += -I$(ROOT)/apps/nvram
-#CFLAGS += -D_SC_DEBUG_ENABLE_
+MFLAGS  = -Os -s -Wall -Werror -MD -Wno-strict-aliasing
+MFLAGS += -I$(SHARED_DIR) -I../include
+MFLAGS += -I$(ROOT)../Kernel/bcm963xx/bcmdrivers/broadcom/include/bcm963xx
+MFLAGS += -I$(ROOT)/apps/nvram
 
 LDFLAGS += -L$(ROOT)/apps/nvram -lscnvram
 
 ifeq ($(CHIP_VENDOR),BCM)
-CFLAGS += -DBCM_BSP
+MFLAGS += -DBCM_BSP
 else
 ifeq ($(CHIP_VENDOR),RTL)
-CFLAGS += -DRTL_BSP
+MFLAGS += -DRTL_BSP
 endif
 endif
 
 ifeq ($(EXTERNAL_SWITCH),1)
-CFLAGS += -DEXTERNAL_SWITCH
+MFLAGS += -DEXTERNAL_SWITCH
 endif
 
 ifeq ($(DUAL_WAN),1)
-CFLAGS += -DDUAL_WAN
+MFLAGS += -DDUAL_WAN
 endif
 
 ifeq ($(3G_FEATURE),1)
-CFLAGS += -D_3G_FEATURE_
+MFLAGS += -D_3G_FEATURE_
 endif
 
 ifeq ($(MTCODE),1)
-CFLAGS += -DMT_CODE
+MFLAGS += -DMT_CODE
 endif
 
 DESTLIB = $(TARGET_HOME)/target/lib
@@ -40,15 +39,16 @@
 all: $(TARGET)
 
 liblink.so : $(OBJS)
-	$(CC) -Os -s $(LDFLAGS) -shared -Wl,-soname,liblink.so -o $@ $(LIBS) $^
+	$(CC) $(CFLAGS) -Os -s $(LDFLAGS) -shared -Wl,-soname,liblink.so -o $@ $(LIBS) $^
+	$(STRIP) $@
+
 test: main.o 
-	$(CC) -Os -s  -o $@ $(LIBS) $^
+	$(CC) $(CFLAGS) -Os -s  -o $@ $(LIBS) $^
 
 main.o: main.c lib_adslinfo.c
 
 %.o : %.c
-	echo $(CFLAGS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -fPIC -c $< -o $@
+	$(CC) $(CFLAGS) $(MFLAGS) $(LDFLAGS) -fPIC -c $< -o $@
 
 install :
 	cp -f $(TARGET) $(DESTLIB)
diff -urN D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/Makefile D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/Makefile
--- D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/Makefile	2016-10-11 17:56:23.000000000 +0200
+++ D7000_V1.0.1.54_WW_src/Source/apps/sc_libs/Makefile	2017-04-29 23:43:36.779402732 +0200
@@ -1,8 +1,6 @@
 include ../../Rules.mak
 export CC=$(CROSS)gcc
 
-CFLAGS  = -Os -s -Wall
-
 DESTLIB = ../../target/lib
 DESTBIN = ../../target/usr/sbin
 
