diff -urN work/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/src/nettest_bsd.c work/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/src/nettest_bsd.c
--- work/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/src/nettest_bsd.c	2018-06-13 23:32:44.000000000 +0200
+++ work/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/netperf-bcb868bde7f0203bbab69609f65d4088ba7398db/src/nettest_bsd.c	2020-11-27 22:23:28.443573000 +0100
@@ -4666,7 +4666,7 @@
 
       /* you can look at netlib.h for a description of the fields we
 	 are passing to sendfile(). 08/2000 */
-      if (netperf_sendfile(send_socket, send_ring) != send_size) {
+      if (len = netperf_sendfile(send_socket, send_ring) != send_size) {
 	/* the test was interrupted, must be the end of test. the
 	   send_tcp_stream code has some WIN32 ifdefs that we do not
 	   need here. */
