--- work/sqlite-src-3380500/sqlite-src-3380500/Makefile.in
+++ work/sqlite-src-3380500/sqlite-src-3380500/Makefile.in
@@ -636,7 +636,7 @@
 # This is the default Makefile target.  The objects listed here
 # are what get build when you type just "make" with no arguments.
 #
-all:	sqlite3.h libsqlite3.la sqlite3$(TEXE) $(HAVE_TCL:1=libtclsqlite3.la)
+all:	sqlite3.h libsqlite3.la $(HAVE_TCL:1=libtclsqlite3.la)
 
 Makefile: $(TOP)/Makefile.in
 	./config.status
