diff -urN D7000_V1.0.1.64_WW_src/Source/apps/mediaserver/library/zlib-1.2.3/Makefile D7000_V1.0.1.64_WW_src/Source/apps/mediaserver/library/zlib-1.2.3/Makefile
--- D7000_V1.0.1.64_WW_src/Source/apps/mediaserver/library/zlib-1.2.3/Makefile	2016-10-13 17:51:19.000000000 +0200
+++ D7000_V1.0.1.64_WW_src/Source/apps/mediaserver/library/zlib-1.2.3/Makefile	2017-05-05 14:57:20.115087727 +0200
@@ -22,7 +22,7 @@
 
 #CC=gcc
 
-CFLAGS=-fPIC -Os -s -DUSE_MMAP
+CFLAGS +=  -fPIC -Os -s -DUSE_MMAP
 #CFLAGS=-O -DMAX_WBITS=14 -DMAX_MEM_LEVEL=7
 #CFLAGS=-g -DDEBUG
 #CFLAGS=-O3 -Wall -Wwrite-strings -Wpointer-arith -Wconversion \
@@ -37,8 +37,8 @@
 SHAREDLIBV=libz.so.1.2.3
 SHAREDLIBM=libz.so.1
 
-AR=ar rc
-RANLIB=ranlib
+AR=$(CROSS)ar rc
+RANLIB=$(CROSS)ranlib
 TAR=tar
 SHELL=/bin/sh
 EXE=
