diff -urN D7000_V1.0.1.70_WW_src/Source/apps/mediaserver/library/sqlite-3.5.9/Makefile D7000_V1.0.1.70_WW_src/Source/apps/mediaserver/library/sqlite-3.5.9/Makefile
--- D7000_V1.0.1.70_WW_src/Source/apps/mediaserver/library/sqlite-3.5.9/Makefile	2016-10-13 17:51:19.000000000 +0200
+++ D7000_V1.0.1.70_WW_src/Source/apps/mediaserver/library/sqlite-3.5.9/Makefile	2017-05-05 14:57:20.115087727 +0200
@@ -26,12 +26,12 @@
 # C Compiler and options for use in building executables that
 # will run on the platform that is doing the build.
 #
-BCC = gcc -s -Os
+BCC = gcc $(CFLAGS) -s -Os
 # C Compile and options for use in building executables that 
 # will run on the target platform.  (BCC and TCC are usually the
 # same unless your are cross-compiling.)
 #
-TCC = $(CC) -s -Os -DOS_UNIX=1 -DHAVE_USLEEP=1 -DHAVE_FDATASYNC=1 -I. -I${TOP}/src
+TCC = $(CC) $(CFLAGS) -s -Os -DOS_UNIX=1 -DHAVE_USLEEP=1 -DHAVE_FDATASYNC=1 -I. -I${TOP}/src
 
 # Define this for the autoconf-based build, so that the code knows it can
 # include the generated config.h
