diff -urN D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/minidlna/Makefile D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/minidlna/Makefile
--- D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/minidlna/Makefile	2016-10-13 17:51:19.000000000 +0200
+++ D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/minidlna/Makefile	2017-05-05 14:57:20.115087727 +0200
@@ -3,7 +3,7 @@
 DESTBIN = $(ROOT)/target/usr/sbin
 FFMPEG_DIR=$(ROOT)/apps/mediaserver/library/ffmpeg/ffmpeg-3.0.1
 
-CFLAGS = -s -Os -fPIC
+CFLAGS += -s -Os -fPIC
 CFLAGS += -I$(ROOT)/apps/mediaserver/library/include/ -I$(FFMPEG_DIR)/include/ -I$(FFMPEG_DIR)/
 CFLAGS += -I$(FFMPEG_DIR)/libavformat/
 CFLAGS += -I$(FFMPEG_DIR)/libavutil/
@@ -15,9 +15,7 @@
 CFLAGS += -I$(ROOT)/apps/mediaserver/library/libogg-1.3.0/include/
 CFLAGS += -I$(ROOT)/apps/mediaserver/library/libexif-0.6.20/
 CFLAGS += -I$(ROOT)/apps/libiconv.1.8/include/
-#CFLAGS += -I/opt/toolchains/crosstools-arm-gcc-4.6-linux-3.4-uclibc-0.9.32-binutils-2.21-NPTL/usr/arm-unknown-linux-uclibcgnueabi/sysroot/usr/include
-CFLAGS += -I$(ROOT)/../Kernel/bcm963xx_4.16L.03_data_src/kernel/linux-3.4rt/include
-
+CFLAGS += -I$(ROOT)/../Kernel/bcm963xx/kernel/linux-3.4rt/include
 
 LDFLAGS =  -L$(ROOT)/apps/mediaserver/library/lib/
 LDFLAGS += -L$(FFMPEG_DIR)/libavformat/
@@ -45,10 +43,10 @@
 	
 configure:
 	cd minidlna-1.1.5 && touch aclocal.m4 compile config.guess  config.h.in config.rpath config.sub configure depcomp INSTALL install-sh Makefile.in
-	cd minidlna-1.1.5 && ./configure --enable-tivo --host=arm-linux CC=$(CROSS)gcc CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" LIBS="-lz -lexif -ljpeg -lsqlite3  -lavformat -lavcodec  -lavutil -lid3tag -lFLAC -logg -lvorbis -liconv"
+	cd minidlna-1.1.5 && ./configure --enable-tivo --host=arm-linux CC=$(CROSS)gcc CFLAGS="$(CFLAGS) -fcommon" LDFLAGS="$(LDFLAGS)" LIBS="-lz -lexif -ljpeg -lsqlite3  -lavformat -lavcodec  -lavutil -lid3tag -lFLAC -logg -lvorbis -liconv"
 
 install:	
-	$(STRIP) minidlna-1.1.5/minidlnad
+	$(SSTRIP) minidlna-1.1.5/minidlnad
 	cp -af minidlna-1.1.5/minidlnad ../../../target/usr/sbin/minidlna
 	$(INSTALL) --mode=0644 minidlna-1.1.5/minidlna.conf ../../../target/usr/etc/
 
