diff -urN D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/Makefile D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/Makefile
--- D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/Makefile	2016-10-13 17:51:19.000000000 +0200
+++ D7000_V1.0.1.48_WW_src/Source/apps/mediaserver/Makefile	2017-05-05 14:57:20.115087727 +0200
@@ -75,12 +75,6 @@
 install:
 	for i in ${SUBLIBS} ; do make -C $$i install || exit 1 ; done
 	for i in ${SUBDIRS} ; do make -C $$i install || exit 1 ; done
-ifeq ($(USE_MINIDLNA_VERSION),1.0.26)
-	#$(STRIP) $(FFMPEG_DIR)/libavutil/libavutil.so* $(FFMPEG_DIR)/libavformat/libavformat.so* $(FFMPEG_DIR)/libavcodec/libavcodec.so.*
-	#cp -a $(FFMPEG_DIR)/libavutil/libavutil.so* ../../target/lib/
-	#cp -a $(FFMPEG_DIR)/libavformat/libavformat.so* ../../target/lib/
-	#cp -a $(FFMPEG_DIR)/libavcodec/libavcodec.so.* ../../target/lib/
-endif
 	$(STRIP) ../../target/lib/libexif.so.* 
 	$(STRIP) ../../target/lib/libFLAC.so.* 
 	$(STRIP) ../../target/lib/libid3tag.so.*  
