diff -urN D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.mak D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.mak
--- D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.mak	2016-10-11 17:55:16.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.mak	2017-05-04 20:06:18.838808358 +0200
@@ -21,7 +21,7 @@
 CP=cp -p
 LN_S=ln -sf
 CPPFLAGS= -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200112 -D_XOPEN_SOURCE=600 -DPIC
-CFLAGS= -fPIC  -std=c99 -fomit-frame-pointer -fPIC -pthread -D_GNU_SOURCE=1 -D_REENTRANT -Wdeclaration-after-statement -Wall -Wno-parentheses -Wno-switch -Wno-format-zero-length -Wdisabled-optimization -Wpointer-arith -Wredundant-decls -Wno-pointer-sign -Wwrite-strings -Wtype-limits -Wundef -Wmissing-prototypes -Wno-pointer-to-int-cast -Wstrict-prototypes -Os -fno-math-errno -fno-signed-zeros -fno-tree-vectorize -Werror=implicit-function-declaration -Werror=missing-prototypes
+CFLAGS+= -fPIC  -std=c99 -fomit-frame-pointer -fPIC -pthread -D_GNU_SOURCE=1 -D_REENTRANT -Wdeclaration-after-statement -Wall -Wno-parentheses -Wno-switch -Wno-format-zero-length -Wdisabled-optimization -Wpointer-arith -Wredundant-decls -Wno-pointer-sign -Wwrite-strings -Wtype-limits -Wundef -Wmissing-prototypes -Wno-pointer-to-int-cast -Wstrict-prototypes -Os -fno-math-errno -fno-signed-zeros -fno-tree-vectorize -Werror=implicit-function-declaration -Werror=missing-prototypes
 CXXFLAGS=  -D__STDC_CONSTANT_MACROS
 ASFLAGS= -fPIC  -fPIC
 AS_O=-o $@
diff -urN D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/library.mak D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/library.mak
--- D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/library.mak	2016-10-11 17:55:16.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/library.mak	2017-05-04 20:11:43.261393787 +0200
@@ -35,14 +35,14 @@
 
 define RULES
 $(EXAMPLES) $(TESTPROGS) $(TOOLS): %$(EXESUF): %.o
-	$$(LD) $(LDFLAGS) -o $$@ $$^ -l$(FULLNAME) $(FFEXTRALIBS) $$(ELIBS)
+	$$(CC) $(LDFLAGS) -o $$@ $$^ -l$(FULLNAME) $(FFEXTRALIBS) $$(ELIBS)
 
 $(SUBDIR)$(SLIBNAME): $(SUBDIR)$(SLIBNAME_WITH_MAJOR)
 	$(Q)cd ./$(SUBDIR) && $(LN_S) $(SLIBNAME_WITH_MAJOR) $(SLIBNAME)
 
 $(SUBDIR)$(SLIBNAME_WITH_MAJOR): $(OBJS) $(SUBDIR)lib$(NAME).ver
 	$(SLIB_CREATE_DEF_CMD)
-	$$(LD) $(SHFLAGS) $(LDFLAGS) -o $$@ $$(filter %.o,$$^) $(FFEXTRALIBS) $(EXTRAOBJS)
+	$$(CC) $(SHFLAGS) $(LDFLAGS) -o $$@ $$(filter %.o,$$^) $(FFEXTRALIBS) $(EXTRAOBJS)
 	$(SLIB_EXTRA_CMD)
 
 ifdef SUBDIR
diff -urN D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/Makefile D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/Makefile
--- D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/Makefile	2016-10-13 17:51:19.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/Makefile	2017-05-05 14:57:20.115087727 +0200
@@ -56,7 +56,7 @@
 	$(STRIP) $@$(PROGSSUF)
 
 $(TOOLS): %$(EXESUF): %.o
-	$(LD) $(LDFLAGS) -o $@ $< $(ELIBS)
+	$(CC) $(LDFLAGS) -o $@ $< $(ELIBS)
 
 tools/cws2fws$(EXESUF): ELIBS = -lz
 
@@ -92,7 +92,7 @@
 ffserver_g$(EXESUF): LDFLAGS += $(FFSERVERLDFLAGS)
 
 %$(PROGSSUF)_g$(EXESUF): %.o cmdutils.o $(FF_DEP_LIBS)
-	$(LD) $(LDFLAGS) -o $@ $< cmdutils.o $(FF_EXTRALIBS)
+	$(CC) $(LDFLAGS) -o $@ $< cmdutils.o $(FF_EXTRALIBS)
 
 OBJDIRS += tools
 
diff -urN D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.h D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.h
--- D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.h	2016-10-11 17:55:16.000000000 +0200
+++ D7000_V1.0.1.74_WW_src/Source/apps/mediaserver/library/ffmpeg-0.11.2/config.h	2017-05-04 18:16:55.628951043 +0200
@@ -7,7 +7,7 @@
 #define AVCONV_DATADIR "/usr/local/share/ffmpeg"
 #define CC_TYPE "gcc"
 #define CC_VERSION __VERSION__
-#define restrict restrict
+#define restrict
 #define EXTERN_PREFIX ""
 #define EXTERN_ASM 
 #define SLIBSUF ".so"
