diff -urN D7000_V1.0.1.78_WW_src/Source/apps/mediaserver/library/flac-1.2.1/Makefile D7000_V1.0.1.78_WW_src/Source/apps/mediaserver/library/flac-1.2.1/Makefile
--- D7000_V1.0.1.78_WW_src/Source/apps/mediaserver/library/flac-1.2.1/Makefile	2016-10-13 17:51:19.000000000 +0200
+++ D7000_V1.0.1.78_WW_src/Source/apps/mediaserver/library/flac-1.2.1/Makefile	2017-05-05 14:57:20.115087727 +0200
@@ -81,7 +81,7 @@
 CCAS = $(CC)
 CCASFLAGS = $(CFLAGS)
 CCDEPMODE = depmode=gcc3
-CFLAGS = -O3 -funroll-loops -finline-functions -Wall -W -Winline -g -O2
+CFLAGS +=  -Os -funroll-loops -finline-functions -Wall -W -Winline
 CPP = gcc -E
 CPPFLAGS = -DFLaC__INLINE=__inline__ -DNDEBUG -I$(top_builddir) -I$(srcdir)/include -I$(top_srcdir)/include 
 CXX = g++
