diff -urN work/mini-snmpd-1.6/Makefile work/mini-snmpd-1.6/Makefile
--- work/mini-snmpd-1.6/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ work/mini-snmpd-1.6/Makefile	2018-01-05 12:00:11.000000000 +0100
@@ -0,0 +1,58 @@
+include ../Rules.mak
+
+BINS			:= $(MINI_SNMPD)/mini-snmpd
+STARTUP			:= rc_snmp
+PRIOS			:= 11
+PRIOF			:= 11
+WEB			:= ../www/$(MINI_SNMPD_NAME).htm
+
+all: config
+	$(MAKE) -C $(MINI_SNMPD)
+
+config:
+	@if [ ! -f h_config ]; then \
+	$(MAKE) configure || exit 1; \
+	touch h_config; \
+	fi
+
+configure:
+	cd $(MINI_SNMPD) && ./configure --build=$(BUILD_HOST) --host=$(TARGET_NAME) CFLAGS="$(CFLAGS) -Os -s" --without-config --without-systemd
+
+install: control
+	$(SSTRIP) $(BINS)
+	for D in $(BINDIR) $(APPDIR) $(WWWDIR) $(RCSDIR) $(FWDIR); do install -m 0755 -d $$D; done
+	install -m 0755 $(BINS) $(BINDIR)
+	install -m 0755 $(STARTUP) $(APPDIR)
+	install -m 0664 $(WEB) $(WWWDIR)
+	ln -sf $(USRSBINRCAPP)/$(STARTUP) $(RCSDIR)/S$(PRIOS)$(MINI_SNMPD_NAME)
+	ln -sf $(USRSBINRCAPP)/$(STARTUP) $(FWDIR)/F$(PRIOF)$(MINI_SNMPD_NAME)
+
+clean:
+	$(MAKE) -C $(MINI_SNMPD) clean
+
+distclean:
+	@rm -f h_config
+	$(MAKE) -C $(MINI_SNMPD) distclean
+
+ifndef BUILD
+control:
+else
+POSTINST		:= $(CTRL_DIR)/postinst
+PRERM			:= $(CTRL_DIR)/prerm
+
+control: postinst prerm
+
+postinst:
+	@echo "#!/bin/sh" > $(POSTINST)
+	@echo "ln -sf $(USRSBINRCAPP)/$(STARTUP) $(TMPETC)/$(INITD)/$(RCSD)/S$(PRIOS)$(MINI_SNMPD_NAME)" >> $(POSTINST)
+	@echo "ln -sf $(USRSBINRCAPP)/$(STARTUP) $(TMPETC)/$(INITD)/$(RCFW)/F$(PRIOF)$(MINI_SNMPD_NAME)" >> $(POSTINST)
+	@echo "$(USRSBINRCAPP)/$(STARTUP) start" >> $(POSTINST)
+	@echo "exit 0" >> $(POSTINST)
+
+prerm:
+	@echo "#!/bin/sh" > $(PRERM)
+	@echo "rm -f $(TMPETC)/$(INITD)/$(RCSD)/S$(PRIOS)$(MINI_SNMPD_NAME) $(TMPETC)/$(INITD)/$(RCFW)/F$(PRIOF)$(MINI_SNMPD_NAME)" >> $(PRERM)
+	@echo "$(USRSBINRCAPP)/$(STARTUP) stop" >> $(PRERM)
+	@echo "exit 0" >> $(PRERM)
+endif
+
