diff -urN work/opkg-0.3.4/opkg-0.3.4/src/opkg.c work/opkg-0.3.4/opkg-0.3.4/src/opkg.c
--- work/opkg-0.3.4/opkg-0.3.4/src/opkg.c	2016-12-22 18:38:14.000000000 +0100
+++ work/opkg-0.3.4/opkg-0.3.4/src/opkg.c	2017-03-22 00:12:25.566249000 +0100
@@ -266,6 +266,9 @@
         }
     }
 
+	if(!opkg_config->conf_file && !opkg_config->offline_root)
+		opkg_config->conf_file = xstrdup("/etc/opkg.conf");
+
     if (parse_err)
         return parse_err;
     else
