diff -urN work/opkg-0.3.4/opkg-0.3.4/utils/opkg.conf work/opkg-0.3.4/opkg-0.3.4/utils/opkg.conf
--- work/opkg-0.3.4/opkg-0.3.4/utils/opkg.conf	1970-01-01 01:00:00.000000000 +0100
+++ work/opkg-0.3.4/opkg-0.3.4/utils/opkg.conf	2017-03-23 13:30:14.381800000 +0100
@@ -0,0 +1,12 @@
+src/gz local file:///mnt/shares/U/local
+#src/gz local file:///mnt/shares/V/local
+src/gz packages https://github.com/negan07
+dest root /
+dest ram /tmp
+#dest hdd /mnt/shares/U/
+#dest usb /mnt/shares/V/
+option lists_dir /tmp/var/opkg-lists
+arch D7000 1
+arch arm 10
+arch all 100
+
