--- work/curl-7.83.1/curl-7.83.1/lib/vtls/mbedtls.c
+++ work/curl-7.83.1/curl-7.83.1/lib/vtls/mbedtls.c
@@ -993,7 +993,7 @@
 {
 #ifdef MBEDTLS_VERSION_C
   /* if mbedtls_version_get_number() is available it is better */
-  unsigned int version = mbedtls_version_get_number();
+  unsigned int version = MBEDTLS_VERSION_NUMBER;
   return msnprintf(buffer, size, "mbedTLS/%u.%u.%u", version>>24,
                    (version>>16)&0xff, (version>>8)&0xff);
 #else
