--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/tagutils/tagutils-misc.c
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/tagutils/tagutils-misc.c
@@ -203,7 +203,12 @@
 	}
 	strncpy(strbuf, comment, length);
 	strbuf[length] = '\0';
-
+	
+	// Xiph.org lists recommended field names for interoperability between programs.
+	// Beyond these software may use other tag names, and because the files we are
+	// tasked with reading may come from a variety of sources we include other commonly
+	// used tags where we can.
+	// https://xiph.org/vorbis/doc/v-comment.html
 	// ALBUM, ARTIST, PUBLISHER, COPYRIGHT, DISCNUMBER, ISRC, EAN/UPN, LABEL, LABELNO,
 	// LICENSE, OPUS, SOURCEMEDIA, TITLE, TRACKNUMBER, VERSION, ENCODED-BY, ENCODING,
 	// -- following tags are muliples
@@ -232,6 +237,16 @@
 	{
 		psong->contributor_sort[ROLE_BAND] = strdup(strbuf + 16);
 	}
+	else if(!strncasecmp(strbuf, "COMPOSER=", 9))
+	{
+		if( *(strbuf+9) )
+			psong->contributor[ROLE_COMPOSER] = strdup(strbuf + 9);
+	}
+	else if(!strncasecmp(strbuf, "CONDUCTOR=", 10))
+	{
+		if( *(strbuf+10) )
+			psong->contributor[ROLE_CONDUCTOR] = strdup(strbuf + 10);
+	}
 	else if(!strncasecmp(strbuf, "TITLE=", 6))
 	{
 		if( *(strbuf+6) )
@@ -241,9 +256,17 @@
 	{
 		psong->track = atoi(strbuf + 12);
 	}
+	else if(!strncasecmp(strbuf, "TRACKTOTAL=", 11))
+	{
+		psong->total_tracks = atoi(strbuf + 11);
+	}
 	else if(!strncasecmp(strbuf, "DISCNUMBER=", 11))
 	{
 		psong->disc = atoi(strbuf + 11);
+	}
+	else if(!strncasecmp(strbuf, "DISCTOTAL=", 10))
+	{
+		psong->total_discs = atoi(strbuf + 10);
 	}
 	else if(!strncasecmp(strbuf, "GENRE=", 6))
 	{
