--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna.c
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna.c
@@ -729,6 +729,10 @@
 			if (!strtobool(ary_options[i].value))
 				CLEARFLAG(INOTIFY_MASK);
 			break;
+		case DISPLAY_THUMBS_ART:
+			if (strtobool(ary_options[i].value))
+				SETFLAG(THUMBS_ART_MASK);
+			break;
 		case ENABLE_TIVO:
 			if (strtobool(ary_options[i].value))
 				SETFLAG(TIVO_MASK);
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna.conf
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna.conf
@@ -42,6 +42,9 @@
 # note: the default is yes
 inotify=yes
 
+# enable sending thumbnails
+#thumbnails=yes
+
 # set this to yes to enable support for streaming .jpg and .mp3 files to a TiVo supporting HMO
 enable_tivo=no
 
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna.conf.5
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna.conf.5
@@ -141,6 +141,10 @@
 
 .fi
 
+.IP "\fBthumbnails\fP"
+Set to 'yes' to enable sending artwork from file's metadata.
+By default, artwork is being sent.
+
 .IP "\fBenable_tivo\fP"
 Set this to yes to enable support for streaming .jpg and .mp3 files to a TiVo 
 supporting HMO, default is no.
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/options.c
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/options.c
@@ -53,6 +53,7 @@
 	{ UPNPFRIENDLYNAME, "friendly_name"},
 	{ UPNPMEDIADIR, "media_dir"},
 	{ UPNPALBUMART_NAMES, "album_art_names"},
+	{ DISPLAY_THUMBS_ART, "thumbnails" },
 	{ UPNPINOTIFY, "inotify" },
 	{ UPNPDBDIR, "db_dir" },
 	{ UPNPLOGDIR, "log_dir" },
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/options.h
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/options.h
@@ -48,6 +48,7 @@
 	UPNPALBUMART_NAMES,		/* list of '/'-delimited file names to check for album art */
 	UPNPINOTIFY,			/* enable inotify on the media directories */
 	UPNPDBDIR,			/* base directory to store the database and album art cache */
+	DISPLAY_THUMBS_ART,		/* show art thumbnails */
 	UPNPLOGDIR,			/* base directory to store the log file */
 	UPNPLOGLEVEL,			/* logging verbosity */
 	UPNPMINISSDPDSOCKET,		/* minissdpdsocket */
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/upnpglobalvars.h
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/upnpglobalvars.h
@@ -209,6 +209,7 @@
 #define SUBTITLES_MASK        0x0400
 #define FORCE_ALPHASORT_MASK  0x0800
 #define NO_EXTRACT_NAMES_MASK 0x1000
+#define THUMBS_ART_MASK       0x2000
 
 #define SETFLAG(mask)	runtime_flags |= mask
 #define GETFLAG(mask)	(runtime_flags & mask)
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/monitor.c
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/monitor.c
@@ -153,7 +153,7 @@
 	struct stat st;
 
 	/* Is it cover art for another file? */
-	if (mtype == TYPE_IMAGE)
+	if (mtype == TYPE_IMAGE && GETFLAG(THUMBS_ART_MASK))
 		update_if_album_art(path);
 	else if (mtype == TYPE_CAPTION)
 		check_for_captions(path, 0);
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/metadata.c
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/metadata.c
@@ -528,7 +528,7 @@
 		}
 	}
 
-	album_art = find_album_art(path, song.image, song.image_size);
+	if(GETFLAG(THUMBS_ART_MASK)) album_art = find_album_art(path, song.image, song.image_size);
 
 	ret = sql_exec(db, "INSERT into DETAILS"
 	                   " (PATH, SIZE, TIMESTAMP, DURATION, CHANNELS, BITRATE, SAMPLERATE, DATE,"
@@ -1643,7 +1643,7 @@
 		}
 	}
 
-	album_art = find_album_art(path, m.thumb_data, m.thumb_size);
+	if(GETFLAG(THUMBS_ART_MASK)) album_art = find_album_art(path, m.thumb_data, m.thumb_size);
 	freetags(&video);
 	lav_close(ctx);
 
--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/scanner.c
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/scanner.c
@@ -435,7 +435,7 @@
 		return 0;
 	}
 
-	detailID = GetFolderMetadata(name, path, NULL, NULL, find_album_art(path, NULL, 0));
+	detailID = GetFolderMetadata(name, path, NULL, NULL, (GETFLAG(THUMBS_ART_MASK) ? find_album_art(path, NULL, 0) : 0) );
 	sql_exec(db, "INSERT into OBJECTS"
 	             " (OBJECT_ID, PARENT_ID, DETAIL_ID, CLASS, NAME) "
 	             "VALUES"
