--- work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/upnphttp.c
+++ work/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/minidlna-git-66939d5ac4986020764c0518455f55b68422369f/upnphttp.c
@@ -2002,7 +2002,7 @@ SendResp_dlnafile(struct upnphttp *h, ch
 	{
 		if( strncmp(last_file.mime, "image", 5) == 0 )
 		{
-			DPRINTF(E_WARN, L_HTTP, "Client tried to specify transferMode as Streaming with an image!\n");
+			DPRINTF(E_WARN, L_HTTP, "Client tried to specify transferMode as Streaming with an image! [%s]\n", last_file.path);
 			Send406(h);
 			goto error;
 		}
@@ -2011,13 +2011,13 @@ SendResp_dlnafile(struct upnphttp *h, ch
 	{
 		if( h->reqflags & FLAG_REALTIMEINFO )
 		{
-			DPRINTF(E_WARN, L_HTTP, "Bad realTimeInfo flag with Interactive request!\n");
+			DPRINTF(E_WARN, L_HTTP, "Bad realTimeInfo flag with Interactive request! [%s]\n", last_file.path);
 			Send400(h);
 			goto error;
 		}
 		if( strncmp(last_file.mime, "image", 5) != 0 )
 		{
-			DPRINTF(E_WARN, L_HTTP, "Client tried to specify transferMode as Interactive without an image!\n");
+			DPRINTF(E_WARN, L_HTTP, "Client tried to specify transferMode as Interactive without an image! [%s]\n", last_file.path);
 			/* Samsung TVs (well, at least the A950) do this for some reason,
 			 * and I don't see them fixing this bug any time soon. */
 			if( !(cflags & FLAG_SAMSUNG) || GETFLAG(DLNA_STRICT_MASK) )
@@ -2062,14 +2062,14 @@ SendResp_dlnafile(struct upnphttp *h, ch
 		}
 		if( (h->req_RangeStart > h->req_RangeEnd) || (h->req_RangeStart < 0) )
 		{
-			DPRINTF(E_WARN, L_HTTP, "Specified range was invalid!\n");
+			DPRINTF(E_WARN, L_HTTP, "Specified range was invalid! [%s]\n", last_file.path);
 			Send400(h);
 			close(sendfh);
 			goto error;
 		}
 		if( h->req_RangeEnd >= size )
 		{
-			DPRINTF(E_WARN, L_HTTP, "Specified range was outside file boundaries!\n");
+			DPRINTF(E_WARN, L_HTTP, "Specified range was outside file boundaries! [%s]\n", last_file.path);
 			Send416(h);
 			close(sendfh);
 			goto error;
